From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Manganiello <adamantike@users.noreply.github.com>
Date: Sat, 26 Mar 2022 13:15:31 -0300
Subject: [PATCH] Teach the multiple configure files that serenity is a thing

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index ba131b5fedb26d83625f3a91584f01aecf8ca4c4..fdb47e5c9eb594156a48904014f37f58ae4397b0 100755
--- a/configure
+++ b/configure
@@ -14823,7 +14823,7 @@ case $host in
     cpu_type="x86"
     newoldwritesample=enabled
   ;;
-  x86_64-*-linux*|x86_64-*-kfreebsd*-gnu)
+  x86_64-*-linux*|x86_64-*-kfreebsd*-gnu|x86_64-*-serenity*)
     cpu_type="x86-64"
   ;;
   *-*-linux*|*-*-kfreebsd*-gnu)
