From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:19:42 +0200
Subject: [PATCH] Add SerenityOS platform

---
 qb/qb.system.sh | 5 +++--
 1 file changed, 3 insertions(+), 2 deletions(-)

diff --git a/qb/qb.system.sh b/qb/qb.system.sh
index c7ac14110bafc0c1715a5b02fc933dc7c45dc449..f7f198dc9c290d5c46079cedd13c5cbb42bb678b 100644
--- a/qb/qb.system.sh
+++ b/qb/qb.system.sh
@@ -1,5 +1,5 @@
-PLATFORM_NAME="$(uname -s)"
-ARCHITECTURE_NAME="$(uname -m)"
+PLATFORM_NAME="SerenityOS"
+ARCHITECTURE_NAME="${SERENITY_ARCH}"
 
 if [ -n "${CROSS_COMPILE:=}" ]; then
 	case "$CROSS_COMPILE" in
@@ -18,6 +18,7 @@ if [ -z "$CROSS_COMPILE" ] || [ -z "$OS" ]; then
 		'Haiku') OS='Haiku';;
 		'MINGW'*) OS='Win32';;
 		'SunOS') OS='SunOS';;
+		'SerenityOS') OS='SerenityOS';;
 		*) OS="Win32";;
 	esac
 fi
