From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:20:35 +0200
Subject: [PATCH] Add strlcat()

---
 Makefile.common | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile.common b/Makefile.common
index 2a715f4786a2714f32cb127691e6ed1a390b5188..3b0b4ca0b8f63d3ab79938005934df178e9c446f 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -184,6 +184,10 @@ ifneq ($(findstring Linux,$(OS)),)
    HAVE_UNIX = 1
 endif
 
+ifneq ($(findstring SerenityOS,$(OS)),)
+	OBJ += $(LIBRETRO_COMM_DIR)/compat/compat_strl.o
+endif
+
 ifeq ($(HAVE_UNIX), 1)
    OBJ += frontend/drivers/platform_unix.o
 
