From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:20:09 +0200
Subject: [PATCH] Find libgl

---
 qb/config.libs.sh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/qb/config.libs.sh b/qb/config.libs.sh
index 19cd2a6..672fb02 100644
--- a/qb/config.libs.sh
+++ b/qb/config.libs.sh
@@ -402,7 +402,7 @@ if [ "$HAVE_OPENGL" != 'no' ] && [ "$HAVE_OPENGLES" != 'yes' ]; then
       check_lib '' OPENGL -lopengl32
    else
       check_header '' OPENGL "GL/gl.h"
-      check_lib '' OPENGL -lGL
+      check_lib '' OPENGL -lgl
    fi
 
    if [ "$HAVE_OPENGL" = 'yes' ]; then
