From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:20:35 +0200
Subject: [PATCH] Add strlcat()

---
 Makefile.common | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile.common b/Makefile.common
index 40c3ad2d5a077dc8f84932506d4253c7d98e326f..f2d6bcf526a1c7d3aba90bdd4ef6cd1ebc3ca91c 100644
--- a/Makefile.common
+++ b/Makefile.common
@@ -186,6 +186,10 @@ ifneq ($(findstring Linux,$(OS)),)
    HAVE_UNIX = 1
 endif
 
+ifneq ($(findstring SerenityOS,$(OS)),)
+	OBJ += $(LIBRETRO_COMM_DIR)/compat/compat_strl.o
+endif
+
 ifeq ($(HAVE_UNIX), 1)
    OBJ += frontend/drivers/platform_unix.o
 
