From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:20:52 +0200
Subject: [PATCH] Disable pthread_attr_setschedpolicy()

---
 libretro-common/rthreads/rthreads.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/libretro-common/rthreads/rthreads.c b/libretro-common/rthreads/rthreads.c
index 2a748edd53a19304de81faade4bd28e3014df713..2660345dfb76688922c1300630d603f90d9164cd 100644
--- a/libretro-common/rthreads/rthreads.c
+++ b/libretro-common/rthreads/rthreads.c
@@ -158,7 +158,7 @@ sthread_t *sthread_create(void (*thread_func)(void*), void *userdata)
 }
 
 /* TODO/FIXME - this needs to be implemented for Switch/3DS */
-#if !defined(SWITCH) && !defined(USE_WIN32_THREADS) && !defined(_3DS) && !defined(GEKKO) && !defined(__HAIKU__) && !defined(EMSCRIPTEN)
+#if !defined(SWITCH) && !defined(USE_WIN32_THREADS) && !defined(_3DS) && !defined(GEKKO) && !defined(__HAIKU__) && !defined(EMSCRIPTEN) && !defined(__serenity__)
 #define HAVE_THREAD_ATTR
 #endif
 
