From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gouchi <gouchi@free.fr>
Date: Sat, 7 May 2022 18:21:50 +0200
Subject: [PATCH] Use SDL software instead of hardware rendering

---
 gfx/drivers/sdl2_gfx.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gfx/drivers/sdl2_gfx.c b/gfx/drivers/sdl2_gfx.c
index 4dbe33f446815aa212daf0dbb58c3d2e183e88ec..dd67b4aaec550cf375ea3da088363e266dc5aa31 100644
--- a/gfx/drivers/sdl2_gfx.c
+++ b/gfx/drivers/sdl2_gfx.c
@@ -182,7 +182,7 @@ static void sdl2_render_msg(sdl2_video_t *vid, const char *msg)
 
 static void sdl2_init_renderer(sdl2_video_t *vid)
 {
-   unsigned flags = SDL_RENDERER_ACCELERATED;
+   unsigned flags = SDL_RENDERER_SOFTWARE;
 
    if (vid->video.vsync)
       flags |= SDL_RENDERER_PRESENTVSYNC;
