From 9b61ddd52828042b106d7d4c3fef582be9c933e7 Mon Sep 17 00:00:00 2001
From: Vincent Sanders <vince@kyllikki.org>
Date: Sun, 6 Oct 2019 11:11:16 +0100
Subject: [PATCH 2/2] Stub out getprogname() for serenity

---
 srclib/getprogname.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/srclib/getprogname.c b/srclib/getprogname.c
index 22d71eb..7cc112d 100644
--- a/srclib/getprogname.c
+++ b/srclib/getprogname.c
@@ -244,6 +244,8 @@ getprogname (void)
         }
     }
   return NULL;
+# elif defined __serenity__
+  return "?";
 # else
 #  error "getprogname module not ported to this OS"
 # endif
-- 
2.36.1

