From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gunnar@beutner.name>
Date: Sun, 23 Oct 2022 12:49:01 +0200
Subject: [PATCH] Remove incorrect linker library path

---
 src/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.am b/src/Makefile.am
index 51a1cc1c819a1f3be4ab245abd2c8f8af3560571..2eee4b103359c8061626024a9ec1654aad4a2933 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -81,7 +81,7 @@ pacman_LDFLAGS = -I/usr/include/SDL2 -D_REENTRANT #$(SDL2_CFLAGS)
 pacman_LDADD =  \
 	$(SDL2_LIBS) \
 	-lSDL2_mixer \
-	-L/usr/lib64 -lSDL2
+	-lSDL2
 
 
 
