From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "Mr.UNIX" <mrunix00@protonmail.com>
Date: Mon, 25 Dec 2023 12:17:14 +0100
Subject: [PATCH] fix-missing-select.h-header

---
 include/libssh2.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/include/libssh2.h b/include/libssh2.h
index 97ac5898fa9feed41ef6ecf018854c87a075986f..eb5c40ce7bb19ee70d0b88154ef37f2ebaf6d18d 100644
--- a/include/libssh2.h
+++ b/include/libssh2.h
@@ -101,6 +101,7 @@ extern "C" {
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <sys/select.h>
 
 /* Allow alternate API prefix from CFLAGS or calling app */
 #ifndef LIBSSH2_API
