From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Wed, 14 Apr 2021 04:32:19 +0200
Subject: [PATCH] Include sys/time.h

---
 src/assuan-socket.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/assuan-socket.c b/src/assuan-socket.c
index 97b2312..39079bc 100644
--- a/src/assuan-socket.c
+++ b/src/assuan-socket.c
@@ -34,6 +34,7 @@
 #endif
 #else
 # include <sys/types.h>
+# include <sys/time.h>
 # include <sys/socket.h>
 # include <netinet/in.h>
 # include <arpa/inet.h>
