diff -Naur libassuan-2.5.5/src/system-posix.c libassuan-2.5.5.serenity/src/system-posix.c
--- libassuan-2.5.5/src/system-posix.c	2021-04-14 02:40:14.020341296 +0200
+++ libassuan-2.5.5.serenity/src/system-posix.c	2021-04-14 02:39:56.823341349 +0200
@@ -412,7 +412,12 @@
 __assuan_socketpair (assuan_context_t ctx, int namespace, int style,
 		     int protocol, assuan_fd_t filedes[2])
 {
+#ifndef __serenity__
   return socketpair (namespace, style, protocol, filedes);
+#else
+  errno = ENOTSUP;
+  return -1;
+#endif
 }
 
 
