From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Mon, 12 Apr 2021 09:18:56 +0200
Subject: [PATCH] Skip building helpers by default

---
 configure.ac    | 3 +++
 src/Makefile.am | 2 ++
 2 files changed, 5 insertions(+)

diff --git a/configure.ac b/configure.ac
index a247d95cc63f4434f5af3dcf741abb61170578c9..f99ba6093cdc65d8a846aa3fc7eacd5bfe87d3bf 100644
--- a/configure.ac
+++ b/configure.ac
@@ -228,5 +228,8 @@ AC_ARG_ENABLE(lineno, AS_HELP_STRING(--disable-lineno, \
 if test "$enable_lineno" != "no"; then
 	AC_DEFINE([WITH_LINENO], 1, [Define if you build with -DWITH_LINENO])
 fi
+AC_ARG_ENABLE(helpers, AS_HELP_STRING(--enable-helpers, \
+				     [Build helpers]))
+AM_CONDITIONAL([BUILD_HELPERS], [test x$enable_helpers = xyes])
 AC_CONFIG_FILES([Makefile src/Makefile])
 AC_OUTPUT
diff --git a/src/Makefile.am b/src/Makefile.am
index 173246535de4c8058a5f61ffbc3df569a44ec556..a128d0f2eb979e4a6aa0fddb2c2ee4e1bc622f76 100644
--- a/src/Makefile.am
+++ b/src/Makefile.am
@@ -64,7 +64,9 @@ syntax.c syntax.h: mksyntax
 signames.c: mksignames
 	$(AM_V_GEN)./$^
 
+if BUILD_HELPERS
 mksyntax: token.h
 
 $(HELPERS): %: %.c
 	$(AM_V_CC)$(COMPILE_FOR_BUILD) -o $@ $<
+endif
