--- dialog-1.3-20210324/config.sub	2020-08-16 20:36:41.000000000 -0700
+++ dialog-1.3-20210324/config.sub	2021-04-11 20:15:18.580516177 -0700
@@ -1693,6 +1693,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
