From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Mon, 16 May 2022 16:20:52 +0430
Subject: [PATCH] Make configure assume all dependencies are okay

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index de9c48a..1ace244 100755
--- a/configure
+++ b/configure
@@ -6828,6 +6828,10 @@ bsdi[45]*)
   lt_cv_file_magic_test_file=/shlib/libc.so
   ;;
 
+*serenity*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
   lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
