From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Liav A <liavalb@gmail.com>
Date: Sun, 16 Oct 2022 20:27:51 +0300
Subject: [PATCH] Allow to override the default C compiler

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 4ca97b8a130800f56111578dc1be1f140524985e..160ab6a42c2db3d2baa2c077813b027927b81ba5 100644
--- a/Makefile
+++ b/Makefile
@@ -18,7 +18,7 @@
 
 PREFIX=/usr/local
 
-CC=gcc
+CC=
 INSTALL=install
 
 VERSION=2.0.4
