--- SDL2_gfx-1.0.4/config.sub	2021-04-04 18:40:50.418255684 +0200
+++ SDL2_gfx-1.0.4/config.sub	2021-04-04 18:42:32.801888295 +0200
@@ -1355,7 +1355,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -serenity*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
