From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Liav A <liavalb@gmail.com>
Date: Fri, 27 May 2022 10:19:35 +0300
Subject: [PATCH] Disable sprintf debug message with formatting issues

---
 source/compiler/dtfield.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/source/compiler/dtfield.c b/source/compiler/dtfield.c
index f931631..f9fb310 100644
--- a/source/compiler/dtfield.c
+++ b/source/compiler/dtfield.c
@@ -280,9 +280,9 @@ DtCompileString (
 
     if (Length > ByteLength)
     {
-        sprintf (AslGbl_MsgBuffer,
-            "Maximum %u characters, found %u characters [%s]",
-            ByteLength, Length, Field->Value);
+        //sprintf (AslGbl_MsgBuffer,
+        //    "Maximum %u characters, found %u characters [%s]",
+        //    ByteLength, Length, Field->Value);
         DtError (ASL_ERROR, ASL_MSG_STRING_LENGTH, Field, AslGbl_MsgBuffer);
         Length = ByteLength;
     }
