From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Wed, 14 Apr 2021 04:32:34 +0200
Subject: [PATCH] Include errno.h

---
 src/mkerrcodes.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/mkerrcodes.c b/src/mkerrcodes.c
index 29c1cc2..a1cbd52 100644
--- a/src/mkerrcodes.c
+++ b/src/mkerrcodes.c
@@ -22,6 +22,7 @@
    system, while this file will be run on the build system.  */
 
 #include <stdio.h>
+#include <errno.h>
 
 #include "mkerrcodes.h"
 
