diff -Naur libgpg-error-1.42/build-aux/config.sub libgpg-error-1.42.serenity/build-aux/config.sub
--- libgpg-error-1.42/build-aux/config.sub	2018-10-24 10:00:20.000000000 +0200
+++ libgpg-error-1.42.serenity/build-aux/config.sub	2021-04-14 01:15:34.040231143 +0200
@@ -1333,7 +1333,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
-	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
