From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Thu, 5 May 2022 18:49:17 +0200
Subject: [PATCH] Add build system support for SerenityOS

---
 configure | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/configure b/configure
index 133f4e3235125222083bf77368ea714b7cc0b703..004e2d5893a4834d4cc209ce4c0f91978b89422f 100755
--- a/configure
+++ b/configure
@@ -365,6 +365,8 @@ elif check_define __NetBSD__; then
   targetos=netbsd
 elif check_define __APPLE__; then
   targetos=darwin
+elif check_define __serenity__; then
+  targetos=serenity
 else
   # This is a fatal error, but don't report it yet, because we
   # might be going to just print the --help text, or it might
@@ -378,6 +380,7 @@ mingw32="no"
 bsd="no"
 linux="no"
 solaris="no"
+serenity="no"
 case $targetos in
 windows)
   mingw32="yes"
@@ -413,6 +416,9 @@ haiku)
 linux)
   linux="yes"
 ;;
+serenity)
+  serenity="yes"
+;;
 esac
 
 if test ! -z "$cpu" ; then
