diff -urp a/bootstrap b/bootstrap
--- a/bootstrap	2021-02-12 16:01:09.075367557 +0330
+++ b/bootstrap	2021-02-12 15:59:27.811749370 +0330
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 # Distributed under the OSI-approved BSD 3-Clause License.  See accompanying
 # file Copyright.txt or https://cmake.org/licensing for details.
 
diff -urp a/configure b/configure
--- a/configure	2021-02-12 16:02:00.042148704 +0330
+++ b/configure	2021-02-12 16:02:07.035496388 +0330
@@ -1,3 +1,3 @@
-#!/bin/sh
+#!/bin/bash
 cmake_source_dir=`cd "\`dirname \"$0\"\`";pwd`
 exec "${cmake_source_dir}/bootstrap" "$@"
