--- a/Source/kwsys/SystemInformation.cxx	2021-02-12 21:49:51.581563656 +0330
+++ b/Source/kwsys/SystemInformation.cxx	2021-02-12 21:49:54.454669016 +0330
@@ -3635,7 +3635,7 @@ long long SystemInformationImplementatio
   if (hostLimitEnvVarName) {
     const char* hostLimitEnvVarValue = getenv(hostLimitEnvVarName);
     if (hostLimitEnvVarValue) {
-      long long hostLimit = std::atoll(hostLimitEnvVarValue);
+      long long hostLimit = atoll(hostLimitEnvVarValue);
       if (hostLimit > 0) {
         memTotal = min(hostLimit, memTotal);
       }
@@ -3659,7 +3659,7 @@ long long SystemInformationImplementatio
   if (procLimitEnvVarName) {
     const char* procLimitEnvVarValue = getenv(procLimitEnvVarName);
     if (procLimitEnvVarValue) {
-      long long procLimit = std::atoll(procLimitEnvVarValue);
+      long long procLimit = atoll(procLimitEnvVarValue);
       if (procLimit > 0) {
         memAvail = min(procLimit, memAvail);
       }
