--- a/Utilities/cmcurl/lib/curl_config.h.cmake	2021-02-12 22:12:54.848790862 +0330
+++ b/Utilities/cmcurl/lib/curl_config.h.cmake	2021-02-12 22:12:57.841458768 +0330
@@ -1052,9 +1052,6 @@ ${SIZEOF_TIME_T_CODE}
 /* Define to empty if `const' does not conform to ANSI C. */
 #cmakedefine const ${const}
 
-/* Type to use in place of in_addr_t when system does not provide it. */
-#cmakedefine in_addr_t ${in_addr_t}
-
 /* Define to `__inline__' or `__inline' if that's what the C compiler
    calls it, or to nothing if 'inline' is not supported under any name.  */
 #ifndef __cplusplus
