--- a/Utilities/cmjsoncpp/src/lib_json/json_writer.cpp	2021-02-12 22:35:57.078226288 +0330
+++ b/Utilities/cmjsoncpp/src/lib_json/json_writer.cpp	2021-02-12 22:36:02.937764094 +0330
@@ -41,7 +41,9 @@
 #endif
 #else
 #include <cmath>
-#if !(defined(__QNXNTO__)) // QNX already defines isfinite
+#if defined(__serenity__) // isfinite is already defined in math.h
+#include <math.h> 
+#elif !(defined(__QNXNTO__)) // QNX already defines isfinite
 #define isfinite std::isfinite
 #endif
 #endif
