From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Seal Sealy <butilkama@gmail.com>
Date: Mon, 7 Aug 2023 07:18:49 +0300
Subject: [PATCH] Fix path to clear utility

---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 5308cb634834d78b9de9df93c7a562299bfb4125..b203853250a84322a8dc67d38f200381336b815e 100644
--- a/main.c
+++ b/main.c
@@ -206,5 +206,5 @@ void crapout_nobug(const char *fmt, ...) {
 }
 
 void clearscreen(void) {
-	system("/usr/bin/clear");
+	system("/bin/clear");
 }
