From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nathan Ell <nathanjamesell@gmail.com>
Date: Tue, 7 Dec 2021 20:00:58 -0700
Subject: [PATCH] Link with `-lintl`

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 2edc74d..4ec93a6 100644
--- a/Makefile
+++ b/Makefile
@@ -316,6 +316,8 @@ ifeq (hp-ux,$(D2U_OS))
         EXTRA_DEFS += -Dhpux -D_HPUX_SOURCE
 endif
 
+LIBS_EXTRA = -lintl 
+
 
 # ............................................................ flags ...
 
