From 56aeba261bf70ff3728e029cf231ac068b84122d Mon Sep 17 00:00:00 2001
From: Timur Sultanov <SultanovTS@yandex.ru>
Date: Sun, 6 Feb 2022 20:05:40 +0300
Subject: [PATCH 11/24] Silencing int conversion warning

---
 tools/ipptool.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/tools/ipptool.c b/tools/ipptool.c
index a3a694d..67aafff 100644
--- a/tools/ipptool.c
+++ b/tools/ipptool.c
@@ -2148,7 +2148,7 @@ init_data(_cups_testdata_t *data)	/* I - Data */
   data->errors       = cupsArrayNew3(NULL, NULL, NULL, 0, (cups_acopy_func_t)strdup, (cups_afree_func_t)free);
   data->pass         = 1;
   data->prev_pass    = 1;
-  data->request_id   = (CUPS_RAND() % 1000) * 137 + 1;
+  data->request_id   = (int)((CUPS_RAND() % 1000) * 137 + 1);
   data->show_header  = 1;
 }
 
@@ -3155,7 +3155,7 @@ token_cb(_ipp_file_t      *f,		/* I - IPP file data */
 	}
 	else if (!_cups_strcasecmp(temp, "random"))
 	{
-	  data->request_id = (CUPS_RAND() % 1000) * 137 + 1;
+	  data->request_id = (int)((CUPS_RAND() % 1000) * 137 + 1);
 	}
 	else
 	{
-- 
2.25.1

