From 67a6bf15246546e493b508d449236cd3c6e94f9c Mon Sep 17 00:00:00 2001
From: Timur Sultanov <SultanovTS@yandex.ru>
Date: Sun, 6 Feb 2022 20:07:04 +0300
Subject: [PATCH 12/24] Silencing more truncation errors

---
 backend/snmp-supplies.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/backend/snmp-supplies.c b/backend/snmp-supplies.c
index 687e7ae..b78a18b 100644
--- a/backend/snmp-supplies.c
+++ b/backend/snmp-supplies.c
@@ -559,7 +559,7 @@ backend_init_supplies(
     cachedir = CUPS_CACHEDIR;
 
   snprintf(cachefilename, sizeof(cachefilename), "%s/%s.snmp", cachedir,
-           addrstr);
+           addrstr) < 0 ? abort() : (void)0;
 
   if ((cachefile = cupsFileOpen(cachefilename, "r")) != NULL)
   {
-- 
2.25.1

