From d60f40f495e2132853435eb821df544b64f19969 Mon Sep 17 00:00:00 2001
From: Timur Sultanov <SultanovTS@yandex.ru>
Date: Sun, 6 Feb 2022 20:08:13 +0300
Subject: [PATCH 13/24] Fixing missing select.h header

---
 backend/ipp.c  | 1 +
 backend/snmp.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/backend/ipp.c b/backend/ipp.c
index 3f3e186..8592d22 100644
--- a/backend/ipp.c
+++ b/backend/ipp.c
@@ -18,6 +18,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <sys/wait.h>
+#include <sys/select.h>
 #if defined(HAVE_GSSAPI) && defined(HAVE_XPC)
 #  include <xpc/xpc.h>
 #  define kPMPrintUIToolAgent	"com.apple.printuitool.agent"
diff --git a/backend/snmp.c b/backend/snmp.c
index 66ac884..dbe3bba 100644
--- a/backend/snmp.c
+++ b/backend/snmp.c
@@ -17,6 +17,7 @@
 #include <cups/file.h>
 #include <cups/http-private.h>
 #include <regex.h>
+#include <sys/select.h>
 
 
 /*
-- 
2.25.1

