From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominika Liberda <ja@sdomi.pl>
Date: Sun, 2 Jul 2023 05:50:58 +0200
Subject: [PATCH] Remove reserved word from goto in mcpp

---
 lib/mcpp/system.c | 10 +++++-----
 1 file changed, 5 insertions(+), 5 deletions(-)

diff --git a/lib/mcpp/system.c b/lib/mcpp/system.c
index a5b0d85c74203bd26365d99a874e5517aa78d2fb..1a61d240fec52c9cc96960566a03cca09084216c 100644
--- a/lib/mcpp/system.c
+++ b/lib/mcpp/system.c
@@ -3435,7 +3435,7 @@ search:
     if (! fullname)                 /* Non-existent or directory    */
         return  FALSE;
     if (standard && included( fullname))        /* Once included    */
-        goto  true;
+        goto  foo;
         
     if ((max_open != 0 && max_open <= include_nest)
                             /* Exceed the known limit of open files */
@@ -3462,12 +3462,12 @@ search:
         if ((fp = fopen( fullname, "r")) == NULL) {
             file->fp = fopen( cur_fullname, "r");
             fseek( file->fp, file->pos, SEEK_SET);
-            goto  false;
+            goto  bar;
         }
         if (max_open == 0)      /* Remember the limit of the system */
             max_open = include_nest;
     } else if (fp == NULL)                  /* No read permission   */ 
-        goto  false;
+        goto  bar;
     /* Truncate buffer of the includer to save memory   */
     len = (int) (file->bptr - file->buffer);
     if (len) {
@@ -3514,9 +3514,9 @@ search:
     if (mkdep && ((mkdep & MD_SYSHEADER) || ! infile->sys_header))
         put_depend( fullname);          /* Output dependency line   */
 
-true:
+foo:
     return  TRUE;
-false:
+bar:
     free( fullname);
     return  FALSE;
 }
