diff -Naur diffutils-3.5/lib/getdtablesize.c diffutils-3.5.serenity/lib/getdtablesize.c
--- diffutils-3.5/lib/getdtablesize.c	2021-04-11 06:54:25.712284497 +0200
+++ diffutils-3.5.serenity/lib/getdtablesize.c	2021-04-11 06:52:52.543810671 +0200
@@ -86,6 +86,7 @@
 
 #else
 
+#ifndef __serenity__
 # include <limits.h>
 # include <sys/resource.h>
 
@@ -117,5 +118,6 @@
 
   return INT_MAX;
 }
+#endif
 
 #endif
