From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Wed, 10 Nov 2021 03:29:21 +0100
Subject: [PATCH] Add SerenityOS to config.guess

---
 llvm/cmake/config.guess | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/llvm/cmake/config.guess b/llvm/cmake/config.guess
index 71abbf939..f48962367 100644
--- a/llvm/cmake/config.guess
+++ b/llvm/cmake/config.guess
@@ -816,6 +816,9 @@ EOF
     i*:PW*:*)
 	echo ${UNAME_MACHINE}-pc-pw32
 	exit ;;
+    *:SerenityOS:*:*)
+	echo ${UNAME_MACHINE}-pc-serenity
+	exit ;;
     *:Interix*:*)
 	case ${UNAME_MACHINE} in
 	    x86)
