diff -ur stress-ng-master/Makefile stress-ng-master-2/Makefile
--- stress-ng-master/Makefile	2020-11-07 10:52:22.000000000 -0800
+++ stress-ng-master-2/Makefile	2020-11-08 23:36:28.587668500 -0800
@@ -21,7 +21,7 @@
 # Codename "synthetic system strainer"
 #
 
-CFLAGS += -Wall -Wextra -DVERSION='"$(VERSION)"' -O2 -std=gnu99
+CFLAGS += -Wall -Wextra -DVERSION='"$(VERSION)"' -O2 -std=gnu99 -Wno-old-style-declaration
 
 #
 # Pedantic flags
@@ -89,7 +89,6 @@
 	stress-close.c \
 	stress-context.c \
 	stress-copy-file.c \
-	stress-cpu.c \
 	stress-cpu-online.c \
 	stress-crypt.c \
 	stress-cyclic.c \
@@ -99,7 +98,6 @@
 	stress-dev.c \
 	stress-dev-shm.c \
 	stress-dir.c \
-	stress-dirdeep.c \
 	stress-dnotify.c \
 	stress-dup.c \
 	stress-dynlib.c \
@@ -132,19 +130,16 @@
 	stress-hdd.c \
 	stress-heapsort.c \
 	stress-hrtimers.c \
-	stress-hsearch.c \
 	stress-icache.c \
 	stress-icmp-flood.c \
 	stress-idle-page.c \
 	stress-inode-flags.c \
 	stress-inotify.c \
-	stress-iomix.c \
 	stress-ioport.c \
 	stress-ioprio.c \
 	stress-iosync.c \
 	stress-io-uring.c \
 	stress-ipsec-mb.c \
-	stress-itimer.c \
 	stress-judy.c \
 	stress-kcmp.c \
 	stress-key.c \
@@ -158,7 +153,6 @@
 	stress-lockofd.c \
 	stress-longjmp.c \
 	stress-loop.c \
-	stress-lsearch.c \
 	stress-madvise.c \
 	stress-malloc.c \
 	stress-matrix.c \
@@ -222,22 +216,16 @@
 	stress-rename.c \
 	stress-resources.c \
 	stress-revio.c \
-	stress-rlimit.c \
-	stress-rmap.c \
 	stress-rtc.c \
 	stress-sctp.c \
-	stress-schedpolicy.c \
 	stress-seal.c \
 	stress-seccomp.c \
 	stress-seek.c \
 	stress-sem.c \
-	stress-sem-sysv.c \
 	stress-sendfile.c \
 	stress-session.c \
-	stress-set.c \
 	stress-shellsort.c \
 	stress-shm.c \
-	stress-shm-sysv.c \
 	stress-sigabrt.c \
 	stress-sigchld.c \
 	stress-sigfd.c \
@@ -257,7 +245,6 @@
 	stress-sockabuse.c \
 	stress-sockdiag.c \
 	stress-sockfd.c \
-	stress-sockpair.c \
 	stress-sockmany.c \
 	stress-softlockup.c \
 	stress-spawn.c \
@@ -269,7 +256,6 @@
 	stress-swap.c \
 	stress-switch.c \
 	stress-sync-file.c \
-	stress-sysbadaddr.c \
 	stress-sysinfo.c \
 	stress-sysinval.c \
 	stress-sysfs.c \
@@ -280,7 +266,6 @@
 	stress-tmpfs.c \
 	stress-tree.c \
 	stress-tsc.c \
-	stress-tsearch.c \
 	stress-tun.c \
 	stress-udp.c \
 	stress-udp-flood.c \
@@ -288,11 +273,9 @@
 	stress-uprobe.c \
 	stress-urandom.c \
 	stress-userfaultfd.c \
-	stress-utime.c \
 	stress-vdso.c \
 	stress-vecmath.c \
 	stress-verity.c \
-	stress-vforkmany.c \
 	stress-vm.c \
 	stress-vm-addr.c \
 	stress-vm-rw.c \
@@ -474,12 +457,6 @@
 	./stress-ng --seq 0 -t 15 --pathological --verbose --times --tz --metrics
 
 .PHONY: install
-install: stress-ng stress-ng.1.gz
+install: stress-ng
 	mkdir -p ${DESTDIR}${BINDIR}
 	cp stress-ng ${DESTDIR}${BINDIR}
-	mkdir -p ${DESTDIR}${MANDIR}
-	cp stress-ng.1.gz ${DESTDIR}${MANDIR}
-	mkdir -p ${DESTDIR}${JOBDIR}
-	cp -rp example-jobs/*.job ${DESTDIR}${JOBDIR}
-	mkdir -p ${DESTDIR}${BASHDIR}
-	cp bash-completion/stress-ng ${DESTDIR}${BASHDIR}

