From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Thu, 13 Jan 2022 16:13:02 +0330
Subject: [PATCH] Makefile: Install to /usr/local

---
 Makefile | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/Makefile b/Makefile
index 541a20f29091e543fc8af7dbb04bcab7b68e551d..e05f0102b84d30867b7b93ab0d747b566da5023a 100644
--- a/Makefile
+++ b/Makefile
@@ -124,10 +124,10 @@ override LDFLAGS += -static -z muldefs
 override CFLAGS += -DBUILD_STATIC
 endif
 
-BINDIR=/usr/bin
-MANDIR=/usr/share/man/man1
-JOBDIR=/usr/share/stress-ng/example-jobs
-BASHDIR=/usr/share/bash-completion/completions
+BINDIR=/usr/local/bin
+MANDIR=/usr/local/share/man/man1
+JOBDIR=/usr/local/share/stress-ng/example-jobs
+BASHDIR=/usr/local/share/bash-completion/completions
 
 #
 # Header files
