diff -ur stress-ng-master/core-helper.c stress-ng-master-2/core-helper.c
--- stress-ng-master/core-helper.c	2020-11-07 10:52:22.000000000 -0800
+++ stress-ng-master-2/core-helper.c	2020-11-08 23:17:03.777668500 -0800
@@ -1102,7 +1102,7 @@
  */
 uint64_t stress_get_prime64(const uint64_t n)
 {
-	static uint p = 1009;
+	static uint64_t p = 1009;
 
 	if (n != p)
 		return p;
@@ -1149,6 +1149,9 @@
  */
 size_t stress_get_file_limit(void)
 {
+	/* Just return a constant for serenity */
+	return 65536;
+#if 0
 	struct rlimit rlim;
 	size_t i, opened = 0, max = 65536;	/* initial guess */
 
@@ -1161,6 +1164,7 @@
 			opened++;
 	}
 	return max - opened;
+#endif
 }

 /*
diff -ur stress-ng-master/core-job.c stress-ng-master-2/core-job.c
--- stress-ng-master/core-job.c	2020-11-07 10:52:22.000000000 -0800
+++ stress-ng-master-2/core-job.c	2020-11-08 22:32:06.607668500 -0800
@@ -28,7 +28,7 @@
 #define RUN_SEQUENTIAL	(0x01)
 #define RUN_PARALLEL	(0x02)
 
-#define ISBLANK(ch)	isblank((int)(ch))
+#define ISBLANK(ch)	(ch == ' ' || ch == '\t')
 
 /*
  *  stress_chop()
diff -ur stress-ng-master/core-sched.c stress-ng-master-2/core-sched.c
--- stress-ng-master/core-sched.c	2020-11-07 10:52:22.000000000 -0800
+++ stress-ng-master-2/core-sched.c	2020-11-08 23:21:44.497668500 -0800
@@ -263,6 +263,7 @@
 			pr_dbg("%s: setting scheduler class '%s'\n", prefix, sched_name);
 		break;
 	}
+#if 0
 	rc = sched_setscheduler(pid, sched, &param);
 	if (rc < 0) {
 		rc = -errno;
@@ -273,6 +274,7 @@
 				errno, strerror(errno));
 		return rc;
 	}
+#endif
 	return 0;
 }
 #endif
diff -ur stress-ng-master/stress-chown.c stress-ng-master-2/stress-chown.c
--- stress-ng-master/stress-chown.c	2020-11-07 10:52:22.000000000 -0800
+++ stress-ng-master-2/stress-chown.c	2020-11-08 22:35:41.107668500 -0800
@@ -219,6 +219,9 @@
 				pr_fail("%s: chown %s failed, errno=%d (%s)\n",
 					args->name, filename, errno, strerror(errno));
 		}
+
+/* No lchown support in serenity */
+#if 0
 		ret = do_chown(lchown, filename, cap_chown, uid, gid);
 		if (ret < 0) {
 			if (ret == -ENOENT || ret == -ENOTDIR) {
@@ -233,6 +236,7 @@
 				pr_fail("%s: chown %s failed, errno=%d (%s)\n",
 					args->name, filename, errno, strerror(errno));
 		}
+#endif
 		inc_counter(args);
 	} while (keep_stressing());
 
