diff -ur a/stress-str.c b/stress-str.c
--- a/stress-str.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-str.c	2020-11-08 22:55:55.887668500 -0800
@@ -597,8 +597,6 @@
 	{ "all",		stress_str_all,		NULL },	/* Special "all test */
 
 #if defined(HAVE_STRINGS_H)
-	{ "index",		stress_index,		index },
-	{ "rindex",		stress_rindex,		rindex  },
 	{ "strcasecmp",		stress_strcasecmp,	strcasecmp },
 #endif
 #if defined(HAVE_STRLCAT)
diff -ur a/stress-sigsegv.c b/stress-sigsegv.c
--- a/stress-sigsegv.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-sigsegv.c	2020-11-08 22:50:22.327668500 -0800
@@ -143,11 +143,13 @@
 				pr_fail("%s: expecting SIGSEGV/SIGILL/SIGBUS, got %s instead\n",
 					args->name, strsignal(signo));
 			}
+#if 0
 			if (verify && (signo == SIGBUS) && (code != SEGV_ACCERR)) {
 				pr_fail("%s: expecting SIGBUS si_code SEGV_ACCERR (%d), got %d instead\n",
 					args->name, SEGV_ACCERR, code);
 			}
 #endif
+#endif
 			inc_counter(args);
 		} else {
 #if defined(SA_SIGINFO)
diff -ur a/stress-mmapmany.c b/stress-mmapmany.c
--- a/stress-mmapmany.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-mmapmany.c	2020-11-08 22:47:27.567668500 -0800
@@ -35,7 +35,8 @@
 static int stress_mmapmany_child(const stress_args_t *args, void *context)
 {
 	const size_t page_size = args->page_size;
-	ssize_t max = sysconf(_SC_MAPPED_FILES);
+	/* Just make up a constant for Serenity */
+	ssize_t max = 6400; 
 	uint8_t **mappings;
 	max = STRESS_MAXIMUM(max, MMAP_MAX);
 
d
