diff -ur a/stress-fork.c b/stress-fork.c
--- a/stress-fork.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-fork.c	2020-11-08 23:05:17.627668500 -0800
@@ -198,6 +198,7 @@
 }
 
 
+#if 0
 /*
  *  stress_vfork()
  *	stress by vforking and exiting
@@ -218,16 +219,13 @@
 	return stress_fork_fn(args, vfork, "vfork", vfork_max);
 }
 STRESS_PRAGMA_POP
+#endif
 
 static const stress_opt_set_func_t fork_opt_set_funcs[] = {
 	{ OPT_fork_max,		stress_set_fork_max },
 	{ 0,			NULL }
 };
 
-static const stress_opt_set_func_t vfork_opt_set_funcs[] = {
-	{ OPT_vfork_max,	stress_set_vfork_max },
-	{ 0,			NULL }
-};
 
 stressor_info_t stress_fork_info = {
 	.stressor = stress_fork,
@@ -236,9 +234,16 @@
 	.help = fork_help
 };
 
+#if 0
+static const stress_opt_set_func_t vfork_opt_set_funcs[] = {
+	{ OPT_vfork_max,	stress_set_vfork_max },
+	{ 0,			NULL }
+};
+
 stressor_info_t stress_vfork_info = {
 	.stressor = stress_vfork,
 	.class = CLASS_SCHEDULER | CLASS_OS,
 	.opt_set_funcs = vfork_opt_set_funcs,
 	.help = vfork_help
 };
+#endif
diff -ur a/stress-get.c b/stress-get.c
--- a/stress-get.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-get.c	2020-11-08 22:42:06.027668500 -0800
@@ -162,7 +162,9 @@
 #endif
 		struct timeval tv;
 		struct timezone tz;
+#if 0
 		struct rlimit rlim;
+#endif
 		time_t t, t1, t2;
 		pid_t pid;
 		gid_t gid;
@@ -324,6 +326,8 @@
 			check_do_run();
 		}
 #endif
+
+#if 0
 		/* Invalid getrlimit syscall and ignoring failure */
 		(void)getrlimit(INT_MAX, &rlim);
 
@@ -334,6 +338,7 @@
 					args->name, i, errno, strerror(errno));
 			check_do_run();
 		}
+#endif 0
 
 #if defined(__NR_ugetrlimit)
 		/*
diff -ur a/stress-hdd.c b/stress-hdd.c
--- a/stress-hdd.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-hdd.c	2020-11-08 23:19:59.907668500 -0800
@@ -277,11 +277,11 @@
 #endif
 		default:
 			/* 50% */
-			return readv(fd, iov, HDD_IO_VEC_MAX);
+			break;
 		}
-	} else {
-		return read(fd, buf, count);
-	}
+	} 
+
+	return read(fd, buf, count);
 }
 
 
diff -ur a/stress-ng.c b/stress-ng.c
--- a/stress-ng.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-ng.c	2020-11-08 23:29:08.447668500 -0800
@@ -321,10 +321,6 @@
 	{ "dir",	1,	0,	OPT_dir },
 	{ "dir-ops",	1,	0,	OPT_dir_ops },
 	{ "dir-dirs",	1,	0,	OPT_dir_dirs },
-	{ "dirdeep",	1,	0,	OPT_dirdeep },
-	{ "dirdeep-ops",1,	0,	OPT_dirdeep_ops },
-	{ "dirdeep-dirs",1,	0,	OPT_dirdeep_dirs },
-	{ "dirdeep-inodes",1,	0,	OPT_dirdeep_inodes },
 	{ "dry-run",	0,	0,	OPT_dry_run },
 	{ "dnotify",	1,	0,	OPT_dnotify },
 	{ "dnotify-ops",1,	0,	OPT_dnotify_ops },
@@ -409,9 +405,6 @@
 	{ "hrtimers",	1,	0,	OPT_hrtimers },
 	{ "hrtimers-ops",1,	0,	OPT_hrtimers_ops },
 	{ "help",	0,	0,	OPT_help },
-	{ "hsearch",	1,	0,	OPT_hsearch },
-	{ "hsearch-ops",1,	0,	OPT_hsearch_ops },
-	{ "hsearch-size",1,	0,	OPT_hsearch_size },
 	{ "icache",	1,	0,	OPT_icache },
 	{ "icache-ops",	1,	0,	OPT_icache_ops },
 	{ "icmp-flood",	1,	0,	OPT_icmp_flood },
@@ -425,9 +418,6 @@
 	{ "inotify-ops",1,	0,	OPT_inotify_ops },
 	{ "io",		1,	0,	OPT_io },
 	{ "io-ops",	1,	0,	OPT_io_ops },
-	{ "iomix",	1,	0,	OPT_iomix },
-	{ "iomix-bytes",1,	0,	OPT_iomix_bytes },
-	{ "iomix-ops",	1,	0,	OPT_iomix_ops },
 	{ "ionice-class",1,	0,	OPT_ionice_class },
 	{ "ionice-level",1,	0,	OPT_ionice_level },
 	{ "ioport",	1,	0,	OPT_ioport },
@@ -440,10 +430,6 @@
 	{ "ipsec-mb",	1,	0,	OPT_ipsec_mb },
 	{ "ipsec-mb-ops",1,	0,	OPT_ipsec_mb_ops },
 	{ "ipsec-mb-feature",1,	0,	OPT_ipsec_mb_feature },
-	{ "itimer",	1,	0,	OPT_itimer },
-	{ "itimer-ops",	1,	0,	OPT_itimer_ops },
-	{ "itimer-freq",1,	0,	OPT_itimer_freq },
-	{ "itimer-rand",0,	0,	OPT_itimer_rand },
 	{ "job",	1,	0,	OPT_job },
 	{ "judy",	1,	0,	OPT_judy },
 	{ "judy-ops",	1,	0,	OPT_judy_ops },
@@ -477,9 +463,6 @@
 	{ "longjmp-ops",1,	0,	OPT_longjmp_ops },
 	{ "loop",	1,	0,	OPT_loop },
 	{ "loop-ops",	1,	0,	OPT_loop_ops },
-	{ "lsearch",	1,	0,	OPT_lsearch },
-	{ "lsearch-ops",1,	0,	OPT_lsearch_ops },
-	{ "lsearch-size",1,	0,	OPT_lsearch_size },
 	{ "madvise",	1,	0,	OPT_madvise },
 	{ "madvise-ops",1,	0,	OPT_madvise_ops },
 	{ "malloc",	1,	0,	OPT_malloc },
@@ -669,21 +652,14 @@
 	{ "revio-ops",	1,	0,	OPT_revio_ops },
 	{ "revio-opts",	1,	0,	OPT_revio_opts },
 	{ "revio-bytes",1,	0,	OPT_revio_bytes },
-	{ "rlimit",	1,	0,	OPT_rlimit },
-	{ "rlimit-ops",	1,	0,	OPT_rlimit_ops },
-	{ "rmap",	1,	0,	OPT_rmap },
-	{ "rmap-ops",	1,	0,	OPT_rmap_ops },
 	{ "rtc",	1,	0,	OPT_rtc },
 	{ "rtc-ops",	1,	0,	OPT_rtc_ops },
 	{ "sched",	1,	0,	OPT_sched },
 	{ "sched-prio",	1,	0,	OPT_sched_prio },
-	{ "schedpolicy",1,	0,	OPT_schedpolicy },
-	{ "schedpolicy-ops",1,	0,	OPT_schedpolicy_ops },
 	{ "sched-period",1,	0,	OPT_sched_period },
 	{ "sched-runtime",1,	0,	OPT_sched_runtime },
 	{ "sched-deadline",1,	0,	OPT_sched_deadline },
 	{ "sched-reclaim",0,    0,      OPT_sched_reclaim },
-	{ "schedpolicy",1,	0,	OPT_schedpolicy },
 	{ "sctp",	1,	0,	OPT_sctp },
 	{ "sctp-ops",	1,	0,	OPT_sctp_ops },
 	{ "sctp-domain",1,	0,	OPT_sctp_domain },
@@ -699,17 +675,12 @@
 	{ "sem",	1,	0,	OPT_sem },
 	{ "sem-ops",	1,	0,	OPT_sem_ops },
 	{ "sem-procs",	1,	0,	OPT_sem_procs },
-	{ "sem-sysv",	1,	0,	OPT_sem_sysv },
-	{ "sem-sysv-ops",1,	0,	OPT_sem_sysv_ops },
-	{ "sem-sysv-procs",1,	0,	OPT_sem_sysv_procs },
 	{ "sendfile",	1,	0,	OPT_sendfile },
 	{ "sendfile-ops",1,	0,	OPT_sendfile_ops },
 	{ "sendfile-size",1,	0,	OPT_sendfile_size },
 	{ "sequential",	1,	0,	OPT_sequential },
 	{ "session",	1,	0,	OPT_session },
 	{ "session-ops",1,	0,	OPT_session_ops },
-	{ "set",	1,	0,	OPT_set },
-	{ "set-ops",	1,	0,	OPT_set_ops },
 	{ "shellsort",	1,	0,	OPT_shellsort },
 	{ "shellsort-ops",1,	0,	OPT_shellsort_ops },
 	{ "shellsort-size",1,	0,	OPT_shellsort_integers },
@@ -717,10 +688,6 @@
 	{ "shm-ops",	1,	0,	OPT_shm_ops },
 	{ "shm-bytes",	1,	0,	OPT_shm_bytes },
 	{ "shm-objs",	1,	0,	OPT_shm_objects },
-	{ "shm-sysv",	1,	0,	OPT_shm_sysv },
-	{ "shm-sysv-ops",1,	0,	OPT_shm_sysv_ops },
-	{ "shm-sysv-bytes",1,	0,	OPT_shm_sysv_bytes },
-	{ "shm-sysv-segs",1,	0,	OPT_shm_sysv_segments },
 	{ "sigabrt",	1,	0,	OPT_sigabrt },
 	{ "sigabrt-ops",1,	0,	OPT_sigabrt_ops },
 	{ "sigchld",	1,	0,	OPT_sigchld },
@@ -769,8 +736,6 @@
 	{ "sockfd-port",1,	0,	OPT_sockfd_port },
 	{ "sockmany",	1,	0,	OPT_sockmany },
 	{ "sockmany-ops",1,	0,	OPT_sockmany_ops },
-	{ "sockpair",	1,	0,	OPT_sockpair },
-	{ "sockpair-ops",1,	0,	OPT_sockpair_ops },
 	{ "softlockup",	1,	0,	OPT_softlockup },
 	{ "softlockup-ops",1,	0,	OPT_softlockup_ops },
 	{ "spawn",	1,	0,	OPT_spawn },
@@ -802,8 +767,6 @@
 	{ "sync-file",	1,	0,	OPT_sync_file },
 	{ "sync-file-ops", 1,	0,	OPT_sync_file_ops },
 	{ "sync-file-bytes", 1,	0,	OPT_sync_file_bytes },
-	{ "sysbadaddr",	1,	0,	OPT_sysbadaddr },
-	{ "sysbadaddr-ops",1,	0,	OPT_sysbadaddr_ops },
 	{ "sysfs",	1,	0,	OPT_sysfs },
 	{ "sysfs-ops",1,	0,	OPT_sysfs_ops },
 	{ "sysinfo",	1,	0,	OPT_sysinfo },
@@ -839,9 +802,6 @@
 	{ "tree-size",	1,	0,	OPT_tree_size },
 	{ "tsc",	1,	0,	OPT_tsc },
 	{ "tsc-ops",	1,	0,	OPT_tsc_ops },
-	{ "tsearch",	1,	0,	OPT_tsearch },
-	{ "tsearch-ops",1,	0,	OPT_tsearch_ops },
-	{ "tsearch-size",1,	0,	OPT_tsearch_size },
 	{ "thrash",	0,	0,	OPT_thrash },
 	{ "times",	0,	0,	OPT_times },
 	{ "timestamp",	0,	0,	OPT_timestamp },
@@ -866,9 +826,6 @@
 	{ "userfaultfd",1,	0,	OPT_userfaultfd },
 	{ "userfaultfd-ops",1,	0,	OPT_userfaultfd_ops },
 	{ "userfaultfd-bytes",1,0,	OPT_userfaultfd_bytes },
-	{ "utime",	1,	0,	OPT_utime },
-	{ "utime-ops",	1,	0,	OPT_utime_ops },
-	{ "utime-fsync",0,	0,	OPT_utime_fsync },
 	{ "vdso",	1,	0,	OPT_vdso },
 	{ "vdso-ops",	1,	0,	OPT_vdso_ops },
 	{ "vdso-func",	1,	0,	OPT_vdso_func },
@@ -879,11 +836,6 @@
 	{ "verity",	1,	0,	OPT_verity },
 	{ "verity-ops",	1,	0,	OPT_verity_ops },
 	{ "version",	0,	0,	OPT_version },
-	{ "vfork",	1,	0,	OPT_vfork },
-	{ "vfork-ops",	1,	0,	OPT_vfork_ops },
-	{ "vfork-max",	1,	0,	OPT_vfork_max },
-	{ "vforkmany",	1,	0,	OPT_vforkmany },
-	{ "vforkmany-ops", 1,	0,	OPT_vforkmany_ops },
 	{ "vm",		1,	0,	OPT_vm },
 	{ "vm-bytes",	1,	0,	OPT_vm_bytes },
 	{ "vm-hang",	1,	0,	OPT_vm_hang },
@@ -2964,6 +2916,7 @@
  */
 static inline void stress_mlock_executable(void)
 {
+#if 0
 #if defined(MLOCKED_SECTION)
 	extern void *__start_mlocked_text;
 	extern void *__stop_mlocked_text;
@@ -2973,6 +2926,7 @@
 	stress_mlock_region(&__start_mlocked_text, &__stop_mlocked_text);
 	stress_mlock_region(&__start_mlocked_data, &__stop_mlocked_data);
 #endif
+#endif
 }
 
 int main(int argc, char **argv, char **envp)
diff -ur a/stress-ng.h b/stress-ng.h
--- a/stress-ng.h	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-ng.h	2020-11-08 23:25:40.027668500 -0800
@@ -56,7 +56,6 @@
 #include <math.h>
 #include <pwd.h>
 #include <sched.h>
-#include <search.h>
 #include <setjmp.h>
 #include <signal.h>
 #include <stdarg.h>
@@ -1093,14 +1092,6 @@
 #define PACKED
 #endif
 
-/* force inlining hint */
-#if defined(__GNUC__) && NEED_GNUC(3,4,0)	/* or possibly earlier */ \
- && ((!defined(__s390__) && !defined(__s390x__)) || NEED_GNUC(6,0,1))
-#define ALWAYS_INLINE	__attribute__ ((always_inline))
-#else
-#define ALWAYS_INLINE
-#endif
-
 /* force no inlining hint */
 #if defined(__GNUC__) && NEED_GNUC(3,4,0)	/* or possibly earier */
 #define NOINLINE	__attribute__ ((noinline))
@@ -2230,7 +2221,6 @@
 		uint64_t timeout[STRESS_PROCS_MAX];	/* Shared futex timeouts */
 	} futex;
 	struct {
-		key_t key_id;				/* System V semaphore key id */
 		int sem_id;				/* System V semaphore id */
 		bool init;				/* System V semaphore initialized */
 	} sem_sysv;
@@ -2288,7 +2278,6 @@
 	MACRO(close)		\
 	MACRO(context)		\
 	MACRO(copy_file)	\
-	MACRO(cpu)		\
 	MACRO(cpu_online)	\
 	MACRO(crypt)		\
 	MACRO(cyclic)		\
@@ -2298,7 +2287,6 @@
 	MACRO(dev)		\
 	MACRO(dev_shm)		\
 	MACRO(dir)		\
-	MACRO(dirdeep)		\
 	MACRO(dnotify)		\
 	MACRO(dup)		\
 	MACRO(dynlib)		\
@@ -2331,19 +2319,16 @@
 	MACRO(hdd)		\
 	MACRO(heapsort)		\
 	MACRO(hrtimers)		\
-	MACRO(hsearch)		\
 	MACRO(icache)		\
 	MACRO(icmp_flood)	\
 	MACRO(idle_page)	\
 	MACRO(inode_flags)	\
 	MACRO(inotify)		\
 	MACRO(io)		\
-	MACRO(iomix)		\
 	MACRO(ioport)		\
 	MACRO(ioprio)		\
 	MACRO(io_uring)		\
 	MACRO(ipsec_mb)		\
-	MACRO(itimer)		\
 	MACRO(judy)		\
 	MACRO(kcmp)		\
 	MACRO(key)		\
@@ -2357,7 +2342,6 @@
 	MACRO(lockofd)		\
 	MACRO(longjmp)		\
 	MACRO(loop)		\
-	MACRO(lsearch)		\
 	MACRO(madvise)		\
 	MACRO(malloc)		\
 	MACRO(matrix)		\
@@ -2421,22 +2405,16 @@
 	MACRO(rename)		\
 	MACRO(resources)	\
 	MACRO(revio)		\
-	MACRO(rlimit)		\
-	MACRO(rmap)		\
 	MACRO(rtc)		\
-	MACRO(schedpolicy)	\
 	MACRO(sctp)		\
 	MACRO(seal)		\
 	MACRO(seccomp)		\
 	MACRO(seek)		\
 	MACRO(sem)		\
-	MACRO(sem_sysv)		\
 	MACRO(sendfile)		\
 	MACRO(session)		\
-	MACRO(set)		\
 	MACRO(shellsort)	\
 	MACRO(shm)		\
-	MACRO(shm_sysv)		\
 	MACRO(sigabrt)		\
 	MACRO(sigchld)		\
 	MACRO(sigfd)		\
@@ -2456,7 +2434,6 @@
 	MACRO(sockabuse)	\
 	MACRO(sockdiag)		\
 	MACRO(sockfd)		\
-	MACRO(sockpair)		\
 	MACRO(sockmany)		\
 	MACRO(softlockup)	\
 	MACRO(spawn)		\
@@ -2469,7 +2446,6 @@
 	MACRO(switch)		\
 	MACRO(symlink)		\
 	MACRO(sync_file)	\
-	MACRO(sysbadaddr)	\
 	MACRO(sysinfo)		\
 	MACRO(sysinval)		\
 	MACRO(sysfs)		\
@@ -2480,7 +2456,6 @@
 	MACRO(tmpfs)		\
 	MACRO(tree)		\
 	MACRO(tsc)		\
-	MACRO(tsearch)		\
 	MACRO(tun)		\
 	MACRO(udp)		\
 	MACRO(udp_flood)	\
@@ -2488,12 +2463,9 @@
 	MACRO(uprobe)		\
 	MACRO(urandom)		\
 	MACRO(userfaultfd)	\
-	MACRO(utime)		\
 	MACRO(vdso)		\
 	MACRO(vecmath)		\
 	MACRO(verity)		\
-	MACRO(vfork)		\
-	MACRO(vforkmany)	\
 	MACRO(vm)		\
 	MACRO(vm_addr)		\
 	MACRO(vm_rw)		\
@@ -2709,11 +2681,6 @@
 	OPT_dir_ops,
 	OPT_dir_dirs,
 
-	OPT_dirdeep,
-	OPT_dirdeep_ops,
-	OPT_dirdeep_dirs,
-	OPT_dirdeep_inodes,
-
 	OPT_dnotify,
 	OPT_dnotify_ops,
 
@@ -2825,10 +2792,6 @@
 	OPT_hrtimers,
 	OPT_hrtimers_ops,
 
-	OPT_hsearch,
-	OPT_hsearch_ops,
-	OPT_hsearch_size,
-
 	OPT_icache,
 	OPT_icache_ops,
 
@@ -2846,10 +2809,6 @@
 	OPT_inotify,
 	OPT_inotify_ops,
 
-	OPT_iomix,
-	OPT_iomix_bytes,
-	OPT_iomix_ops,
-
 	OPT_ioport,
 	OPT_ioport_ops,
 	OPT_ioport_opts,
@@ -2869,11 +2828,6 @@
 	OPT_ipsec_mb_ops,
 	OPT_ipsec_mb_feature,
 
-	OPT_itimer,
-	OPT_itimer_ops,
-	OPT_itimer_freq,
-	OPT_itimer_rand,
-
 	OPT_judy,
 	OPT_judy_ops,
 	OPT_judy_size,
@@ -2919,10 +2873,6 @@
 	OPT_loop,
 	OPT_loop_ops,
 
-	OPT_lsearch,
-	OPT_lsearch_ops,
-	OPT_lsearch_size,
-
 	OPT_madvise,
 	OPT_madvise_ops,
 
@@ -3170,9 +3120,6 @@
 	OPT_revio_opts,
 	OPT_revio_bytes,
 
-	OPT_rlimit,
-	OPT_rlimit_ops,
-
 	OPT_rmap,
 	OPT_rmap_ops,
 
@@ -3182,9 +3129,6 @@
 	OPT_sched,
 	OPT_sched_prio,
 
-	OPT_schedpolicy,
-	OPT_schedpolicy_ops,
-
 	OPT_sched_period,
 	OPT_sched_runtime,
 	OPT_sched_deadline,
@@ -3214,16 +3158,9 @@
 	OPT_sem_ops,
 	OPT_sem_procs,
 
-	OPT_sem_sysv,
-	OPT_sem_sysv_ops,
-	OPT_sem_sysv_procs,
-
 	OPT_session,
 	OPT_session_ops,
 
-	OPT_set,
-	OPT_set_ops,
-
 	OPT_shellsort,
 	OPT_shellsort_ops,
 	OPT_shellsort_integers,
@@ -3233,11 +3170,6 @@
 	OPT_shm_bytes,
 	OPT_shm_objects,
 
-	OPT_shm_sysv,
-	OPT_shm_sysv_ops,
-	OPT_shm_sysv_bytes,
-	OPT_shm_sysv_segments,
-
 	OPT_sequential,
 
 	OPT_sigabrt,
@@ -3307,9 +3239,6 @@
 	OPT_sockmany,
 	OPT_sockmany_ops,
 
-	OPT_sockpair,
-	OPT_sockpair_ops,
-
 	OPT_softlockup,
 	OPT_softlockup_ops,
 
@@ -3352,9 +3281,6 @@
 	OPT_sync_file_ops,
 	OPT_sync_file_bytes,
 
-	OPT_sysbadaddr,
-	OPT_sysbadaddr_ops,
-
 	OPT_sysinfo,
 	OPT_sysinfo_ops,
 
@@ -3408,10 +3334,6 @@
 	OPT_tsc,
 	OPT_tsc_ops,
 
-	OPT_tsearch,
-	OPT_tsearch_ops,
-	OPT_tsearch_size,
-
 	OPT_tun,
 	OPT_tun_ops,
 	OPT_tun_tap,
@@ -3438,10 +3360,6 @@
 	OPT_userfaultfd_ops,
 	OPT_userfaultfd_bytes,
 
-	OPT_utime,
-	OPT_utime_ops,
-	OPT_utime_fsync,
-
 	OPT_vdso,
 	OPT_vdso_ops,
 	OPT_vdso_func,
@@ -3454,13 +3372,6 @@
 	OPT_verity,
 	OPT_verity_ops,
 
-	OPT_vfork,
-	OPT_vfork_ops,
-	OPT_vfork_max,
-
-	OPT_vforkmany,
-	OPT_vforkmany_ops,
-
 	OPT_vm_bytes,
 	OPT_vm_hang,
 	OPT_vm_keep,
iff -ur a/stress-schedpolicy.c b/stress-schedpolicy.c
--- a/stress-schedpolicy.c	2020-11-07 10:52:22.000000000 -0800
+++ b/stress-schedpolicy.c	2020-11-08 23:22:11.257668500 -0800
@@ -117,7 +117,9 @@
 		case SCHED_OTHER:
 #endif
 			param.sched_priority = 0;
+#if 0
 			ret = sched_setscheduler(pid, new_policy, &param);
+#endif
 			break;
 #if defined(SCHED_RR)
 		case SCHED_RR:

