From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Gianforcaro <b.gianfo@gmail.com>
Date: Mon, 27 Dec 2021 23:46:44 -0800
Subject: [PATCH] serenity: Disable rand48 cpu stressor, it's not implemented
 for Serenity

---
 stress-cpu.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/stress-cpu.c b/stress-cpu.c
index 0a08f1d..c1f1fc5 100644
--- a/stress-cpu.c
+++ b/stress-cpu.c
@@ -668,6 +668,7 @@ static void HOT OPTIMIZE3 stress_cpu_logmap(const char *name)
 	stress_double_put(x);
 }
 
+#if !defined(__serenity__)
 /*
  *  stress_cpu_rand48()
  *	generate random values using rand48 family of functions
@@ -688,6 +689,7 @@ static void HOT OPTIMIZE3 stress_cpu_rand48(const char *name)
 	stress_double_put(d);
 	stress_uint64_put((uint64_t)l);
 }
+#endif
 
 /*
  *  stress_cpu_lfsr32()
@@ -3116,7 +3118,9 @@ static const stress_cpu_method_info_t cpu_methods[] = {
 	{ "psi",		stress_cpu_psi },
 	{ "queens",		stress_cpu_queens },
 	{ "rand",		stress_cpu_rand },
+#if !defined(__serenity__)
 	{ "rand48",		stress_cpu_rand48 },
+#endif
 	{ "rgb",		stress_cpu_rgb },
 	{ "sieve",		stress_cpu_sieve },
 	{ "stats",		stress_cpu_stats },
