From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Gianforcaro <b.gianfo@gmail.com>
Date: Fri, 18 Mar 2022 02:35:30 -0700
Subject: [PATCH] serenity: Add bogus O_NDELAY just to allow the port to
 compile

---
 stress-dev.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/stress-dev.c b/stress-dev.c
index 7772097..aa98b0a 100644
--- a/stress-dev.c
+++ b/stress-dev.c
@@ -117,6 +117,11 @@
 #include <termio.h>
 #endif
 
+#if defined(__serenity__)
+    /* Serenity doesn't yet support O_NDELAY, so just make it compile */
+    #define  O_NDELAY (0)
+#endif
+
 /*
  *  Device information is held in a linked list of dev_info_t objects. Each
  *  nth element in the list also points to a unique device state which is
