From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Gianforcaro <b.gianfo@gmail.com>
Date: Fri, 18 Mar 2022 02:36:17 -0700
Subject: [PATCH] serenity: Disable nice() stress workload as we do not
 implement it

---
 stress-syncload.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/stress-syncload.c b/stress-syncload.c
index ca89124..7dd9e52 100644
--- a/stress-syncload.c
+++ b/stress-syncload.c
@@ -130,6 +130,7 @@ static void stress_syncload_loop(void)
 	}
 }
 
+#if !defined(__serenity__)
 static void stress_syncload_nice(void)
 {
 	int niceness;
@@ -137,6 +138,7 @@ static void stress_syncload_nice(void)
 	niceness = nice(0);
 	(void)niceness;
 }
+#endif
 
 static void stress_syncload_spinwrite(void)
 {
@@ -196,7 +198,10 @@ static const stress_syncload_op_t stress_syncload_ops[] = {
 #if defined(HAVE_VECMATH)
 	stress_syncload_vecmath,
 #endif
+
+#if !defined(__serenity__)
 	stress_syncload_nice,
+#endif
 	stress_syncload_spinwrite,
 };
 
