From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Gianforcaro <b.gianfo@gmail.com>
Date: Fri, 18 Mar 2022 02:36:55 -0700
Subject: [PATCH] serenity: Disable prctl stressor on serenity

---
 stress-usersyscall.c | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/stress-usersyscall.c b/stress-usersyscall.c
index 26732db..a4eb903 100644
--- a/stress-usersyscall.c
+++ b/stress-usersyscall.c
@@ -26,7 +26,8 @@ static const stress_help_t help[] = {
 };
 
 #if defined(SA_SIGINFO) &&	\
-    defined(HAVE_SYS_PRCTL_H)
+    defined(HAVE_SYS_PRCTL_H) && \
+    !defined(__serenity__)
 
 #include <sys/prctl.h>
 
