diff -Naur stress-ng-0.11.23/Makefile stress-ng-0.11.23.serenity/Makefile
--- stress-ng-0.11.23/Makefile	2021-04-16 00:30:33.161310251 +0200
+++ stress-ng-0.11.23.serenity/Makefile	2021-04-16 00:33:59.731692012 +0200
@@ -53,10 +53,10 @@
 CFLAGS += -DBUILD_STATIC
 endif
 
-BINDIR=/usr/bin
-MANDIR=/usr/share/man/man1
-JOBDIR=/usr/share/stress-ng/example-jobs
-BASHDIR=/usr/share/bash-completion/completions
+BINDIR=/usr/local/bin
+MANDIR=/usr/local/share/man/man1
+JOBDIR=/usr/local/share/stress-ng/example-jobs
+BASHDIR=/usr/local/share/bash-completion/completions
 
 #
 #  Stressors
