diff -ru freetype-2.10.4/builds/unix/config.sub freetype-2.10.4-patched/builds/unix/config.sub
--- freetype-2.10.4/builds/unix/config.sub	2020-10-20 07:15:01.000000000 +0200
+++ freetype-2.10.4-patched/builds/unix/config.sub	2021-03-05 21:50:35.776894354 +0100
@@ -1690,7 +1690,7 @@
 	# Now accept the basic system types.
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
-	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | serenity* | irix* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
