From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jesse Buhagiar <jooster669@gmail.com>
Date: Fri, 25 Mar 2022 09:45:12 +1100
Subject: [PATCH] Meta: Add `-ldl` library for Serenity target

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index b26ceb7..035fea7 100644
--- a/Makefile
+++ b/Makefile
@@ -860,7 +860,7 @@ ifeq ($(PLATFORM),serenity)
   SHLIBLDFLAGS=-shared $(LDFLAGS)
 
   THREAD_LIBS=-lpthread
-  LIBS=-lm
+  LIBS=-lm -ldl
 
   CLIENT_LIBS =
 
