From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dante Catalfamo <dante.catalfamo@gmail.com>
Date: Fri, 20 Aug 2021 00:38:31 -0400
Subject: [PATCH] Include sys/select.h

---
 mrbgems/mruby-io/src/io.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mrbgems/mruby-io/src/io.c b/mrbgems/mruby-io/src/io.c
index 0469c05..0dae6ca 100644
--- a/mrbgems/mruby-io/src/io.c
+++ b/mrbgems/mruby-io/src/io.c
@@ -49,6 +49,7 @@
   #include <sys/wait.h>
   #include <sys/time.h>
   #include <unistd.h>
+  #include <sys/select.h>
   typedef size_t fsize_t;
   typedef time_t ftime_t;
   typedef suseconds_t fsuseconds_t;
