From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gurkirat Singh <tbhaxor@gmail.com>
Date: Mon, 21 Aug 2023 05:47:52 +0530
Subject: [PATCH] Pretend that SerenityOS is Linux-like

---
 deps/filesystem/filesystem.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/filesystem/filesystem.hpp b/deps/filesystem/filesystem.hpp
index 53f4ad4efc0947583ae53b9b2a64a9f50557537b..60c0f194b669e62d079ec32d4c4bb8fa7d0ecbb4 100644
--- a/deps/filesystem/filesystem.hpp
+++ b/deps/filesystem/filesystem.hpp
@@ -54,7 +54,7 @@
 #ifndef GHC_OS_DETECTED
 #if defined(__APPLE__) && defined(__MACH__)
 #define GHC_OS_MACOS
-#elif defined(__linux__)
+#elif defined(__linux__) || defined(__serenity__)
 #define GHC_OS_LINUX
 #if defined(__ANDROID__)
 #define GHC_OS_ANDROID
