From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Mon, 16 May 2022 15:45:32 +0430
Subject: [PATCH] Stub out some wctype functions

---
 gettext-tools/gnulib-lib/fnmatch.c | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/gettext-tools/gnulib-lib/fnmatch.c b/gettext-tools/gnulib-lib/fnmatch.c
index b33a127d9802be15eeafef4622e63aac709bef8e..87f1f7c218c1708f84f3e4394eaa219e1d7c5858 100644
--- a/gettext-tools/gnulib-lib/fnmatch.c
+++ b/gettext-tools/gnulib-lib/fnmatch.c
@@ -112,6 +112,18 @@ typedef ptrdiff_t idx_t;
 # define CHAR_CLASS_MAX_LENGTH 256
 #endif
 
+#ifdef __serenity__
+wctype_t wctype(const char*)
+{
+    asm("int3");
+}
+
+int iswctype(wint_t, wctype_t)
+{
+    asm("int3");
+}
+#endif
+
 #define IS_CHAR_CLASS(string) wctype (string)
 
 /* Avoid depending on library functions or files
