From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: implicitfield <114500360+implicitfield@users.noreply.github.com>
Date: Mon, 5 Feb 2024 01:05:50 +0400
Subject: [PATCH] Install examples

---
 example/meson.build | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/example/meson.build b/example/meson.build
index 3ee2630..4f39c68 100644
--- a/example/meson.build
+++ b/example/meson.build
@@ -20,14 +20,14 @@ threaded_examples = [ 'notify_inval_inode',
 foreach ex : examples
     executable(ex, ex + '.c',
                dependencies: [ libfuse_dep ],
-               install: false)
+               install: true)
 endforeach
 
 
 foreach ex : threaded_examples
     executable(ex, ex + '.c',
                dependencies: [ thread_dep, libfuse_dep ],
-               install: false)
+               install: true)
 endforeach
 
 if not platform.endswith('bsd') and platform != 'dragonfly' and platform != 'serenity' and add_languages('cpp', required : false)
