From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: implicitfield <114500360+implicitfield@users.noreply.github.com>
Date: Mon, 5 Feb 2024 19:34:40 +0400
Subject: [PATCH] Teach the mount helpers about serenity

---
 lib/helper.c    | 4 ++++
 lib/mount_bsd.c | 4 ++++
 2 files changed, 8 insertions(+)

diff --git a/lib/helper.c b/lib/helper.c
index 357a0b6b60df315e5cc349d03f4ec213d6689d3c..9cdca9755922ec96221f0fa21ccc2e68cf60a4d3 100644
--- a/lib/helper.c
+++ b/lib/helper.c
@@ -178,6 +178,9 @@ static int fuse_helper_opt_proc(void *data, const char *arg, int key,
    function actually sets the fsname */
 static int add_default_subtype(const char *progname, struct fuse_args *args)
 {
+#ifdef __serenity__
+	return 0;
+#else
 	int res;
 	char *subtype_opt;
 
@@ -200,6 +203,7 @@ static int add_default_subtype(const char *progname, struct fuse_args *args)
 	res = fuse_opt_add_arg(args, subtype_opt);
 	free(subtype_opt);
 	return res;
+#endif
 }
 
 int fuse_parse_cmdline_312(struct fuse_args *args,
diff --git a/lib/mount_bsd.c b/lib/mount_bsd.c
index d49f62aacdbd9b9354468f27d5064694d530b3b1..3785ca5f9733b463f69a5a16161905dd47d57b3a 100644
--- a/lib/mount_bsd.c
+++ b/lib/mount_bsd.c
@@ -26,7 +26,11 @@
 #include <errno.h>
 #include <string.h>
 
+#ifdef __serenity__
+#define FUSERMOUNT_PROG		"fusermount"
+#else
 #define FUSERMOUNT_PROG		"mount_fusefs"
+#endif
 #define FUSE_DEV_TRUNK		"/dev/fuse"
 
 enum {
