From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Make configure build for serenity

---
 configure | 39 ++++++++++++---------------------------
 1 file changed, 12 insertions(+), 27 deletions(-)

diff --git a/configure b/configure
index 5507de7..16a074b 100755
--- a/configure
+++ b/configure
@@ -35,14 +35,13 @@ echo "file config.log: writing..."
 
 SOURCEDIR=`dirname "$0"`
 
-MANPATH_BASE="/usr/share/man:/usr/X11R6/man"
-MANPATH_DEFAULT="/usr/share/man:/usr/X11R6/man:/usr/local/man"
-OSENUM=
-OSNAME=
+MANPATH_BASE="/usr/share/man:/usr/local/share/man"
+MANPATH_DEFAULT="/usr/share/man:/usr/local/share/man"
+OSENUM=MANDOC_OS_OTHER
+OSNAME=Serenity
 UTF8_LOCALE=
 
-CC=`printf "all:\\n\\t@echo \\\$(CC)\\n" | env -i make -sf -`
-CFLAGS=
+CFLAGS="-O2 -pipe"
 LDADD=
 LDFLAGS=
 LD_NANOSLEEP=
@@ -62,17 +61,17 @@ HAVE_ENDIAN=
 HAVE_ERR=
 HAVE_FTS=
 HAVE_FTS_COMPARE_CONST=
-HAVE_GETLINE=
+HAVE_GETLINE=1
 HAVE_GETSUBOPT=
 HAVE_ISBLANK=
 HAVE_LESS_T=
-HAVE_MKDTEMP=
+HAVE_MKDTEMP=1
 HAVE_NANOSLEEP=
-HAVE_NTOHL=
+HAVE_NTOHL=1
 HAVE_O_DIRECTORY=
 HAVE_OHASH=
 HAVE_PATH_MAX=
-HAVE_PLEDGE=
+HAVE_PLEDGE=0
 HAVE_PROGNAME=
 HAVE_REALLOCARRAY=
 HAVE_RECALLOCARRAY=
@@ -84,7 +83,7 @@ HAVE_STRCASESTR=
 HAVE_STRINGLIST=
 HAVE_STRLCAT=
 HAVE_STRLCPY=
-HAVE_STRNDUP=
+HAVE_STRNDUP=1
 HAVE_STRPTIME=
 HAVE_STRSEP=
 HAVE_STRTONUM=
@@ -361,20 +360,6 @@ else
 	echo 1>&3
 fi
 
-# --- nanosleep ---
-if [ -n "${LD_NANOSLEEP}" ]; then
-	runtest nanosleep NANOSLEEP "${LD_NANOSLEEP}" || true
-elif singletest nanosleep NANOSLEEP; then
-	:
-elif runtest nanosleep NANOSLEEP "-lrt"; then
-	LD_NANOSLEEP="-lrt"
-fi
-if [ "${HAVE_NANOSLEEP}" -eq 0 ]; then
-	echo "FATAL: nanosleep: no" 1>&2
-	echo "FATAL: nanosleep: no" 1>&3
-	exit 1
-fi
-
 if [ ${BUILD_CATMAN} -gt 0 ]; then
 	# --- recvmsg ---
 	if [ -n "${LD_RECVMSG}" ]; then
@@ -420,7 +405,7 @@ if [ "${HAVE_OHASH}" -eq 0 ]; then
 fi
 
 # --- LDADD ---
-LDADD="${LDADD} ${LD_NANOSLEEP} ${LD_RECVMSG} ${LD_OHASH} -lz"
+LDADD="${LDADD} ${LD_NANOSLEEP} ${LD_RECVMSG} ${LD_OHASH} -lpcre2-posix -lpcre2-8 -lz"
 echo "selected LDADD=\"${LDADD}\"" 1>&2
 echo "selected LDADD=\"${LDADD}\"" 1>&3
 echo 1>&3
@@ -572,7 +557,7 @@ exec > Makefile.local
 [ -z "${BIN_FROM_SBIN}"   ] && BIN_FROM_SBIN="../bin"
 [ -z "${INCLUDEDIR}"      ] && INCLUDEDIR="${PREFIX}/include/mandoc"
 [ -z "${LIBDIR}"          ] && LIBDIR="${PREFIX}/lib/mandoc"
-[ -z "${MANDIR}"          ] && MANDIR="${PREFIX}/man"
+[ -z "${MANDIR}"          ] && MANDIR="${PREFIX}/share/man"
 
 [ -z "${HTDOCDIR}"        ] && HTDOCDIR="${WWWPREFIX}/htdocs"
 [ -z "${CGIBINDIR}"       ] && CGIBINDIR="${WWWPREFIX}/cgi-bin"
