From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Use pcre2 for regex

---
 dba_read.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dba_read.c b/dba_read.c
index 4fc3ee5eddcb49076ff6445f495aa1cee257068b..6ede1aaf2366aeea004c3256c48479d4e2e3aa32 100644
--- a/dba_read.c
+++ b/dba_read.c
@@ -21,7 +21,7 @@
  */
 #include "config.h"
 
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdio.h>
