From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Use pcre2 for regex

---
 dba_read.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dba_read.c b/dba_read.c
index e976057..6183dcb 100644
--- a/dba_read.c
+++ b/dba_read.c
@@ -19,7 +19,7 @@
  * The interface is defined in "dba.h".
  * This file is seperate from dba.c because this also uses "dbm.h".
  */
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <stdio.h>
