From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Use pcre2 for regex

---
 dbm_map.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbm_map.c b/dbm_map.c
index 87c085d22ec18fcccfd4bd0ce674c8976fdc485e..d5c620e0cfb2bcfaa6028b21fc31057db47858f0 100644
--- a/dbm_map.c
+++ b/dbm_map.c
@@ -36,7 +36,7 @@
 #endif
 #include <errno.h>
 #include <fcntl.h>
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdint.h>
 #include <stdlib.h>
 #include <string.h>
