From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Use pcre2 for regex

---
 dbm.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/dbm.c b/dbm.c
index f6b1259..3dc034a 100644
--- a/dbm.c
+++ b/dbm.c
@@ -31,7 +31,7 @@
 #include <err.h>
 #endif
 #include <errno.h>
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdint.h>
 #include <stdio.h>
 #include <stdlib.h>
