From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Build the patched-in glob implementation

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index 48c4741812b62a3be4520d648ea8392a695c5688..c5c835fdfc77d95d96314facff86250565b02382 100644
--- a/Makefile
+++ b/Makefile
@@ -65,6 +65,7 @@ SRCS		 = arch.c \
 		   compat_fts.c \
 		   compat_getline.c \
 		   compat_getsubopt.c \
+		   compat_glob.c \
 		   compat_isblank.c \
 		   compat_mkdtemp.c \
 		   compat_mkstemps.c \
@@ -257,6 +258,7 @@ ALL_COBJS	 = compat_err.o \
 		   compat_fts.o \
 		   compat_getline.o \
 		   compat_getsubopt.o \
+		   compat_glob.o \
 		   compat_isblank.o \
 		   compat_mkdtemp.o \
 		   compat_mkstemps.o \
