From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Build the patched-in glob implementation

---
 Makefile | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/Makefile b/Makefile
index f4e2954..51a9816 100644
--- a/Makefile
+++ b/Makefile
@@ -63,6 +63,7 @@ SRCS		 = arch.c \
 		   compat_fts.c \
 		   compat_getline.c \
 		   compat_getsubopt.c \
+		   compat_glob.c \
 		   compat_isblank.c \
 		   compat_mkdtemp.c \
 		   compat_ohash.c \
@@ -251,6 +252,7 @@ COMPAT_OBJS	 = compat_err.o \
 		   compat_fts.o \
 		   compat_getline.o \
 		   compat_getsubopt.o \
+		   compat_glob.o \
 		   compat_isblank.o \
 		   compat_mkdtemp.o \
 		   compat_ohash.o \
