From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sat, 25 Jan 2020 21:20:39 -0500
Subject: [PATCH] Fix mansearch.c

---
 mansearch.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/mansearch.c b/mansearch.c
index 0c457f9..b036f39 100644
--- a/mansearch.c
+++ b/mansearch.c
@@ -26,9 +26,9 @@
 #endif
 #include <errno.h>
 #include <fcntl.h>
-#include <glob.h>
+#include "glob.h"
 #include <limits.h>
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdio.h>
 #include <stdint.h>
 #include <stddef.h>
