From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Ledbetter <timledbetter@gmail.com>
Date: Fri, 18 Aug 2023 06:36:54 +0100
Subject: [PATCH] Link to zstd library

---
 BackwardConfig.cmake | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/BackwardConfig.cmake b/BackwardConfig.cmake
index cda0568bf9eaf76e7a3e263c1bd1aa950d1308a3..cd93ee07af2daf32fa2f1b6f550fcc8844c7215e 100644
--- a/BackwardConfig.cmake
+++ b/BackwardConfig.cmake
@@ -156,6 +156,9 @@ if (${STACK_DETAILS_AUTO_DETECT})
 			if (LIBSFRAME_LIBRARY)
 				list(APPEND _BACKWARD_LIBRARIES ${LIBSFRAME_LIBRARY})
 			endif()
+			if (LIBZSTD_LIBRARY)
+				list(APPEND _BACKWARD_LIBRARIES ${LIBZSTD_LIBRARY})
+			endif()
 
 			list(APPEND _BACKWARD_LIBRARIES iberty z)
 		endif()
