From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brendan Coles <bcoles@gmail.com>
Date: Thu, 8 Apr 2021 12:14:24 +0000
Subject: [PATCH] Use /bin/bash instead of /bin/sh for shebang

---
 bin/cob-config.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/bin/cob-config.in b/bin/cob-config.in
index 6c07e7d..3086d8c 100644
--- a/bin/cob-config.in
+++ b/bin/cob-config.in
@@ -1,4 +1,4 @@
-#!/bin/sh
+#!/bin/bash
 #
 # cob-config
 #
