--- gnucobol-3.1.2/build_aux/config.sub	2021-04-08 02:49:20.863867201 -0700
+++ gnucobol-3.1.2/build_aux/config.sub	2021-04-08 02:49:22.715869376 -0700
@@ -1339,6 +1339,7 @@
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
 	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
