From 0a0fd1875d7a58753e1c3da90e767e04ed06ef7f Mon Sep 17 00:00:00 2001
From: Snow <i@xkun.me>
Date: Sun, 15 May 2022 11:41:03 +0800
Subject: [PATCH 3/4] Teach config.sub about serenity

---
 config.sub | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index 9633db7..fe9a63e 100755
--- a/config.sub
+++ b/config.sub
@@ -1374,7 +1374,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -serenity*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
-- 
2.36.1

