From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: circl <circl.lastname@gmail.com>
Date: Sat, 19 Mar 2022 20:07:39 +0100
Subject: [PATCH] Serenity doesn't have setitimer

---
 src/lj_arch.h | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/lj_arch.h b/src/lj_arch.h
index ba7be19..0bd7b1b 100644
--- a/src/lj_arch.h
+++ b/src/lj_arch.h
@@ -490,6 +490,9 @@
 #define LJ_HASFFI		1
 #endif
 
+// FIXME: Disabled due to lack of setitimer
+#define LUAJIT_DISABLE_PROFILE
+
 #if defined(LUAJIT_DISABLE_PROFILE)
 #define LJ_HASPROFILE		0
 #elif LJ_TARGET_POSIX
