From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Thu, 23 Jan 2020 23:43:31 -0500
Subject: [PATCH] Disable pcre2test's `-S` flag on serenity

---
 src/pcre2test.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/pcre2test.c b/src/pcre2test.c
index ea52a20..e9454c4 100644
--- a/src/pcre2test.c
+++ b/src/pcre2test.c
@@ -8917,7 +8917,7 @@ while (argc > 1 && argv[op][0] == '-' && argv[op][1] != 0)
   else if (strcmp(arg, "-S") == 0 && argc > 2 &&
       ((uli = strtoul(argv[op+1], &endptr, 10)), *endptr == 0))
     {
-#if defined(_WIN32) || defined(WIN32) || defined(__HAIKU__) || defined(NATIVE_ZOS) || defined(__VMS)
+#if defined(_WIN32) || defined(WIN32) || defined(__HAIKU__) || defined(NATIVE_ZOS) || defined(__VMS) || defined(__serenity__)
     fprintf(stderr, "pcre2test: -S is not supported on this OS\n");
     exit(1);
 #else
