From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Mon, 16 May 2022 15:31:45 +0430
Subject: [PATCH] Disable logging

---
 mem.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/mem.h b/mem.h
index 8c96952..6c39fd6 100644
--- a/mem.h
+++ b/mem.h
@@ -7,7 +7,7 @@
 #define MAX(a,b)          ((a)>(b)?(a):(b))
 #define MIN(a,b)          ((a)<(b)?(a):(b))
 
-#define DISABLE_LOGGING   0
+#define DISABLE_LOGGING   1
 
 void mem_init(int romsize);
 int load_bin(const char *fn);
