From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: aabajyan <arsen.abajyan@pm.me>
Date: Sun, 7 Mar 2021 22:30:13 +0400
Subject: [PATCH] Add a missing cstdlib include

---
 mem.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/mem.cpp b/mem.cpp
index fd36d68..4cf0b30 100644
--- a/mem.cpp
+++ b/mem.cpp
@@ -10,6 +10,7 @@ extern "C" {
 #include "vdp.h"
 #include "cpu.h"
 #include "ioports.h"
+#include <cstdlib>
 
 uint8_t *ROM;
 uint8_t RAM[0x10000];
