From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Mon, 16 May 2022 15:32:14 +0430
Subject: [PATCH] Use software rendering

---
 hw.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/hw.c b/hw.c
index 6b864df..1f34423 100644
--- a/hw.c
+++ b/hw.c
@@ -1,5 +1,5 @@
 #include "hw.h"
-#include <SDL.h>
+#include <SDL2/SDL.h>
 #include <assert.h>
 #include <time.h>
 
@@ -105,7 +105,7 @@ void hw_enable_video(int enable)
         screen = SDL_CreateWindow("Genemu - Sega Genesis Emulator",
             SDL_WINDOWPOS_UNDEFINED, SDL_WINDOWPOS_UNDEFINED,
             WINDOW_WIDTH, WINDOW_WIDTH*3/4, SDL_WINDOW_RESIZABLE);
-        renderer = SDL_CreateRenderer(screen, -1, SDL_RENDERER_PRESENTVSYNC);
+        renderer = SDL_CreateRenderer(screen, -1, SDL_RENDERER_SOFTWARE);
 
         SDL_SetHint(SDL_HINT_RENDER_SCALE_QUALITY, "nearest");  // make the scaled rendering look smoother.
         SDL_RenderSetLogicalSize(renderer, 320, 240);
