From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Liav A <liavalb@gmail.com>
Date: Sun, 16 Oct 2022 13:14:06 +0300
Subject: [PATCH] Don't try to use <malloc.h> header file in the codebase

---
 configure     | 2 +-
 src/miniacc.h | 1 -
 2 files changed, 1 insertion(+), 2 deletions(-)

diff --git a/configure b/configure
index 77ac278aaf10a04c0a4d97ffaef5f083912f3f51..7a00285647bf78fdeb3e159dbd55be412b841743 100755
--- a/configure
+++ b/configure
@@ -5759,7 +5759,7 @@ $as_echo "#define TIME_WITH_SYS_TIME 1" >>confdefs.h
 
 fi
 
-for ac_header in assert.h ctype.h dirent.h errno.h fcntl.h float.h limits.h malloc.h memory.h setjmp.h signal.h stdarg.h stddef.h stdint.h stdio.h stdlib.h string.h strings.h time.h unistd.h utime.h sys/mman.h sys/resource.h sys/stat.h sys/time.h sys/types.h sys/wait.h
+for ac_header in assert.h ctype.h dirent.h errno.h fcntl.h float.h limits.h memory.h setjmp.h signal.h stdarg.h stddef.h stdint.h stdio.h stdlib.h string.h strings.h time.h unistd.h utime.h sys/mman.h sys/resource.h sys/stat.h sys/time.h sys/types.h sys/wait.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
diff --git a/src/miniacc.h b/src/miniacc.h
index 5f44085fad3e906785b090320a6d03e5fdcc17d3..4d75d7bb60ead2560f7db454ff52ee677b8dbcf6 100644
--- a/src/miniacc.h
+++ b/src/miniacc.h
@@ -3365,7 +3365,6 @@ ACC_COMPILE_TIME_ASSERT_HEADER(sizeof(acc_int_fast64_t) == sizeof(acc_uint_fast6
 #define HAVE_FCNTL_H 1
 #define HAVE_FLOAT_H 1
 #define HAVE_LIMITS_H 1
-#define HAVE_MALLOC_H 1
 #define HAVE_MEMORY_H 1
 #define HAVE_SETJMP_H 1
 #define HAVE_SIGNAL_H 1
