From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brendan Coles <bcoles@gmail.com>
Date: Wed, 14 Apr 2021 17:24:58 +0000
Subject: [PATCH] Include <strings.h>

---
 src/codepage.c | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/codepage.c b/src/codepage.c
index a67e230dde65783a88a2ef26c493e2352f43ba5b..26fdb1ae591d9dc58e2c34a60bd04cfa2621bcc3 100644
--- a/src/codepage.c
+++ b/src/codepage.c
@@ -3,6 +3,7 @@
 #include "env.h"
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 
 /* List of code-pages */
 struct cp_data
