From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Thu, 15 Apr 2021 15:43:18 +0200
Subject: [PATCH] Install to /usr/local

---
 Makefile | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 069b06c..34b38d8 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,9 @@ distcheck: $(distdir).tar.gz
 	@echo "*** Package $(distdir).tar.gz is ready for distribution."
 
 install: all
-	install src/nyancat /usr/bin/${package}
-	gzip -9 -c < nyancat.1 > /usr/share/man/man1/nyancat.1.gz
+	mkdir -p ${DESTDIR}/usr/local/bin
+	install src/nyancat ${DESTDIR}/usr/local/bin/${package}
+	mkdir -p ${DESTDIR}/usr/local/share/man/man1
+	gzip -9 -c < nyancat.1 > ${DESTDIR}/usr/local/share/man/man1/nyancat.1.gz
 
 .PHONY: FORCE all clean check dist distcheck install
