From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Thu, 15 Apr 2021 15:43:18 +0200
Subject: [PATCH] Use %d for time diff printing

---
 src/nyancat.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/nyancat.c b/src/nyancat.c
index 537225c..f2965c1 100644
--- a/src/nyancat.c
+++ b/src/nyancat.c
@@ -901,7 +901,7 @@ int main(int argc, char ** argv) {
 			 * The \033[0m prevents the Apple ][ from flipping everything, but
 			 * makes the whole nyancat less bright on the vt220
 			 */
-			printf("\033[1;37mYou have nyaned for %0.0f seconds!\033[J\033[0m", diff);
+			printf("\033[1;37mYou have nyaned for %d seconds!\033[J\033[0m", (int)diff);
 		}
 		/* Reset the last color so that the escape sequences rewrite */
 		last = 0;
