diff --git a/Makefile b/Makefile
index 1dd50ef..e435044 100644
--- a/Makefile
+++ b/Makefile
@@ -33,7 +33,9 @@ distcheck: $(distdir).tar.gz
 	@echo "*** Package $(distdir).tar.gz is ready for distribution."
 
 install: all
-	install src/nyancat /usr/bin/${package}
-	gzip -9 -c < nyancat.1 > /usr/share/man/man1/nyancat.1.gz
+	mkdir -p ${DESTDIR}/usr/local/bin
+	install src/nyancat ${DESTDIR}/usr/local/bin/${package}
+	mkdir -p ${DESTDIR}/usr/local/share/man/man1
+	gzip -9 -c < nyancat.1 > ${DESTDIR}/usr/local/share/man/man1/nyancat.1.gz
 
 .PHONY: FORCE all clean check dist distcheck install
diff --git a/src/nyancat.c b/src/nyancat.c
index 537225c..f2965c1 100644
--- a/src/nyancat.c
+++ b/src/nyancat.c
@@ -901,7 +901,7 @@ int main(int argc, char ** argv) {
 			 * The \033[0m prevents the Apple ][ from flipping everything, but
 			 * makes the whole nyancat less bright on the vt220
 			 */
-			printf("\033[1;37mYou have nyaned for %0.0f seconds!\033[J\033[0m", diff);
+			printf("\033[1;37mYou have nyaned for %d seconds!\033[J\033[0m", (int)diff);
 		}
 		/* Reset the last color so that the escape sequences rewrite */
 		last = 0;
