From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?S=C3=B6nke=20Holz?= <soenke.holz@serenityos.org>
Date: Thu, 2 Apr 2026 18:26:43 +0200
Subject: [PATCH] feat(provider): support serenity clipboard

---
 runtime/autoload/provider/clipboard.vim | 12 ++++++++++++
 1 file changed, 12 insertions(+)

diff --git a/runtime/autoload/provider/clipboard.vim b/runtime/autoload/provider/clipboard.vim
index fbd692ce4a557e17682a4c6d63657b18ebea89cb..2acb997115ecd78648c3bba673744d8f38123c62 100644
--- a/runtime/autoload/provider/clipboard.vim
+++ b/runtime/autoload/provider/clipboard.vim
@@ -174,6 +174,14 @@ function! s:set_tmux() abort
   return 'tmux'
 endfunction
 
+function! s:set_serenity() abort
+  let s:copy['+'] = ['copy', '--type', 'text/plain']
+  let s:paste['+'] = ['paste', '--no-newline']
+  let s:copy['*'] = s:copy['+']
+  let s:paste['*'] = s:paste['+']
+  return 'serenity'
+endfunction
+
 let s:cache_enabled = 1
 let s:err = ''
 
@@ -191,6 +199,8 @@ function! provider#clipboard#Executable() abort
         return s:set_osc52()
       elseif 'pbcopy' == g:clipboard
         return s:set_pbcopy()
+      elseif 'serenity' == g:clipboard
+        return s:set_serenity()
       elseif 'wl-copy' == g:clipboard
         return s:set_wayland()
       elseif 'wayclip' == g:clipboard
@@ -235,6 +245,8 @@ function! provider#clipboard#Executable() abort
     return get(g:clipboard, 'name', 'g:clipboard')
   elseif has('mac')
     return s:set_pbcopy()
+  elseif has('serenity')
+    return s:set_serenity()
   elseif !empty($WAYLAND_DISPLAY) && executable('wl-copy') && executable('wl-paste')
     return s:set_wayland()
   elseif !empty($WAYLAND_DISPLAY) && executable('waycopy') && executable('waypaste')
