From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andre Herbst <moormaster@gmx.net>
Date: Sat, 8 Jul 2023 14:17:19 +0200
Subject: [PATCH] zlib: Fix implicit write() method declaration error

---
 zlib/gzguts.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zlib/gzguts.h b/zlib/gzguts.h
index eba72085bb756bc4fac374c59e8f5fdd1e660eab..4fe9d4811185b38c08114858212d9a35ad7b4e21 100644
--- a/zlib/gzguts.h
+++ b/zlib/gzguts.h
@@ -18,6 +18,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 #include "zlib.h"
 #ifdef STDC
 #  include <string.h>
