From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: sin-ack <sin-ack@users.noreply.github.com>
Date: Mon, 17 Jul 2023 03:26:04 +0300
Subject: [PATCH] build: Set Zig version to 0.15.0-dev.208+8acedfd5b

---
 build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/build b/build
index 66f61dcab57f5cf451a512b308968b80989e6915..dfb96847b0a9a0bcbfab2b5f164e025f80312c72 100755
--- a/build
+++ b/build
@@ -7,7 +7,7 @@ TARGET="$1" # Example: riscv64-linux-gnu
 MCPU="$2" # Examples: `baseline`, `native`, `generic+v7a`, or `arm1176jzf_s`
 
 ROOTDIR="$(pwd)"
-ZIG_VERSION="0.14.0-dev.3299+31f353cd9"
+ZIG_VERSION="0.15.0-dev.208+8acedfd5b"
 
 TARGET_OS_AND_ABI=${TARGET#*-} # Example: linux-gnu
 
