From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andre Herbst <moormaster@gmx.net>
Date: Sat, 8 Jul 2023 14:17:19 +0200
Subject: [PATCH] zlib: Fix implicit write() method declaration error

---
 zlib/gzguts.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/zlib/gzguts.h b/zlib/gzguts.h
index 57faf37165a3546c3d0a783457a51e7dedff49ae..3c700c2a908cf4f675299fb4769f905533945831 100644
--- a/zlib/gzguts.h
+++ b/zlib/gzguts.h
@@ -19,6 +19,7 @@
 #endif
 
 #include <stdio.h>
+#include <unistd.h>
 #include "zlib.h"
 #ifdef STDC
 #  include <string.h>
