diff -Nru vttest-orig/config.sub vttest-20210210/config.sub
--- vttest-orig/config.sub	2020-08-17 05:36:41.000000000 +0200
+++ vttest-20210210/config.sub	2021-02-20 16:22:14.126987759 +0100
@@ -1700,7 +1700,7 @@
 	     | os9* | macos* | osx* | ios* \
 	     | mpw* | magic* | mmixware* | mon960* | lnews* \
 	     | amigaos* | amigados* | msdos* | newsos* | unicos* | aof* \
-	     | aos* | aros* | cloudabi* | sortix* | twizzler* \
+	     | aos* | aros* | cloudabi* | sortix* | twizzler* | serenity* \
 	     | nindy* | vxsim* | vxworks* | ebmon* | hms* | mvs* \
 	     | clix* | riscos* | uniplus* | iris* | isc* | rtu* | xenix* \
 	     | mirbsd* | netbsd* | dicos* | openedition* | ose* \
