diff --git a/config.sub b/config.sub
index 5b158ac4..cfd2eb81 100755
--- a/config.sub
+++ b/config.sub
@@ -1342,6 +1342,7 @@ case $os in
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
 	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
