--- openssl-1.1.1k/Configurations/shared-info.pl	2021-03-25 21:28:38.000000000 +0800
+++ openssl-1.1.1k/Configurations/shared-info.pl	2021-09-26 00:05:04.340004623 +0800
@@ -34,6 +34,13 @@
             shared_defflag    => '-Wl,--version-script=',
         };
     },
+    'serenity-shared' => {
+        shared_ldflag         => '-shared',
+        shared_sonameflag     => '-Wl,-soname=',
+        # -Map is not really what should be used here, but we need a flag that
+        # accepts a filename and that it's '-version-map'
+        shared_defflag        => "-Wl,-Map=",
+    },
     'bsd-gcc-shared' => sub { return $shared_info{'linux-shared'}; },
     'bsd-shared' => sub {
         return $shared_info{'gnu-shared'} if detect_gnu_ld();
