diff -ru gcc/isl/config.sub gcc-patched/isl/config.sub
--- gcc/isl/config.sub	2016-01-24 21:50:53.000000000 +0100
+++ gcc-patched/isl/config.sub	2019-05-28 16:34:43.396758421 +0200
@@ -1361,6 +1361,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
diff -ru gcc/mpc/config.sub gcc-patched/mpc/config.sub
--- gcc/mpc/config.sub	2015-02-16 13:28:25.000000000 +0100
+++ gcc-patched/mpc/config.sub	2019-05-28 16:34:22.308458127 +0200
@@ -1369,6 +1369,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
diff -ru gcc/mpfr/config.sub gcc-patched/mpfr/config.sub
--- gcc/mpfr/config.sub	2016-03-06 12:33:18.000000000 +0100
+++ gcc-patched/mpfr/config.sub	2019-05-28 16:34:33.052611048 +0200
@@ -1375,6 +1375,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
diff -ru gcc/gmp/configfsf.sub gcc-patched/gmp/configfsf.sub
--- gcc/gmp/configfsf.sub	2015-11-01 16:19:48.000000000 +0100
+++ gcc-patched/gmp/configfsf.sub	2019-05-28 16:47:14.095698619 +0200
@@ -1375,6 +1375,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
