From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Brehm <cooljohnny3@gmail.com>
Date: Mon, 16 May 2022 22:58:31 +0430
Subject: [PATCH] Link with -ldl -liconv on serenity

---
 7zip/CMAKE/7za/CMakeLists.txt | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/7zip/CMAKE/7za/CMakeLists.txt b/7zip/CMAKE/7za/CMakeLists.txt
index 7fc1102..9a3511d 100644
--- a/7zip/CMAKE/7za/CMakeLists.txt
+++ b/7zip/CMAKE/7za/CMakeLists.txt
@@ -357,6 +357,10 @@ add_executable(7za
   "../../../../CPP/myWindows/wine_date_and_time.cpp"
 )
 
+IF(SERENITYOS)
+    TARGET_LINK_LIBRARIES(7za ${CMAKE_THREAD_LIBS_INIT} dl iconv)
+ENDIF(SERENITYOS)
+
 
 IF(APPLE)
    TARGET_LINK_LIBRARIES(7za ${COREFOUNDATION_LIBRARY} ${CMAKE_THREAD_LIBS_INIT})
