From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Brehm <cooljohnny3@gmail.com>
Date: Mon, 16 May 2022 22:58:31 +0430
Subject: [PATCH] Disable wctype related stuff

---
 myWindows/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/myWindows/config.h b/myWindows/config.h
index 497d197..fa402c6 100644
--- a/myWindows/config.h
+++ b/myWindows/config.h
@@ -36,7 +36,7 @@
     #define ENV_HAVE_TOWUPPER
   #endif
 		
-  
+  #undef ENV_HAVE_WCTYPE_H
   
   #if !defined(ENV_BEOS) && !defined(ANDROID_NDK)
 
