From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: John Brehm <cooljohnny3@gmail.com>
Date: Mon, 16 May 2022 22:58:31 +0430
Subject: [PATCH] Add a missing strings.h include

---
 7zip/Archive/Zip/ZipItem.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/7zip/Archive/Zip/ZipItem.cpp b/7zip/Archive/Zip/ZipItem.cpp
index 353e895..ad0a3da 100644
--- a/7zip/Archive/Zip/ZipItem.cpp
+++ b/7zip/Archive/Zip/ZipItem.cpp
@@ -19,6 +19,7 @@
 #include "../Common/ItemNameUtils.h"
 
 #include "ZipItem.h"
+#include <strings.h>
 
 namespace NArchive {
 namespace NZip {
