diff -ruN flatbuffers-1.12.0/src/code_generators.cpp flatbuffers-1.12.0-serenity/src/code_generators.cpp
--- flatbuffers-1.12.0/src/code_generators.cpp	2020-03-12 19:33:39.000000000 -0300
+++ flatbuffers-1.12.0-serenity/src/code_generators.cpp	2021-03-31 20:39:12.000000000 -0300
@@ -23,6 +23,17 @@
 #include "flatbuffers/base.h"
 #include "flatbuffers/util.h"
 
+#if defined(__serenity__)
+  // We do not have those functions inside std namespace...
+
+namespace std {
+  auto isnan(double x) { return ::isnan(x); }
+  auto isinf(double x) { return ::isinf(x); }
+}
+
+#endif
+
+
 #if defined(_MSC_VER)
 #  pragma warning(push)
 #  pragma warning(disable : 4127)  // C4127: conditional expression is constant
