From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: TheFightingCatfish <seekingblues@gmail.com>
Date: Fri, 13 Aug 2021 06:52:59 +0800
Subject: [PATCH] Remove uname detection in config.mak.uname

---
 config.mak.uname | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/config.mak.uname b/config.mak.uname
index d63629fe807f59deda80ed780c1915df011bf862..eae96e0ab2986c4b96a9786715cf778acdb18736 100644
--- a/config.mak.uname
+++ b/config.mak.uname
@@ -4,12 +4,12 @@
 # Microsoft's Safe Exception Handling in libraries (such as zlib).
 # Typically required for VS2013+/32-bit compilation on Vista+ versions.
 
-uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
-uname_M := $(shell sh -c 'uname -m 2>/dev/null || echo not')
-uname_O := $(shell sh -c 'uname -o 2>/dev/null || echo not')
-uname_R := $(shell sh -c 'uname -r 2>/dev/null || echo not')
-uname_P := $(shell sh -c 'uname -p 2>/dev/null || echo not')
-uname_V := $(shell sh -c 'uname -v 2>/dev/null || echo not')
+# uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
+# uname_M := $(shell sh -c 'uname -m 2>/dev/null || echo not')
+# uname_O := $(shell sh -c 'uname -o 2>/dev/null || echo not')
+# uname_R := $(shell sh -c 'uname -r 2>/dev/null || echo not')
+# uname_P := $(shell sh -c 'uname -p 2>/dev/null || echo not')
+# uname_V := $(shell sh -c 'uname -v 2>/dev/null || echo not')
 
 ifneq ($(findstring MINGW,$(uname_S)),)
 	uname_S := MINGW
