--- git-2.26.0/config.mak.uname.orig	2021-03-13 13:30:24.439217597 +0100
+++ git-2.26.0/config.mak.uname	2021-03-13 13:30:32.747203874 +0100
@@ -57,7 +57,7 @@
 	HAVE_GETDELIM = YesPlease
 	SANE_TEXT_GREP=-a
 	FREAD_READS_DIRECTORIES = UnfortunatelyYes
-	BASIC_CFLAGS += -DHAVE_SYSINFO
+	# BASIC_CFLAGS += -DHAVE_SYSINFO
 	PROCFS_EXECUTABLE_PATH = /proc/self/exe
 endif
 ifeq ($(uname_S),GNU/kFreeBSD)
