diff -ru git-2.25.1/config.c git-2.25.1-serenity/config.c
--- git-2.25.1/config.c	2020-02-17 05:47:35.000000000 +0100
+++ git-2.25.1-serenity/config.c	2020-02-20 06:44:32.062023438 +0100
@@ -2841,7 +2841,7 @@
 
 		contents_sz = xsize_t(st.st_size);
 		contents = xmmap_gently(NULL, contents_sz, PROT_READ,
-					MAP_PRIVATE, in_fd, 0);
+					MAP_SHARED, in_fd, 0);
 		if (contents == MAP_FAILED) {
 			if (errno == ENODEV && S_ISDIR(st.st_mode))
 				errno = EISDIR;
diff -ru git-2.25.1/config.mak.uname git-2.25.1-serenity/config.mak.uname
--- git-2.25.1/config.mak.uname	2020-02-17 05:47:35.000000000 +0100
+++ git-2.25.1-serenity/config.mak.uname	2020-02-20 06:20:22.532687832 +0100
@@ -4,12 +4,12 @@
 # Microsoft's Safe Exception Handling in libraries (such as zlib).
 # Typically required for VS2013+/32-bit compilation on Vista+ versions.
 
-uname_S := $(shell sh -c 'uname -s 2>/dev/null || echo not')
-uname_M := $(shell sh -c 'uname -m 2>/dev/null || echo not')
-uname_O := $(shell sh -c 'uname -o 2>/dev/null || echo not')
-uname_R := $(shell sh -c 'uname -r 2>/dev/null || echo not')
-uname_P := $(shell sh -c 'uname -p 2>/dev/null || echo not')
-uname_V := $(shell sh -c 'uname -v 2>/dev/null || echo not')
+uname_S := 
+uname_M := 
+uname_O := 
+uname_R := 
+uname_P := 
+uname_V := 
 
 ifdef MSVC
 	# avoid the MingW and Cygwin configuration sections
