From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Emily Trau <emily@downunderctf.com>
Date: Tue, 6 Jun 2023 14:20:15 -0700
Subject: [PATCH 1/2] Exclude arpa/nameser.h as it does not exist on Serenity

---
 compats.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/compats.c b/compats.c
index 92d3b71..5d6f31a 100644
--- a/compats.c
+++ b/compats.c
@@ -215,7 +215,6 @@ warnx(const char *fmt, ...)
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
-#include <arpa/nameser.h>
 
 #include <ctype.h>
 #include <resolv.h>
-- 
2.37.2

