From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Emily Trau <emily@downunderctf.com>
Date: Tue, 6 Jun 2023 14:21:50 -0700
Subject: [PATCH 2/2] Don't use `getprogname()`

---
 main.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 1d192be..6867fe4 100644
--- a/main.c
+++ b/main.c
@@ -384,7 +384,9 @@ main(int argc, char *argv[])
 		LOWDOWN_LATEX_NUMBERED |
 		LOWDOWN_SMARTY;
 
-	if (strcasecmp(getprogname(), "lowdown-diff") == 0) 
+	char progname[BUFSIZ];
+	get_process_name(progname, sizeof(progname));
+	if (strcasecmp(progname, "lowdown-diff") == 0) 
 		diff = 1;
 
 	while ((c = getopt_long(argc, argv, 
-- 
2.37.2

