From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Manuel Palenzuela <manuelpalenzuelamerino@gmail.com>
Date: Sun, 4 Apr 2021 00:41:48 +0200
Subject: [PATCH] Fix a header include path

---
 src/FireBall.cpp | 4 ++--
 src/Map.h        | 4 ++--
 2 files changed, 4 insertions(+), 4 deletions(-)

diff --git a/src/FireBall.cpp b/src/FireBall.cpp
index 6f3eacc..6f5eade 100644
--- a/src/FireBall.cpp
+++ b/src/FireBall.cpp
@@ -1,4 +1,4 @@
-#include "FireBall.h"
+#include "Fireball.h"
 #include "Core.h"
 #include <math.h>
 
@@ -68,4 +68,4 @@ void FireBall::updateXPos() {
 
 void FireBall::collisionWithPlayer(bool TOP) {
 	CCore::getMap()->playerDeath(true, false);
-}
\ No newline at end of file
+}
diff --git a/src/Map.h b/src/Map.h
index b812b49..f6abbb9 100644
--- a/src/Map.h
+++ b/src/Map.h
@@ -28,7 +28,7 @@
 #include "Flower.h"
 #include "Lakito.h"
 #include "Spikey.h"
-#include "FireBall.h"
+#include "Fireball.h"
 #include "Fire.h"
 #include "Pipe.h"
 #include "Spring.h"
@@ -406,4 +406,4 @@ public:
 	void setInEvent(bool inEvent);
 };
 
-#endif
\ No newline at end of file
+#endif
