From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Fri, 8 Dec 2023 23:13:54 +0100
Subject: [PATCH] Remove the dependency on termcap

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 8bf1951683863566f383c4b1d450f0b8057ab8e6..32a3ea253783c9a3338a77bfa782aefd95d13ca1 100644
--- a/Makefile
+++ b/Makefile
@@ -49,7 +49,7 @@ OUTPUT_OPTION = -MMD -MP -o $@
 LD = $(CXX)
 
 # CImg requires pthread, for some reason
-LDLIBS = $(LIBS) -ltermcap -lm -lpthread
+LDLIBS = $(LIBS) -lm -lpthread
 
 # Get the source files.
 SOURCES = $(wildcard src/*.c) $(wildcard src/*.cc)
