From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Kaster <akaster@serenityos.org>
Date: Sat, 8 Jan 2022 05:33:27 -0700
Subject: [PATCH] Teach configure about serenity

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index 367f505..df6217c 100755
--- a/configure
+++ b/configure
@@ -11067,6 +11067,8 @@ $as_echo "$lt_cv_irix_exported_symbol" >&6; }
       hardcode_shlibpath_var=no
       ;;
 
+    serenity*)
+      ;;
     *)
       ld_shlibs=no
       ;;
@@ -12147,6 +12149,8 @@ uts4*)
   shlibpath_var=LD_LIBRARY_PATH
   ;;
 
+serenity*)
+  ;;
 *)
   dynamic_linker=no
   ;;
