diff -Naur lua-5.3.6/src/Makefile lua-5.3.6.serenity/src/Makefile
--- lua-5.3.6/src/Makefile	2020-07-13 20:38:14.000000000 +0200
+++ lua-5.3.6.serenity/src/Makefile	2021-12-27 21:33:37.015610414 +0100
@@ -26,7 +26,7 @@
 
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
-PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
+PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris serenity
 
 LUA_A=	liblua.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
@@ -100,7 +100,6 @@
 	@echo '*** C89 does not guarantee 64-bit integers for Lua.'
 	@echo ''
 
-
 freebsd:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_LINUX -I/usr/include/edit" SYSLIBS="-Wl,-E -ledit" CC="cc"
 
@@ -124,6 +123,13 @@
 solaris:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN -D_REENTRANT" SYSLIBS="-ldl"
 
+serenity:
+	$(MAKE) $(ALL) \
+		CC="$(CC) -std=gnu99" \
+		AR="$(AR) rcu" \
+		SYSCFLAGS="-DLUA_USE_DLOPEN" \
+		SYSLIBS="-ldl"
+
 # list targets that do not create files (but not all makes understand .PHONY)
 .PHONY: all $(PLATS) default o a clean depend echo none
 
