diff -Naur lua-5.3.5/Makefile lua-5.3.5.serenity/Makefile
--- lua-5.3.5/Makefile	2016-12-20 11:26:08.000000000 -0500
+++ lua-5.3.5.serenity/Makefile	2020-04-18 03:41:23.000000000 -0400
@@ -36,7 +36,7 @@
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
 # Convenience platforms targets.
-PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
+PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris serenity
 
 # What to install.
 TO_BIN= lua luac
diff -Naur lua-5.3.5/src/Makefile lua-5.3.5.serenity/src/Makefile
--- lua-5.3.5/src/Makefile	2018-06-25 13:46:36.000000000 -0400
+++ lua-5.3.5.serenity/src/Makefile	2020-04-18 03:51:15.000000000 -0400
@@ -26,7 +26,7 @@
 
 # == END OF USER SETTINGS -- NO NEED TO CHANGE ANYTHING BELOW THIS LINE =======
 
-PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris
+PLATS= aix bsd c89 freebsd generic linux macosx mingw posix solaris serenity
 
 LUA_A=	liblua.a
 CORE_O=	lapi.o lcode.o lctype.o ldebug.o ldo.o ldump.o lfunc.o lgc.o llex.o \
@@ -124,6 +124,13 @@
 solaris:
 	$(MAKE) $(ALL) SYSCFLAGS="-DLUA_USE_POSIX -DLUA_USE_DLOPEN -D_REENTRANT" SYSLIBS="-ldl"
 
+serenity:
+	$(MAKE) $(ALL) \
+		CC="i686-pc-serenity-gcc -std=gnu99" \
+		AR="i686-pc-serenity-ar rcu" \
+		RANLIB="i686-pc-serenity-ranlib" \
+		SYSCFLAGS="-DLUA_USE_DLOPEN"
+
 # list targets that do not create files (but not all makes understand .PHONY)
 .PHONY: all $(PLATS) default o a clean depend echo none
 
