From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Beckett Normington <beckett@b0ba.dev>
Date: Thu, 10 Aug 2023 09:48:21 -0400
Subject: [PATCH] parse_exec: replace wait3() with waitpid()

---
 xorriso/parse_exec.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/xorriso/parse_exec.c b/xorriso/parse_exec.c
index e94abcbac719e82a617f4135a77686632d04866b..52fb60dc516ed30f2debd921e4e52145b63cfb54 100644
--- a/xorriso/parse_exec.c
+++ b/xorriso/parse_exec.c
@@ -2988,7 +2988,7 @@ int Xorriso_execv(struct XorrisO *xorriso, char *cmd,
 
  Xorriso_alloc_meM(prog, char, 5 * SfileadrL);
 
- wait3(NULL,WNOHANG,NULL); /* just to remove any old dead child */
+ waitpid(-1,NULL,WNOHANG); /* just to remove any old dead child */
 
  if(flag & 2) {
    ret= Xorriso_make_argv_with_null(xorriso, in_argc, in_argv,
