From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Timur Sultanov <SultanovTS@yandex.ru>
Date: Thu, 17 Feb 2022 09:37:02 +0300
Subject: [PATCH] make: Remove CUPS dependency

---
 make/autoconf/libraries.m4 | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/make/autoconf/libraries.m4 b/make/autoconf/libraries.m4
index a65d91ee974bff4248cd39cce662159f23f84720..7c779d01201661e3f5cd124c2e86a8d554b9d502 100644
--- a/make/autoconf/libraries.m4
+++ b/make/autoconf/libraries.m4
@@ -26,7 +26,7 @@
 # Major library component reside in separate files.
 m4_include([lib-alsa.m4])
 m4_include([lib-bundled.m4])
-m4_include([lib-cups.m4])
+# m4_include([lib-cups.m4])
 m4_include([lib-ffi.m4])
 m4_include([lib-freetype.m4])
 m4_include([lib-std.m4])
