From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Fri, 24 Dec 2021 15:24:16 +0100
Subject: [PATCH] Disable full screen

---
 src/game_config.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/game_config.c b/src/game_config.c
index 6d1e979..35b6036 100644
--- a/src/game_config.c
+++ b/src/game_config.c
@@ -505,7 +505,7 @@ void init_game_configuration()
         "# decreasing this number, at the cost of lower image quality." );
 
     INIT_PARAM_BOOL( 
-	fullscreen, True,
+	fullscreen, False,
 	"# If true then the game will run in full-screen mode." );
 
     INIT_PARAM_INT( 
