From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Grigoris Pavlakis <grigpavl@ece.auth.gr>
Date: Mon, 30 May 2022 01:54:22 +0300
Subject: [PATCH] Exit event loop on SDL_QUIT event

---
 src/winsys.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/winsys.c b/src/winsys.c
index d932a1d..5385cb7 100644
--- a/src/winsys.c
+++ b/src/winsys.c
@@ -392,6 +392,10 @@ void winsys_process_events()
 		}
 		
 		break;
+      
+      case SDL_QUIT:
+        winsys_exit( 0 );
+        break;
 	    }
 
 	    SDL_LockAudio();
