From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sahan Fernando <sahan.h.fernando@gmail.com>
Date: Sat, 22 Jan 2022 00:54:01 +1100
Subject: [PATCH] Manually link against lxml2 and ldl

---
 fontconfig.pc.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/fontconfig.pc.in b/fontconfig.pc.in
index 61b35fb..a3c45eb 100644
--- a/fontconfig.pc.in
+++ b/fontconfig.pc.in
@@ -13,6 +13,6 @@ Description: Font configuration and customization library
 Version: @VERSION@
 Requires: @PKGCONFIG_REQUIRES@
 Requires.private: @PKGCONFIG_REQUIRES_PRIVATELY@
-Libs: -L${libdir} -lfontconfig
+Libs: -L${libdir} -lfontconfig -lxml2 -ldl
 Libs.private: @ICONV_LIBS@ @PKG_EXPAT_LIBS@
 Cflags: -I${includedir} @ICONV_CFLAGS@ @PKG_EXPAT_CFLAGS@
