--- fontconfig-2.13.94/config.sub.orig	2021-10-24 19:47:39.283922896 -0700
+++ fontconfig-2.13.94/config.sub	2021-10-24 19:47:56.679967977 -0700
@@ -1334,6 +1334,7 @@
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
 	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
