diff -Naur curl-7.65.3/lib/nonblock.c curl-7.65.3.serenity/lib/nonblock.c
--- curl-7.65.3/lib/nonblock.c	2021-04-12 18:25:48.687757722 +0200
+++ curl-7.65.3.serenity/lib/nonblock.c	2021-04-12 18:22:32.390375039 +0200
@@ -39,6 +39,10 @@
 
 #include "nonblock.h"
 
+#ifdef __serenity__
+#define USE_BLOCKING_SOCKETS
+#endif
+
 /*
  * curlx_nonblock() set the given socket to either blocking or non-blocking
  * mode based on the 'nonblock' boolean argument. This function is highly
