--- curl-7.81.0/configure.orig	2022-01-08 01:39:27.339369369 +0000
+++ curl-7.81.0/configure	2022-01-08 01:40:30.218717020 +0000
@@ -19836,50 +19836,8 @@
 rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
 
 
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if compiler halts on function prototype mismatch" >&5
-printf %s "checking if compiler halts on function prototype mismatch... " >&6; }
-  cat confdefs.h - <<_ACEOF >conftest.$ac_ext
-/* end confdefs.h.  */
 
 
-#     include <stdlib.h>
-      int rand(int n);
-      int rand(int n)
-      {
-        if(n)
-          return ++n;
-        else
-          return n;
-      }
-
-int main (void)
-{
-
-      int i[2]={0,0};
-      int j = rand(i[0]);
-      if(j)
-        return j;
-
- ;
- return 0;
-}
-
-_ACEOF
-if ac_fn_c_try_compile "$LINENO"
-then :
-
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: no" >&5
-printf "%s\n" "no" >&6; }
-    as_fn_error $? "compiler does not halt on function prototype mismatch." "$LINENO" 5
-
-else $as_nop
-
-    { printf "%s\n" "$as_me:${as_lineno-$LINENO}: result: yes" >&5
-printf "%s\n" "yes" >&6; }
-
-fi
-rm -f core conftest.err conftest.$ac_objext conftest.beam conftest.$ac_ext
-
 
       { printf "%s\n" "$as_me:${as_lineno-$LINENO}: checking if compiler supports hiding library internal symbols" >&5
 printf %s "checking if compiler supports hiding library internal symbols... " >&6; }
--- curl-7.81.0/configure.ac.orig	2022-01-08 01:42:45.505315507 +0000
+++ curl-7.81.0/configure.ac	2022-01-08 01:42:47.917290542 +0000
@@ -420,7 +420,6 @@
 
 CURL_CHECK_COMPILER_HALT_ON_ERROR
 CURL_CHECK_COMPILER_ARRAY_SIZE_NEGATIVE
-CURL_CHECK_COMPILER_PROTOTYPE_MISMATCH
 CURL_CHECK_COMPILER_SYMBOL_HIDING
 
 CURL_CHECK_CURLDEBUG
