From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Timothy <timmot@users.noreply.github.com>
Date: Thu, 3 Jun 2021 14:20:09 -0700
Subject: [PATCH] Enable the windows function stubs for serenity

---
 src/main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/main.c b/src/main.c
index 3e8fdde..3432fe8 100644
--- a/src/main.c
+++ b/src/main.c
@@ -14,7 +14,7 @@
 
 
 // Approximations of some Windows functions to ease portability
-#if defined __GNU_LIBRARY__ || defined __GLIBC__
+#if defined __GNU_LIBRARY__ || defined __GLIBC__ || defined __serenity__
 static int min(int l, int r) { return l < r ? l : r; }
 static void strcat_s(char * dest, int size, char * str) {
     unsigned int dlen = strlen(dest);
