From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Timothy <timmot@users.noreply.github.com>
Date: Mon, 16 May 2022 23:23:21 +0430
Subject: [PATCH] Manually set the SDL2 include path and library dir

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 1153e0c..da26e9d 100644
--- a/Makefile
+++ b/Makefile
@@ -3,8 +3,8 @@ OBJS = reciter.o sam.o render.o main.o debug.o processframes.o createtransitions
 CC = gcc
 
 # libsdl present
-CFLAGS =  -Wall -O2 -DUSESDL `sdl-config --cflags`
-LFLAGS = `sdl-config --libs`
+CFLAGS =  -Wall -O2 -DUSESDL -I$(SERENITY_INSTALL_ROOT)/usr/local/include/SDL2 -D_REENTRANT
+LFLAGS = -lSDL2
 
 # no libsdl present
 #CFLAGS =  -Wall -O2
