From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Thu, 12 Aug 2021 20:25:48 +0200
Subject: [PATCH] Let glib know where our 'resolv.h' is located

---
 gio/meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/gio/meson.build b/gio/meson.build
index bbedf47..b30cd7c 100644
--- a/gio/meson.build
+++ b/gio/meson.build
@@ -34,7 +34,7 @@ endif
 
 network_libs = [ ]
 network_args = [ ]
-if host_system != 'windows'
+if host_system not in ['windows', 'serenity']
   # res_query()
   res_query_test = '''#include <resolv.h>
                       int main (int argc, char ** argv) {
