From f228b35ce89f2ccf6d8c84a8a35c3e7279ddb424 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Thu, 12 Aug 2021 20:51:13 +0200
Subject: [PATCH 08/12] gio/gsocketaddress.c: Add missing macro
 IN6_IS_ADDR_V4MAPPED

---
 gio/gsocketaddress.c | 9 +++++++++
 1 file changed, 9 insertions(+)

diff --git a/gio/gsocketaddress.c b/gio/gsocketaddress.c
index 2b7e83c..88d757c 100644
--- a/gio/gsocketaddress.c
+++ b/gio/gsocketaddress.c
@@ -39,6 +39,15 @@
 #include "gunixsocketaddress.h"
 #endif
 
+#ifndef IN6_IS_ADDR_V4MAPPED
+#define IN6_IS_ADDR_V4MAPPED(a) \
+((((a)->s6_addr[0]) == 0) && \
+(((a)->s6_addr[1]) == 0) && \
+(((a)->s6_addr[2]) == 0) && \
+(((a)->s6_addr[3]) == 0) && \
+(((a)->s6_addr[4]) == 0) && \
+(((a)->s6_addr[5]) == 0xFFFF))
+#endif
 
 /**
  * SECTION:gsocketaddress
-- 
2.25.1

