From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Fri, 13 Aug 2021 22:32:25 +0200
Subject: [PATCH] Exclude arpa/nameser.h as it does not exist on Serenity

---
 gio/gnetworking.h.in | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/gio/gnetworking.h.in b/gio/gnetworking.h.in
index 2fa95ff..1d138cf 100644
--- a/gio/gnetworking.h.in
+++ b/gio/gnetworking.h.in
@@ -43,7 +43,9 @@
 #include <net/if.h>
 
 #include <arpa/inet.h>
+#ifndef __serenity__
 #include <arpa/nameser.h>
+#endif
 @NAMESER_COMPAT_INCLUDE@
 
 #ifndef T_SRV
