From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Mon, 3 Oct 2022 17:19:25 +0200
Subject: [PATCH] Do not flag support for extended attributes (xattr)

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index 6a7ea498e55d51f0fb7547160be7f87e95708039..40f9351ef98943871549911f69ecae7218743366 100644
--- a/meson.build
+++ b/meson.build
@@ -2164,7 +2164,7 @@ if host_system == 'linux'
 endif
 
 xattr_dep = []
-if host_system != 'windows' and get_option('xattr')
+if host_system != 'windows' and host_system != 'serenity' and get_option('xattr')
   # either glibc or libattr can provide xattr support
   # for both of them, we check for getxattr being in
   # the library and a valid xattr header.
