From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Mon, 3 Oct 2022 17:19:25 +0200
Subject: [PATCH] Do not flag support for extended attributes (xattr)

---
 meson.build | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/meson.build b/meson.build
index d8b1e0762c9450652548dd13b06c82380477a7d9..ac5dc864948e819318a56e5067ec89551c459578 100644
--- a/meson.build
+++ b/meson.build
@@ -2394,7 +2394,7 @@ if host_system == 'linux'
 endif
 
 xattr_dep = []
-if host_system != 'windows' and get_option('xattr')
+if host_system != 'windows' and host_system != 'serenity' and get_option('xattr')
   # either glibc or libattr can provide xattr support
   # for both of them, we check for getxattr being in
   # the library and a valid xattr header.
