From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Mon, 16 May 2022 18:03:31 +0430
Subject: [PATCH] Set host to serenity in configure

---
 configure | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/configure b/configure
index a2ff7e4..8283dc9 100755
--- a/configure
+++ b/configure
@@ -5274,6 +5274,8 @@ x86_64-*-cygwin)
 	else
 		icu_cv_host_frag=mh-cygwin-msvc
 	fi ;;
+*-serenity*)
+	icu_cv_host_frag=mh-serenity ;;
 *-*-mingw*)
 	if test "$GCC" = yes; then
                 cat confdefs.h - <<_ACEOF >conftest.$ac_ext
