From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nooga <xnooga@gmail.com>
Date: Sat, 15 May 2021 01:21:59 +0200
Subject: [PATCH] Install the game data to /usr/local

---
 config.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.mk b/config.mk
index b33967b..9655958 100644
--- a/config.mk
+++ b/config.mk
@@ -1,5 +1,5 @@
 # Where to look for game data files (found in 'bin'). Must be without trailing slashes!
-DATADIR := .
+DATADIR := /usr/local/share/games/brogue
 
 # Include terminal support. Requires ncurses
 TERMINAL := NO
