From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: nooga <xnooga@gmail.com>
Date: Sat, 15 May 2021 01:21:59 +0200
Subject: [PATCH] Use pkg-config for SDL2

---
 config.mk | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.mk b/config.mk
index 9655958..0ce2e55 100644
--- a/config.mk
+++ b/config.mk
@@ -7,7 +7,7 @@ TERMINAL := NO
 # Include graphical support. Requires SDL2 and SDL2_image
 GRAPHICS := YES
 # Path to sdl2-config script
-SDL_CONFIG := sdl2-config
+SDL_CONFIG := pkg-config sdl2
 
 # Select web brogue mode. Requires POSIX system.
 WEBBROGUE := NO
