--- bash-5.1.8/support/config.sub	2020-11-18 19:40:12.000000000 +0000
+++ bash-5.1.8.serenity/support/config.sub	2021-07-29 15:33:03.347306019 +0100
@@ -1691,6 +1691,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
