--- bash-5.1.8/locale.c	2020-02-24 20:08:43.000000000 +0000
+++ bash-5.1.8.serenity/locale.c	2021-07-29 15:51:14.523146438 +0100
@@ -76,6 +76,7 @@
 void
 set_default_locale ()
 {
+  return;
 #if defined (HAVE_SETLOCALE)
   default_locale = setlocale (LC_ALL, "");
   if (default_locale)
@@ -88,11 +89,11 @@
 
   locale_mb_cur_max = MB_CUR_MAX;
   locale_utf8locale = locale_isutf8 (default_locale);
-#if defined (HANDLE_MULTIBYTE)
-  locale_shiftstates = mblen ((char *)NULL, 0);
-#else
-  local_shiftstates = 0;
-#endif
+// #if defined (HANDLE_MULTIBYTE)
+//   locale_shiftstates = mblen ((char *)NULL, 0);
+// #else
+//   local_shiftstates = 0;
+// #endif
 }
 
 /* Set default values for LC_CTYPE, LC_COLLATE, LC_MESSAGES, LC_NUMERIC and
@@ -101,6 +102,7 @@
 void
 set_default_locale_vars ()
 {
+  return;
   char *val;
 
 #if defined (HAVE_SETLOCALE)
