From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Elliott <pelliott@ualberta.ca>
Date: Sat, 11 Feb 2023 13:22:20 -0700
Subject: [PATCH] Don't build wayland-server

---
 meson.build     |  4 ++--
 src/meson.build | 54 -------------------------------------------------
 2 files changed, 2 insertions(+), 56 deletions(-)

diff --git a/meson.build b/meson.build
index 8bb47c258228985667f1c6e0cc4ed64a02fa777a..52bf93bb278c75ee77bd3fcb29810752a1bc4ded 100644
--- a/meson.build
+++ b/meson.build
@@ -80,8 +80,8 @@ if get_option('libraries')
 	ffi_dep = dependency('libffi')
 
 	decls = [
-		{ 'header': 'sys/signalfd.h', 'symbol': 'SFD_CLOEXEC' },
-		{ 'header': 'sys/timerfd.h', 'symbol': 'TFD_CLOEXEC' },
+        #{ 'header': 'sys/signalfd.h', 'symbol': 'SFD_CLOEXEC' },
+        #{ 'header': 'sys/timerfd.h', 'symbol': 'TFD_CLOEXEC' },
 		{ 'header': 'time.h', 'symbol': 'CLOCK_MONOTONIC' },
 	]
 
diff --git a/src/meson.build b/src/meson.build
index a8a1d2ba979411a09fd754ef9b8ddf44a6f74653..c20d9b9df042004bcbe0e1ae466c409bd51d3f82 100644
--- a/src/meson.build
+++ b/src/meson.build
@@ -167,58 +167,6 @@ if get_option('libraries')
 		error('We probably need to bump the SONAME of libwayland-server and -client')
 	endif
 
-	wayland_server = library(
-		'wayland-server',
-		sources: [
-			wayland_server_protocol_core_h,
-			wayland_server_protocol_h,
-			wayland_protocol_c,
-			'wayland-server.c',
-			'wayland-shm.c',
-			'event-loop.c'
-		],
-		# To avoid an unnecessary SONAME bump, wayland 1.x.y produces
-		# libwayland-server.so.0.x.y.
-		version: '.'.join(['0', wayland_version[1], wayland_version[2]]),
-		dependencies: [
-			epoll_dep,
-			ffi_dep,
-			wayland_private_dep,
-			wayland_util_dep,
-			mathlib_dep,
-			threads_dep,
-			rt_dep
-		],
-		include_directories: root_inc,
-		install: true
-	)
-
-	wayland_server_dep = declare_dependency(
-		link_with: wayland_server,
-		include_directories: [ root_inc, include_directories('.') ],
-		dependencies: [ epoll_dep, ffi_dep, mathlib_dep, threads_dep ],
-		sources: [
-			wayland_server_protocol_core_h,
-			wayland_server_protocol_h
-		]
-	)
-
-	pkgconfig.generate(
-		wayland_server,
-		name: 'Wayland Server',
-		description: 'Server side implementation of the Wayland protocol',
-		version: meson.project_version(),
-		filebase: 'wayland-server',
-		variables: [
-			'datarootdir=' + join_paths('${prefix}', get_option('datadir')),
-			'pkgdatadir=' + join_paths('${datarootdir}', meson.project_name())
-		]
-	)
-
-	if meson.version().version_compare('>= 0.54.0')
-		meson.override_dependency('wayland-server', wayland_server_dep)
-	endif
-
 	wayland_client = library(
 		'wayland-client',
 		sources: [
@@ -269,8 +217,6 @@ if get_option('libraries')
 
 	install_headers([
 		'wayland-util.h',
-		'wayland-server.h',
-		'wayland-server-core.h',
 		'wayland-client.h',
 		'wayland-client-core.h',
 	])
