From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Elliott <pelliott@ualberta.ca>
Date: Sat, 11 Feb 2023 13:52:12 -0700
Subject: [PATCH] Remove unsupported OS features

---
 src/wayland-os.c | 18 ------------------
 src/wayland-os.h |  3 ---
 2 files changed, 21 deletions(-)

diff --git a/src/wayland-os.c b/src/wayland-os.c
index a9066cae9c93b607ea92ad1da1746f3b37a2bcb3..4872c4837c67178f46e3ab05d9006a8c512664cc 100644
--- a/src/wayland-os.c
+++ b/src/wayland-os.c
@@ -33,7 +33,6 @@
 #include <fcntl.h>
 #include <errno.h>
 #include <string.h>
-#include <sys/epoll.h>
 #include <sys/mman.h>
 #include <sys/un.h>
 #ifdef HAVE_SYS_UCRED_H
@@ -188,23 +187,6 @@ wl_os_recvmsg_cloexec(int sockfd, struct msghdr *msg, int flags)
 	return recvmsg_cloexec_fallback(sockfd, msg, flags);
 }
 
-int
-wl_os_epoll_create_cloexec(void)
-{
-	int fd;
-
-#ifdef EPOLL_CLOEXEC
-	fd = epoll_create1(EPOLL_CLOEXEC);
-	if (fd >= 0)
-		return fd;
-	if (errno != EINVAL)
-		return -1;
-#endif
-
-	fd = epoll_create(1);
-	return set_cloexec_or_close(fd);
-}
-
 int
 wl_os_accept_cloexec(int sockfd, struct sockaddr *addr, socklen_t *addrlen)
 {
diff --git a/src/wayland-os.h b/src/wayland-os.h
index 068fd2fea3da8e0a13a3ab993fcc7d4c609a366d..ca4de54161b39e612b14332364cfb697e53d9afc 100644
--- a/src/wayland-os.h
+++ b/src/wayland-os.h
@@ -41,9 +41,6 @@ wl_os_dupfd_cloexec(int fd, int minfd);
 ssize_t
 wl_os_recvmsg_cloexec(int sockfd, struct msghdr *msg, int flags);
 
-int
-wl_os_epoll_create_cloexec(void);
-
 int
 wl_os_accept_cloexec(int sockfd, struct sockaddr *addr, socklen_t *addrlen);
 
