From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Peter Elliott <pelliott@ualberta.ca>
Date: Sat, 11 Feb 2023 15:17:27 -0700
Subject: [PATCH] Remove call to open_memstream(3) from wl_closure_print

---
 src/connection.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/connection.c b/src/connection.c
index 6ab20ab01c6506513f7c8fa8bcb932a5b9027b10..6199c9890790e44073128e6bb1993a61d68d4d26 100644
--- a/src/connection.c
+++ b/src/connection.c
@@ -1549,12 +1549,7 @@ wl_closure_print(struct wl_closure *closure, struct wl_object *target,
 	struct timespec tp;
 	uint32_t nval;
 	FILE *f;
-	char *buffer;
-	size_t buffer_length;
-
-	f = open_memstream(&buffer, &buffer_length);
-	if (f == NULL)
-		return;
+	f = stderr;
 
 	clock_gettime(CLOCK_REALTIME, &tp);
 	fprintf(f, "%s[%02u:%02u:%02u.%06u] ",
@@ -1659,11 +1654,6 @@ wl_closure_print(struct wl_closure *closure, struct wl_object *target,
 	}
 
 	fprintf(f, ")%s\n", color ? WL_DEBUG_COLOR_RESET : "");
-
-	if (fclose(f) == 0) {
-		fprintf(stderr, "%s", buffer);
-		free(buffer);
-	}
 }
 
 static int
