From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Taj Morton <tajmorton@gmail.com>
Date: Sun, 9 Jul 2023 09:55:42 -0700
Subject: [PATCH] Update pith/Makefile.in to compile helpers for build system,
 not target

---
 pith/Makefile.in | 12 ++++++------
 1 file changed, 6 insertions(+), 6 deletions(-)

diff --git a/pith/Makefile.in b/pith/Makefile.in
index e97b5fa1077708ebea467f3de09b12f7a9494b6c..8bc867d528aca1ee485d30eadfb94e60884b7318 100644
--- a/pith/Makefile.in
+++ b/pith/Makefile.in
@@ -897,12 +897,12 @@ uninstall-am:
 .PRECIOUS: Makefile
 
 
-help_c_gen$(EXEEXT): $(help_c_gen_OBJECTS) $(help_c_gen_DEPENDENCIES)
-	@rm -f help_c_gen$(EXEEXT)
-	$(LINK) $(help_c_gen_OBJECTS) $(help_c_gen_LDADD)
-help_h_gen$(EXEEXT): $(help_h_gen_OBJECTS) $(help_h_gen_DEPENDENCIES)
-	@rm -f help_h_gen$(EXEEXT)
-	$(LINK) $(help_h_gen_OBJECTS) $(help_h_gen_LDADD)
+help_c_gen$(BUILD_EXEEXT): help_c_gen.c
+	@rm -f help_c_gen$(BUILD_EXEEXT)
+	$(CC_FOR_BUILD) help_c_gen.c -o help_c_gen$(BUILD_EXEEXT)
+help_h_gen$(BUILD_EXEEXT):
+	@rm -f help_h_gen$(BUILD_EXEEXT)
+	$(CC_FOR_BUILD) help_h_gen.c -o help_h_gen$(BUILD_EXEEXT)
 
 helptext.c: help_c_gen pine.hlp
 	./help_c_gen < pine.hlp > $@
