--- Python-3.10.1/Include/pyport.h	2021-08-03 18:40:05.313000000 +0100
+++ Python-3.10.1/Include/pyport.h	2021-08-03 18:40:28.722351782 +0100
@@ -843,7 +843,7 @@
 #  error "Py_TRACE_REFS ABI is not compatible with release and debug ABI"
 #endif
 
-#if defined(__ANDROID__) || defined(__VXWORKS__)
+#if defined(__ANDROID__) || defined(__VXWORKS__) || defined(__serenity__)
    // Use UTF-8 as the locale encoding, ignore the LC_CTYPE locale.
    // See _Py_GetLocaleEncoding(), PyUnicode_DecodeLocale()
    // and PyUnicode_EncodeLocale().
