--- Python-3.10.1/Modules/posixmodule.h	2021-08-03 19:01:36.368000000 +0100
+++ Python-3.10.1/Modules/posixmodule.h	2021-08-03 19:01:54.120205211 +0100
@@ -19,7 +19,8 @@
 #endif /* MS_WINDOWS */
 
 #if defined(PYPTHREAD_SIGMASK) || defined(HAVE_SIGWAIT) || \
-        defined(HAVE_SIGWAITINFO) || defined(HAVE_SIGTIMEDWAIT)
+        defined(HAVE_SIGWAITINFO) || defined(HAVE_SIGTIMEDWAIT) || \
+        defined(__serenity__)
 # define HAVE_SIGSET_T
 #endif
 
