--- Python-3.9.4/Programs/python.c	2021-02-21 20:22:43.960024172 +0100
+++ Python-3.9.4/Programs/python.c	2021-02-21 20:35:13.268988119 +0100
@@ -9,7 +9,7 @@
     return Py_Main(argc, argv);
 }
 #else
-int
+Py_EXPORTED_SYMBOL int
 main(int argc, char **argv)
 {
     return Py_BytesMain(argc, argv);
--- Python-3.9.4/Programs/_testembed.c	2021-02-21 20:22:43.959024178 +0100
+++ Python-3.9.4/Programs/_testembed.c	2021-02-21 20:35:13.270988109 +0100
@@ -1711,7 +1711,7 @@
     {NULL, NULL}
 };
 
-int main(int argc, char *argv[])
+Py_EXPORTED_SYMBOL int main(int argc, char *argv[])
 {
     if (argc > 1) {
         for (struct TestCase *tc = TestCases; tc && tc->name; tc++) {
