diff -ur a/Modules/signalmodule.c b/Modules/signalmodule.c
--- a/Modules/signalmodule.c	2021-12-06 19:23:39.000000000 +0100
+++ b/Modules/signalmodule.c	2021-12-22 03:37:56.097759703 +0100
@@ -1151,7 +1151,7 @@
     PyStructSequence_SET_ITEM(result, 4, PyLong_FromLong(0L));
     PyStructSequence_SET_ITEM(result, 5, PyLong_FromLong(0L));
 #else
-    PyStructSequence_SET_ITEM(result, 2, PyLong_FromLong((long)(si->si_errno)));
+    PyStructSequence_SET_ITEM(result, 2, PyLong_FromLong(0L));
     PyStructSequence_SET_ITEM(result, 3, PyLong_FromPid(si->si_pid));
     PyStructSequence_SET_ITEM(result, 4, _PyLong_FromUid(si->si_uid));
     PyStructSequence_SET_ITEM(result, 5,
