--- Python-3.9.2/Python/getversion.c	2021-02-21 20:22:44.092438528 +0100
+++ Python-3.9.2/Python/getversion.c	2021-02-21 20:36:51.249477963 +0100
@@ -9,7 +9,7 @@
 Py_GetVersion(void)
 {
     static char version[250];
-    PyOS_snprintf(version, sizeof(version), "%.80s (%.80s) %.80s",
+    PyOS_snprintf(version, sizeof(version), "%s (%s) %s",
                   PY_VERSION, Py_GetBuildInfo(), Py_GetCompiler());
     return version;
 }
--- Python-3.9.2/Modules/getbuildinfo.c	2021-02-21 20:22:43.945286288 +0100
+++ Python-3.9.2/Modules/getbuildinfo.c	2021-02-21 20:38:09.187987432 +0100
@@ -43,7 +43,7 @@
     if (!(*gitid))
         gitid = "default";
     PyOS_snprintf(buildinfo, sizeof(buildinfo),
-                  "%s%s%s, %.20s, %.9s", gitid, sep, revision,
+                  "%s%s%s, %s, %s", gitid, sep, revision,
                   DATE, TIME);
     return buildinfo;
 }
