--- Python-3.9.6/Lib/webbrowser.py	2021-05-03 15:54:42.000000000 +0100
+++ Python-3.9.6/Lib/webbrowser.py	2021-05-06 14:21:34.242964786 +0100
@@ -574,6 +574,9 @@
             if shutil.which("w3m"):
                 register("w3m", None, GenericBrowser("w3m"))

+    # SerenityOS Browser
+    register("Browser", None, BackgroundBrowser("Browser"))
+
     # OK, now that we know what the default preference orders for each
     # platform are, allow user to override them with the BROWSER variable.
     if "BROWSER" in os.environ:
