From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Mon, 24 Apr 2023 17:24:51 +0200
Subject: [PATCH] Engine: Add missing include for <sys/select.h>

---
 SamTFE/Sources/Engine/GameAgent/GameAgent.cpp | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SamTFE/Sources/Engine/GameAgent/GameAgent.cpp b/SamTFE/Sources/Engine/GameAgent/GameAgent.cpp
index fc6c8441188f19c1cd35377643f86b9df5a7cbb6..1c85633a344181f65e2466950ea2313b7a34c6ec 100644
--- a/SamTFE/Sources/Engine/GameAgent/GameAgent.cpp
+++ b/SamTFE/Sources/Engine/GameAgent/GameAgent.cpp
@@ -35,6 +35,7 @@ typedef int socklen_t;
 #include <fcntl.h>
 #include <netdb.h>
 #include <pthread.h>
+#include <sys/select.h>
 #include <sys/socket.h>
 #include <netinet/in.h>
 #include <arpa/inet.h>
