From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: cflip <cflip@cflip.net>
Date: Fri, 20 Oct 2023 10:46:47 -0600
Subject: [PATCH] Ignore warning with using SDL backend

---
 src/Window_SDL.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/src/Window_SDL.c b/src/Window_SDL.c
index f1ff58f8550f3546ad14ec2a7311b60a4bfd3faa..6a56bc687cf1fe7bf1f9c42018593ea5fd84e359 100644
--- a/src/Window_SDL.c
+++ b/src/Window_SDL.c
@@ -9,8 +9,6 @@
 #include <SDL2/SDL.h>
 static SDL_Window* win_handle;
 
-#error "Some features are missing from the SDL backend. If possible, it is recommended that you use a native windowing backend instead"
-
 static void RefreshWindowBounds(void) {
 	SDL_GetWindowSize(win_handle, &WindowInfo.Width, &WindowInfo.Height);
 }
