From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Hunter Kvalevog <hunterkva@protonmail.com>
Date: Sat, 2 Apr 2022 19:14:23 -0500
Subject: [PATCH] Add definitions for serenity to configure

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index e242e73..1f60a1a 100755
--- a/configure
+++ b/configure
@@ -729,6 +729,10 @@ case $host_os in
     *haiku*)
         SYS="HAIKU"
         ;;
+    serenity)
+        SYS="SERENITY"
+        libm="-lm"
+        ;;
     *)
         die "Unknown system $host, edit the configure"
         ;;
