diff --git a/config.sub b/config.sub
index 72e9265b..2ee38778 100755
--- a/config.sub
+++ b/config.sub
@@ -1514,6 +1514,8 @@ case $os in
 		;;
 	-nacl*)
 		;;
+    -serenity)
+        ;;
 	-none)
 		;;
 	*)
diff --git a/configure b/configure
index e242e73c..1f60a1ab 100755
--- a/configure
+++ b/configure
@@ -729,6 +729,10 @@ case $host_os in
     *haiku*)
         SYS="HAIKU"
         ;;
+    serenity)
+        SYS="SERENITY"
+        libm="-lm"
+        ;;
     *)
         die "Unknown system $host, edit the configure"
         ;;
