From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Tue, 12 Apr 2022 18:01:44 +0200
Subject: [PATCH] Workaround for redundant redeclaration of 'environ'

---
 main.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/main.c b/main.c
index 0f433d2..577afde 100644
--- a/main.c
+++ b/main.c
@@ -14,7 +14,9 @@
 #include <unistd.h>
 #include "frontend.h"
 
+#ifndef __serenity__
 extern char **environ;
+#endif
 
 int main(int argc, char *argv[]) {
 	struct mrsh_state *state = mrsh_state_create();
