From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Wed, 4 May 2022 16:50:45 +0200
Subject: [PATCH] `glob.h` is missing so we disable glob

---
 shell/word.c | 6 ++++++
 1 file changed, 6 insertions(+)

diff --git a/shell/word.c b/shell/word.c
index df274b7..dd94965 100644
--- a/shell/word.c
+++ b/shell/word.c
@@ -1,7 +1,9 @@
 #define _POSIX_C_SOURCE 200809L
 #include <assert.h>
 #include <ctype.h>
+#ifndef __serenity__
 #include <glob.h>
+#endif
 #include <mrsh/buffer.h>
 #include <pwd.h>
 #include <stdbool.h>
@@ -339,10 +341,13 @@ bool expand_pathnames(struct mrsh_array *expanded,
 	for (size_t i = 0; i < fields->len; ++i) {
 		const struct mrsh_word *field = fields->data[i];
 
+#ifndef __serenity__
 		char *pattern = word_to_pattern(field);
 		if (pattern == NULL) {
+#endif
 			mrsh_array_add(expanded, mrsh_word_str(field));
 			continue;
+#ifndef __serenity__
 		}
 
 		glob_t glob_buf;
@@ -361,6 +366,7 @@ bool expand_pathnames(struct mrsh_array *expanded,
 		}
 
 		free(pattern);
+#endif
 	}
 
 	return true;
