diff -Naur emersion-mrsh-d9763a3/shell/redir.c emersion-mrsh-d9763a3.serenity/shell/redir.c
--- emersion-mrsh-d9763a3/shell/redir.c	2020-01-27 16:43:09.000000000 +0100
+++ emersion-mrsh-d9763a3.serenity/shell/redir.c	2021-04-12 10:17:08.980846918 +0200
@@ -46,7 +46,7 @@
 
 	// We can write at most PIPE_BUF bytes without blocking. If we want to write
 	// more, we need to fork and continue writing in another process.
-	size_t remaining = PIPE_BUF;
+	size_t remaining = 4096;
 	bool more = false;
 	size_t i;
 	for (i = 0; i < lines->len; ++i) {
