From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Thu, 14 Jul 2022 20:47:33 +0200
Subject: [PATCH] engine: whereami: Add SerenityOS

---
 engine/common/whereami.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/engine/common/whereami.c b/engine/common/whereami.c
index 79b01dc..02ded98 100644
--- a/engine/common/whereami.c
+++ b/engine/common/whereami.c
@@ -169,7 +169,7 @@ int WAI_PREFIX(getModulePath)(char* out, int capacity, int* dirname_length)
   return length;
 }
 
-#elif defined(__linux__) || defined(__CYGWIN__) || defined(__sun) || defined(WAI_USE_PROC_SELF_EXE)
+#elif defined(__linux__) || defined(__CYGWIN__) || defined(__sun) || defined(__serenity__) || defined(WAI_USE_PROC_SELF_EXE)
 
 #include <stdio.h>
 #include <stdlib.h>
