From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Thu, 14 Apr 2022 00:22:19 +0200
Subject: [PATCH] Include sys/select.h in src/SelectEventPoll.cc

---
 src/SelectEventPoll.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/SelectEventPoll.cc b/src/SelectEventPoll.cc
index d9397b2..49b1f03 100644
--- a/src/SelectEventPoll.cc
+++ b/src/SelectEventPoll.cc
@@ -40,6 +40,7 @@
 #include <cstring>
 #include <algorithm>
 #include <numeric>
+#include <sys/select.h>
 
 #include "Command.h"
 #include "LogFactory.h"
