From b60680828e48ca362e86913de773a37b7a06bc74 Mon Sep 17 00:00:00 2001
From: Tobias Christiansen <tobyase@serenityos.org>
Date: Fri, 11 Mar 2022 19:01:35 +0100
Subject: [PATCH 4/4] Teach config.sub about serenity

---
 config.sub | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index eb0389a..bde853e 100755
--- a/config.sub
+++ b/config.sub
@@ -1290,7 +1290,7 @@ case $os in
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -serenity* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
-- 
2.36.1

