From a83779c90a4a76df21f81d70874997e5a318a95a Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:01:11 +0330
Subject: [PATCH 01/18] Remove -m(tune arch)=native

---
 makefile | 5 ++---
 1 file changed, 2 insertions(+), 3 deletions(-)

diff --git a/makefile b/makefile
index 5bcbf50..681859d 100644
--- a/makefile
+++ b/makefile
@@ -16,9 +16,8 @@ enable_inject ?= true
 enable_ctypes ?= true
 enable_inline_asm ?= false
 
-CFLAGS += -Wall -Wextra -Wno-unused-parameter -mtune=native\
-		  -march=native -D withTermios -D forLinux\
-		  -D CTR_STD_EXTENSION_PATH=\"$(DATADIR)\"
+CFLAGS += -Wall -Wextra -Wno-unused-parameter\
+		  -D withTermios -D CTR_STD_EXTENSION_PATH=\"$(DATADIR)\"
 
 ifeq ($(enable_inject),true)
 	CFLAGS += -DwithInjectNative=1
-- 
2.34.1

