From 5198e9f4d722e94cbfa08e2833837b593857b63a Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:17:46 +0330
Subject: [PATCH 04/18] Use <fcntl.h> on serenity

---
 src/system.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/src/system.c b/src/system.c
index 3b395bd..a117498 100644
--- a/src/system.c
+++ b/src/system.c
@@ -10,7 +10,11 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#ifdef __serenity__
+#include <fcntl.h>
+#else
 #include <sys/fcntl.h>
+#endif
 #include <sys/ioctl.h>
 #include <sys/socket.h>
 #include <sys/wait.h>
-- 
2.34.1

