From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:46:16 +0330
Subject: [PATCH] Don't mess with libsocket

---
 makefile | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/makefile b/makefile
index 1e87eaea439902476d80fb73cb7fea9a2667d10a..e11b39f04dcf14161927f406c31568d34145b9bf 100644
--- a/makefile
+++ b/makefile
@@ -2,7 +2,6 @@ DEBUG_VERSION := 1603
 DEBUG_BUILD_VERSION := "\"$(DEBUG_VERSION)\""
 location = $(CURDIR)/$(word $(words $(MAKEFILE_LIST)),$(MAKEFILE_LIST))
 new_makefile_l1 := $(shell perl -ne '/((DEBUG_VERSION := )(\d+))/ && print (sprintf("%s%s", "$$2", "$$3"+1));' $(location))
-LIBSOCKETDIR = src/lib/libsocket
 BUILDDIR ?= build
 DESTDIR ?= /
 BINDIR ?= $(DESTDIR)/usr/local/bin
@@ -62,7 +61,7 @@ LEXTRACF += $(LDFLAGS) -flto -lstdc++
 OBJS = siphash.o utf8.o memory.o util.o base.o collections.o file.o system.o \
 	   lexer.o lexer_plug.o parser.o walker.o marshal.o reflect.o fiber.o \
 	   importlib.o coroutine.o symbol.o generator.o base_extensions.o citron.o \
-	   promise.o symbol_cxx.o world.o libsocket.so
+	   promise.o symbol_cxx.o world.o
 EXTRAOBJS =
 TCC_STATICS = 
 
@@ -137,7 +136,6 @@ debug: cxx ctr
 
 clean:
 	rm -rf $(BUILDDIR)
-	$(MAKE) -C $(LIBSOCKETDIR) clean
 	$(MAKE) -C src/lib/tcc clean
 
 cxx:
@@ -152,10 +150,6 @@ build_tcc_statics:
 $(BUILDDIR)/%.a: src/lib/tcc/%.a
 	cp $< $@
 
-$(BUILDDIR)/libsocket.so:
-	make -C $(LIBSOCKETDIR)
-	cp $(LIBSOCKETDIR)/libsocket.so $(BUILDDIR)/libsocket.so
-
 $(BUILDDIR)/%.o: src/%.c
 	$(CC) $(CFLAGS) -c $< -o $@
 
