From 669cb84c24184df7912ea5065d731f12191b5e67 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:27:38 +0330
Subject: [PATCH 05/18] Pull arc4random from stdlib.h

---
 src/system.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/system.c b/src/system.c
index a117498..e42d09a 100644
--- a/src/system.c
+++ b/src/system.c
@@ -35,6 +35,8 @@ static struct termios oldTermios, newTermios;
 #ifdef forLinux
 #    include <bsd/stdlib.h>
 
+#elif defined(__serenity__)
+// Nothing, they're in stdlib.h.
 #else
 // Arc4random
 
-- 
2.34.1

