From 345758e53e21b23c716c2efccdaccf5434f16f0e Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:49:59 +0330
Subject: [PATCH 5/9] Disable inject

tcc requires ucontext.
---
 makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/makefile b/makefile
index b262cd0..2e05795 100644
--- a/makefile
+++ b/makefile
@@ -12,7 +12,7 @@ INCLUDE_DIR ?= $(DESTDIR)/usr/local/include/Citron
 
 .PHONY: all ctrconfig libctr ctr install cxx
 
-enable_inject ?= true
+enable_inject ?= false
 enable_ctypes ?= true
 enable_inline_asm ?= false
 enable_boehm_gc ?= true
-- 
2.34.1

