From e4b172e0a9db3eaa0c9c14c477e26185a2a596de Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 11 Feb 2022 16:52:51 +0330
Subject: [PATCH 6/9] Disable openmp

---
 makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/makefile b/makefile
index 2e05795..d916511 100644
--- a/makefile
+++ b/makefile
@@ -96,7 +96,7 @@ $(BUILDDIR):
 	mkdir -p $@
 
 $(BUILDDIR)/ctr: $(TCC_STATICS) $(BUILDDIR) $(OBJS) $(EXTRAOBJS)
-	$(CXX) -fopenmp $(EXTRAOBJS) $(OBJS) $(CXXFLAGS) $(CFLAGS) $(LIBS) $(LEXTRACF) -o $@
+	$(CXX) $(EXTRAOBJS) $(OBJS) $(CXXFLAGS) $(CFLAGS) $(LIBS) $(LEXTRACF) -o $@
 
 ctr: $(BUILDDIR)/ctr
 
@@ -145,7 +145,7 @@ $(BUILDDIR)/%.a: src/lib/tcc/%.a
 	cp $< $@
 
 $(BUILDDIR)/%.o: src/%.c
-	$(CC) -fopenmp $(CFLAGS) -c $< -o $@
+	$(CC) $(CFLAGS) -c $< -o $@
 
 $(BUILDDIR)/%.o: src/%.cpp
 	$(CXX) -g $(CFLAGS) -c $< $(CXXFLAGS) -o $@
-- 
2.34.1

