--- gmp-6.2.1/configfsf.sub	2021-04-07 22:50:17.785886997 -0700
+++ gmp-6.2.1/configfsf.sub.new	2021-04-07 22:50:15.041883314 -0700
@@ -1691,6 +1691,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	gnu* | android* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]* \
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* |  plan9* | psp* | sim* | xray* | os68k* | v88r* \
