--- mpc-1.2.1/build-aux/config.sub	2020-10-20 17:26:31.000000000 +0100
+++ mpc-1.2.1.serenity/build-aux/config.sub	2021-07-28 18:13:47.269493423 +0100
@@ -1339,6 +1339,7 @@
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
 	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
