diff -u -r mbedtls/library/ssl_tls.c mbedtls-patched/library/ssl_tls.c
--- mbedtls/library/ssl_tls.c	2019-06-18 23:23:28.000000000 +1000
+++ mbedtls-patched/library/ssl_tls.c	2019-08-05 00:36:37.049255960 +1000
@@ -49,6 +49,7 @@
 #include "mbedtls/platform_util.h"
 
 #include <string.h>
+#include <stdint.h>
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
 #include "mbedtls/oid.h"
diff -u -r mbedtls/library/x509_crt.c mbedtls-patched/library/x509_crt.c
--- mbedtls/library/x509_crt.c	2019-06-18 23:23:28.000000000 +1000
+++ mbedtls-patched/library/x509_crt.c	2019-08-05 00:24:22.606023827 +1000
@@ -75,6 +75,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <dirent.h>
+#include <unistd.h>
 #endif /* !_WIN32 || EFIX64 || EFI32 */
 #endif
 
