From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "jake.westrip" <jake.westrip@lonsec.com.au>
Date: Tue, 27 Apr 2021 18:34:53 +1000
Subject: [PATCH] Include strings.h

---
 src/ppui/BasicTypes.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/ppui/BasicTypes.h b/src/ppui/BasicTypes.h
index 20eba8b..e353345 100644
--- a/src/ppui/BasicTypes.h
+++ b/src/ppui/BasicTypes.h
@@ -30,6 +30,7 @@ typedef signed short	pp_int16;
 typedef unsigned int	pp_uint32;
 typedef signed int		pp_int32;
 
+#include <strings.h>
 #include "ScanCodes.h"
 
 #if defined(WIN32) || defined(_WIN32_WCE) 
