From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: "jake.westrip" <jake.westrip@lonsec.com.au>
Date: Wed, 28 Apr 2021 21:19:03 +1000
Subject: [PATCH] Link against the needed serenity libraries

---
 src/tracker/CMakeLists.txt | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/tracker/CMakeLists.txt b/src/tracker/CMakeLists.txt
index f4243a0..02f67c0 100644
--- a/src/tracker/CMakeLists.txt
+++ b/src/tracker/CMakeLists.txt
@@ -374,4 +374,7 @@ else()
     set(INSTALL_DEST ${CMAKE_INSTALL_BINDIR})
 endif()
 
+target_link_libraries(tracker -lSDL2 -lgui -lipc -lm)
+set(INSTALL_DEST bin)
+
 install(TARGETS tracker DESTINATION ${INSTALL_DEST})
