From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jaan <jaan_veersoo@hotmail.com>
Date: Sat, 20 Mar 2021 23:34:53 +0200
Subject: [PATCH] Add a missing sys/time.h include

---
 src/input/select.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/input/select.h b/src/input/select.h
index 093bb77..48aad37 100644
--- a/src/input/select.h
+++ b/src/input/select.h
@@ -1,4 +1,5 @@
 #include "../config.h"
+#include <sys/time.h>
 #if HAVE_SYS_SELECT_H
 #include <sys/select.h>
 #elif __DJGPP__
