diff -Naur ntbtls-0.2.0/build-aux/config.sub ntbtls-0.2.0.serenity/build-aux/config.sub
--- ntbtls-0.2.0/build-aux/config.sub	2016-07-13 19:10:49.000000000 +0200
+++ ntbtls-0.2.0.serenity/build-aux/config.sub	2021-04-14 02:52:34.178672014 +0200
@@ -1381,7 +1381,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
