diff -Naur expat-2.2.9/examples/elements.c expat-2.2.9.serenity/examples/elements.c
--- expat-2.2.9/examples/elements.c	2019-09-21 17:23:24.000000000 +0200
+++ expat-2.2.9.serenity/examples/elements.c	2021-04-11 03:55:14.027401389 +0200
@@ -70,6 +70,7 @@
   *depthPtr -= 1;
 }
 
+__attribute__((visibility("default")))
 int
 main(int argc, char *argv[]) {
   char buf[BUFSIZ];
diff -Naur expat-2.2.9/examples/outline.c expat-2.2.9.serenity/examples/outline.c
--- expat-2.2.9/examples/outline.c	2019-09-21 17:23:24.000000000 +0200
+++ expat-2.2.9.serenity/examples/outline.c	2021-04-11 03:55:51.864760235 +0200
@@ -79,6 +79,7 @@
   Depth--;
 }
 
+__attribute__((visibility("default")))
 int
 main(int argc, char *argv[]) {
   XML_Parser p = XML_ParserCreate(NULL);
diff -Naur expat-2.2.9/tests/benchmark/benchmark.c expat-2.2.9.serenity/tests/benchmark/benchmark.c
--- expat-2.2.9/tests/benchmark/benchmark.c	2019-09-21 17:23:24.000000000 +0200
+++ expat-2.2.9.serenity/tests/benchmark/benchmark.c	2021-04-11 03:57:00.892671995 +0200
@@ -54,6 +54,7 @@
   exit(rc);
 }
 
+__attribute__((visibility("default")))
 int
 main(int argc, char *argv[]) {
   XML_Parser parser;
diff -Naur expat-2.2.9/tests/runtests.c expat-2.2.9.serenity/tests/runtests.c
--- expat-2.2.9/tests/runtests.c	2019-09-26 00:03:02.000000000 +0200
+++ expat-2.2.9.serenity/tests/runtests.c	2021-04-11 03:56:30.438792094 +0200
@@ -11566,6 +11566,7 @@
   return s;
 }
 
+__attribute__((visibility("default")))
 int
 main(int argc, char *argv[]) {
   int i, nf;
diff -Naur expat-2.2.9/xmlwf/ct.c expat-2.2.9.serenity/xmlwf/ct.c
--- expat-2.2.9/xmlwf/ct.c	2019-09-21 17:23:24.000000000 +0200
+++ expat-2.2.9.serenity/xmlwf/ct.c	2021-04-11 03:57:26.136455396 +0200
@@ -164,6 +164,7 @@
   }
 }
 
+__attribute__((visibility("default")))
 int
 main(int argc, char **argv) {
   char buf[CHARSET_MAX];
diff -Naur expat-2.2.9/xmlwf/xmlwf.c expat-2.2.9.serenity/xmlwf/xmlwf.c
--- expat-2.2.9/xmlwf/xmlwf.c	2019-09-21 17:23:24.000000000 +0200
+++ expat-2.2.9.serenity/xmlwf/xmlwf.c	2021-04-11 03:59:32.375335879 +0200
@@ -891,6 +891,7 @@
 int wmain(int argc, XML_Char **argv);
 #endif
 
+__attribute__((visibility("default")))
 int
 tmain(int argc, XML_Char **argv) {
   int i, j;
