From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Mattias Nilsson <mattias.nilsson@nshift.com>
Date: Mon, 4 Apr 2022 22:05:38 +0200
Subject: [PATCH] Make the version descriptor match the built tag

---
 main.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/main.c b/main.c
index 986f1a3..e22f097 100644
--- a/main.c
+++ b/main.c
@@ -22,7 +22,7 @@ ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
 THIS SOFTWARE.
 ****************************************************************/
 
-const char	*version = "version 20211208";
+const char	*version = "version 20220122";
 
 #define DEBUG
 #include <stdio.h>
