From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brendan Coles <bcoles@gmail.com>
Date: Mon, 9 Nov 2020 10:52:11 +0000
Subject: [PATCH] Don't build docs

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index cdf7999..ce49203 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -237,7 +237,7 @@ top_build_prefix = @top_build_prefix@
 top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AUTOMAKE_OPTIONS = no-texinfo.tex
-SUBDIRS = intl src doc po man
+SUBDIRS = intl src po
 BUILT_SOURCES = 
 EXTRA_DIST = README \
 		config/config.rpath config/mkinstalldirs \
