diff -u b/dmidecode.c b/dmidecode.c
--- b/dmidecode.c
+++ b/dmidecode.c
@@ -90,7 +90,7 @@
 #define FLAG_NO_FILE_OFFSET     (1 << 0)
 #define FLAG_STOP_AT_EOT        (1 << 1)
 
-#define SYS_FIRMWARE_DIR "/sys/firmware/dmi/tables"
+#define SYS_FIRMWARE_DIR "/sys/bios"
 #define SYS_ENTRY_FILE SYS_FIRMWARE_DIR "/smbios_entry_point"
 #define SYS_TABLE_FILE SYS_FIRMWARE_DIR "/DMI"
 
@@ -3654,7 +3654,5 @@
 {
 	if (addrtype == 0x1) /* IPv4 */
 		return inet_ntop(AF_INET, data, storage, 64);
-	if (addrtype == 0x2) /* IPv6 */
-		return inet_ntop(AF_INET6, data, storage, 64);
 	return out_of_spec;
 }
