diff --git a/dmiopt.c b/dmiopt.c
index d08288f..42f5458 100644
--- a/dmiopt.c
+++ b/dmiopt.c
@@ -277,7 +277,7 @@ int parse_command_line(int argc, char * const argv[])
 		{ "from-dump", required_argument, NULL, 'F' },
 		{ "handle", required_argument, NULL, 'H' },
 		{ "oem-string", required_argument, NULL, 'O' },
-		{ "no-sysfs", no_argument, NULL, 'S' },
+		{ "no-procfs", no_argument, NULL, 'S' },
 		{ "version", no_argument, NULL, 'V' },
 		{ NULL, 0, NULL, 0 }
 	};
@@ -326,7 +326,7 @@ int parse_command_line(int argc, char * const argv[])
 				opt.flags |= FLAG_DUMP;
 				break;
 			case 'S':
-				opt.flags |= FLAG_NO_SYSFS;
+				opt.flags |= FLAG_NO_PROCFS;
 				break;
 			case 'V':
 				opt.flags |= FLAG_VERSION;
@@ -377,7 +377,7 @@ void print_help(void)
 		" -u, --dump             Do not decode the entries\n"
 		"     --dump-bin FILE    Dump the DMI data to a binary file\n"
 		"     --from-dump FILE   Read the DMI data from a binary file\n"
-		"     --no-sysfs         Do not attempt to read DMI data from sysfs files\n"
+		"     --no-procfs        Do not attempt to read DMI data from procfs files\n"
 		"     --oem-string N     Only display the value of the given OEM string\n"
 		" -V, --version          Display the version and exit\n";
 
