diff --git a/dmiopt.h b/dmiopt.h
index 2374637..d215aff 100644
--- a/dmiopt.h
+++ b/dmiopt.h
@@ -45,7 +45,7 @@ extern struct opt opt;
 #define FLAG_QUIET              (1 << 3)
 #define FLAG_DUMP_BIN           (1 << 4)
 #define FLAG_FROM_DUMP          (1 << 5)
-#define FLAG_NO_SYSFS           (1 << 6)
+#define FLAG_NO_PROCFS          (1 << 6)
 
 int parse_command_line(int argc, char * const argv[]);
 void print_help(void);
