From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Larkin <45925460+larb0b@users.noreply.github.com>
Date: Tue, 24 Sep 2019 02:56:39 -0400
Subject: [PATCH] Add missing includes

---
 library/ssl_tls.c  | 1 +
 library/x509_crt.c | 1 +
 2 files changed, 2 insertions(+)

diff --git a/library/ssl_tls.c b/library/ssl_tls.c
index 38690fa..5791783 100644
--- a/library/ssl_tls.c
+++ b/library/ssl_tls.c
@@ -49,6 +49,7 @@
 #include "mbedtls/platform_util.h"
 
 #include <string.h>
+#include <stdint.h>
 
 #if defined(MBEDTLS_X509_CRT_PARSE_C)
 #include "mbedtls/oid.h"
diff --git a/library/x509_crt.c b/library/x509_crt.c
index 97e1d72..9f81247 100644
--- a/library/x509_crt.c
+++ b/library/x509_crt.c
@@ -75,6 +75,7 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 #include <dirent.h>
+#include <unistd.h>
 #endif /* !_WIN32 || EFIX64 || EFI32 */
 #endif
 
