From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Larkin <45925460+larb0b@users.noreply.github.com>
Date: Tue, 24 Sep 2019 02:56:39 -0400
Subject: [PATCH] Remove NET_C from config

---
 include/mbedtls/config.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/include/mbedtls/config.h b/include/mbedtls/config.h
index 654f972..5b37f1f 100644
--- a/include/mbedtls/config.h
+++ b/include/mbedtls/config.h
@@ -2460,7 +2460,7 @@
  *
  * This module provides networking routines.
  */
-#define MBEDTLS_NET_C
+//#define MBEDTLS_NET_C
 
 /**
  * \def MBEDTLS_OID_C
