From 16b8f344dd913ea6ac9ff88fbc31f8e4a93f06d0 Mon Sep 17 00:00:00 2001
From: surrealegg <surrealegg@pm.me>
Date: Thu, 21 Sep 2023 16:46:57 +0400
Subject: [PATCH 2/3] Add missing install options

---
 CMakeLists.txt | 10 ++++++++++
 1 file changed, 10 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 091cb5a..532e628 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -92,3 +92,13 @@ add_library(enet SHARED
 if (MINGW)
     target_link_libraries(enet winmm ws2_32)
 endif()
+
+install(
+    TARGETS enet
+    LIBRARY DESTINATION lib
+)
+
+install(
+    DIRECTORY include/
+    DESTINATION include
+)
-- 
2.41.0

