From 67f1ac3f2f1d2c24035bf8b2cd0a279158b94d2e Mon Sep 17 00:00:00 2001
From: surrealegg <surrealegg@pm.me>
Date: Thu, 21 Sep 2023 22:25:02 +0400
Subject: [PATCH 3/3] Fix cmake deprecation warning

---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 532e628..6281d7e 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -1,4 +1,4 @@
-cmake_minimum_required(VERSION 2.6)
+cmake_minimum_required(VERSION 2.8.12...3.20)
 
 project(enet)
 
-- 
2.41.0

