diff -ruN libgd-2.3.3/config/config.sub libgd-2.3.3.patched/config/config.sub
--- libgd-2.3.3/config/config.sub	2021-09-11 01:12:11.000000000 -0400
+++ libgd-2.3.3.patched/config/config.sub	2022-01-05 15:38:17.115307773 -0500
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -serenity*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
