diff --git a/src/app/CMakeLists.txt b/src/app/CMakeLists.txt
index 95d0a51..be3e4db 100644
--- a/src/app/CMakeLists.txt
+++ b/src/app/CMakeLists.txt
@@ -13,4 +13,4 @@ endif()
 find_package(SDL2 REQUIRED)
 
 target_include_directories(chester PRIVATE ${SDL2_INCLUDE_DIR})
-target_link_libraries(chester libchester ${SDL2_LIBRARY})
+target_link_libraries(chester libchester ${SDL2_LIBRARY} -lm -lgfx -lgui -lipc -lcore)
