From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gloof11 <gloof11@proton.me>
Date: Thu, 29 Jun 2023 22:25:40 +0900
Subject: [PATCH] Added serenity as a proper define for time generation

---
 deps/spdlog/details/os.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/deps/spdlog/details/os.h b/deps/spdlog/details/os.h
index 11b8ee434687f8b1c318c4c85b393bb438ea53b5..ef3ff77bc1338105a49b688d5b633a17e3f76ae8 100644
--- a/deps/spdlog/details/os.h
+++ b/deps/spdlog/details/os.h
@@ -272,7 +272,7 @@ inline int utc_minutes_offset(const std::tm &tm = details::os::localtime())
     return offset;
 #else
 
-#if defined(sun) || defined(__sun) || defined(_AIX) || defined(__VITA__) || defined(__SWITCH__)
+#if defined(sun) || defined(__sun) || defined(_AIX) || defined(__VITA__) || defined(__SWITCH__) || defined(__serenity__)
     // 'tm_gmtoff' field is BSD extension and it's missing on SunOS/Solaris
     struct helper
     {
