From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: gloof11 <gloof11@proton.me>
Date: Thu, 29 Jun 2023 22:27:22 +0900
Subject: [PATCH] Removed hardware acceleration from SDL_CreateRenderer

---
 src/graphics/Renderer.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/graphics/Renderer.cpp b/src/graphics/Renderer.cpp
index 38c26ab47dcd165db86115cdf6c4c5aaa2b15c36..62268b5263671741c5b486d576ac9f749ff962f5 100644
--- a/src/graphics/Renderer.cpp
+++ b/src/graphics/Renderer.cpp
@@ -116,7 +116,7 @@ bool Renderer::initVideo()
 
   if (!_renderer)
   {
-    _renderer = SDL_CreateRenderer(_window, -1, SDL_RENDERER_ACCELERATED);
+    _renderer = SDL_CreateRenderer(_window, -1, SDL_RENDERER_SOFTWARE);
   }
   if (!_renderer)
   {
