--- tiff-4.2.0/config/config.sub.orig	2021-02-16 23:55:55.240259207 +0000
+++ tiff-4.2.0/config/config.sub	2021-02-16 23:55:59.792227377 +0000
@@ -1393,7 +1393,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos*)
+	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* | -tirtos* | -serenity* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
