From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Sat, 8 May 2021 05:23:29 +0200
Subject: [PATCH] Mark a possibly-unused variable as [[maybe_unused]]

---
 src/hb-buffer.hh | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/hb-buffer.hh b/src/hb-buffer.hh
index 8b432b5..ab18f2e 100644
--- a/src/hb-buffer.hh
+++ b/src/hb-buffer.hh
@@ -166,7 +166,7 @@ struct hb_buffer_t
 #ifndef HB_NDEBUG
     unsigned int end = start + count;
     assert (end <= 8);
-    unsigned int bits = (1u<<end) - (1u<<start);
+    [[maybe_unused]] unsigned int bits = (1u<<end) - (1u<<start);
     assert (bits == (allocated_var_bits & bits));
 #endif
   }
