From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dominika Liberda <ja@sdomi.pl>
Date: Tue, 13 Jun 2023 23:09:19 +0200
Subject: [PATCH] Build against freetype

---
 CMakeLists.txt | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 9a1d9032fcde002c27ab5b06c9665c1ca843d13a..c654284a3e6b9511b0433b8c0d25fccb57df30fb 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -37,10 +37,8 @@ option(HB_HAVE_FREETYPE "Enable freetype interop helpers" OFF)
 option(HB_HAVE_GRAPHITE2 "Enable Graphite2 complementary shaper" OFF)
 option(HB_HAVE_GLIB "Enable glib unicode functions" OFF)
 option(HB_HAVE_ICU "Enable icu unicode functions" OFF)
-if (TARGET freetype)
-  set (HB_HAVE_FREETYPE ON)
-  add_definitions(-DHAVE_FREETYPE=1)
-endif ()
+set (HB_HAVE_FREETYPE ON)
+add_definitions(-DHAVE_FREETYPE=1)
 if (APPLE)
   option(HB_HAVE_CORETEXT "Enable CoreText shaper backend on macOS" ON)
   set (CMAKE_MACOSX_RPATH ON)
