--- xproto-7.0.31.old/config.sub	2021-07-22 23:15:08.003666579 -0600
+++ xproto-7.0.31/config.sub	2021-07-22 23:15:45.513816681 -0600
@@ -3,7 +3,7 @@
 #   Copyright (C) 1992, 1993, 1994, 1995, 1996, 1997, 1998, 1999,
 #   2000, 2001, 2002, 2003, 2004, 2005 Free Software Foundation, Inc.
 
-timestamp='2006-02-23'
+timestamp='2021-07-22'
 
 # This file is (in principle) common to ALL GNU software.
 # The presence of a machine in this file suggests that SOME GNU software
@@ -1217,7 +1217,7 @@
 	      | -os2* | -vos* | -palmos* | -uclinux* | -nucleus* \
 	      | -morphos* | -superux* | -rtmk* | -rtmk-nova* | -windiss* \
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
-	      | -skyos* | -haiku* | -rdos*)
+	      | -skyos* | -haiku* | -rdos* | -serenity* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
