From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <ibara@users.noreply.github.com>
Date: Sat, 1 Feb 2020 14:54:04 -0500
Subject: [PATCH] Use stdbool instead of rolling a manual Bool

---
 ed.h | 3 +--
 1 file changed, 1 insertion(+), 2 deletions(-)

diff --git a/ed.h b/ed.h
index db0b4a6..ef0c134 100644
--- a/ed.h
+++ b/ed.h
@@ -18,8 +18,7 @@
 */
 
 #ifndef __cplusplus
-enum Bool { false = 0, true = 1 };
-typedef enum Bool bool;
+#include <stdbool.h>
 #endif
 
 enum Pflags			/* print suffixes */
