From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <ibara@users.noreply.github.com>
Date: Sat, 1 Feb 2020 14:54:04 -0500
Subject: [PATCH] Use pcre2 for regex instead of libc's regex.h

---
 regex.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/regex.c b/regex.c
index 9b38120..325b2f6 100644
--- a/regex.c
+++ b/regex.c
@@ -19,7 +19,7 @@
 
 #include <stddef.h>
 #include <errno.h>
-#include <regex.h>
+#include <pcre2posix.h>
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
