--- ed-1.18/configure.og	2022-03-31 22:58:11.000000000 +0200
+++ ed-1.18/configure	2022-03-31 23:01:31.000000000 +0200
@@ -21,10 +21,10 @@
 infodir='$(datarootdir)/info'
 mandir='$(datarootdir)/man'
 program_prefix=
-CC=gcc
-CPPFLAGS=
-CFLAGS='-Wall -W -O2'
-LDFLAGS=
+CC?=gcc
+CPPFLAGS?=
+CFLAGS?='-Wall -W -O2'
+LDFLAGS?=
 
 # checking whether we are using GNU C.
 /bin/sh -c "${CC} --version" > /dev/null 2>&1 || { CC=cc ; CFLAGS=-O2 ; }
