From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Thu, 23 Jan 2020 23:43:31 -0500
Subject: [PATCH] Replace libc's regex includes with pcre2

---
 src/flexdef.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/flexdef.h b/src/flexdef.h
index 9dac654..2bdf9fb 100644
--- a/src/flexdef.h
+++ b/src/flexdef.h
@@ -72,7 +72,7 @@
 #include <stdbool.h>
 #include <stdarg.h>
 /* Required: regcomp(), regexec() and regerror() in <regex.h> */
-#include <regex.h>
+#include <pcre2posix.h>
 /* Required: strcasecmp() in <strings.h> */
 #include <strings.h>
 #include "flexint.h"
