From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Thu, 23 Jan 2020 23:43:31 -0500
Subject: [PATCH] Link with pcre2

---
 src/Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index e570b87..1f8a305 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -342,7 +342,7 @@ LIBICONV = @LIBICONV@
 LIBINTL = @LIBINTL@
 LIBOBJS = @LIBOBJS@
 LIBPTHREAD = @LIBPTHREAD@
-LIBS = @LIBS@
+LIBS = @LIBS@ -lpcre2-posix -lpcre2-8
 LIBTOOL = @LIBTOOL@
 LIPO = @LIPO@
 LN_S = @LN_S@
