From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Thu, 23 Jan 2020 23:43:31 -0500
Subject: [PATCH] Hardcode the m4 include path

---
 src/Makefile.in | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/Makefile.in b/src/Makefile.in
index 1f8a305..a5dd2e2 100644
--- a/src/Makefile.in
+++ b/src/Makefile.in
@@ -350,7 +350,7 @@ LTLIBICONV = @LTLIBICONV@
 LTLIBINTL = @LTLIBINTL@
 LTLIBOBJS = @LTLIBOBJS@
 LT_SYS_LIBRARY_PATH = @LT_SYS_LIBRARY_PATH@
-M4 = @M4@
+M4 = /usr/local/bin/m4
 MAKEINFO = @MAKEINFO@
 MANIFEST_TOOL = @MANIFEST_TOOL@
 MKDIR_P = @MKDIR_P@
@@ -444,7 +444,7 @@ top_builddir = @top_builddir@
 top_srcdir = @top_srcdir@
 AM_YFLAGS = -d
 AM_CPPFLAGS = -DLOCALEDIR=\"$(localedir)\"
-m4 = @M4@
+m4 = /usr/local/bin/m4
 @ENABLE_LIBFL_TRUE@lib_LTLIBRARIES = libfl.la
 libfl_la_SOURCES = \
 	libmain.c \
