--- flex-2.6.4/configure.orig	Thu Jan 23 22:41:22 2020
+++ flex-2.6.4/configure	Thu Jan 23 22:44:41 2020
@@ -19854,7 +19854,7 @@
 
 # checks for headers
 
-for ac_header in regex.h strings.h sys/stat.h sys/wait.h unistd.h
+for ac_header in pcre2posix.h strings.h sys/stat.h sys/wait.h unistd.h
 do :
   as_ac_Header=`$as_echo "ac_cv_header_$ac_header" | $as_tr_sh`
 ac_fn_c_check_header_mongrel "$LINENO" "$ac_header" "$as_ac_Header" "$ac_includes_default"
@@ -20739,7 +20739,7 @@
 $as_echo "$as_me: WARNING: result $ac_cv_func_realloc_0_nonnull guessed because of cross compilation" >&2;}
 fi
 
-for ac_func in dup2 memset regcomp strcasecmp strchr strdup strtol
+for ac_func in dup2 memset strcasecmp strchr strdup strtol
 do :
   as_ac_var=`$as_echo "ac_cv_func_$ac_func" | $as_tr_sh`
 ac_fn_c_check_func "$LINENO" "$ac_func" "$as_ac_var"
