From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Eric Seifert <seiferteric@gmail.com>
Date: Tue, 21 Sep 2021 23:09:21 -0700
Subject: [PATCH] Remove the inline definitions of labs and llabs

---
 ext/bigdecimal/bigdecimal.h | 17 -----------------
 1 file changed, 17 deletions(-)

diff --git a/ext/bigdecimal/bigdecimal.h b/ext/bigdecimal/bigdecimal.h
index 28f3363..a2c8c7e 100644
--- a/ext/bigdecimal/bigdecimal.h
+++ b/ext/bigdecimal/bigdecimal.h
@@ -74,23 +74,6 @@ extern "C" {
 #endif
 #endif
 
-#ifndef HAVE_LABS
-static inline long
-labs(long const x)
-{
-    if (x < 0) return -x;
-    return x;
-}
-#endif
-
-#ifndef HAVE_LLABS
-static inline LONG_LONG
-llabs(LONG_LONG const x)
-{
-    if (x < 0) return -x;
-    return x;
-}
-#endif
 
 #ifndef HAVE_FINITE
 static int
