--- mgba-0.9.3/src/util/vfs/vfs-fd.c	2021-12-18 01:56:32.000000000 +0000
+++ mgba-0.9.3.serenity/src/util/vfs/vfs-fd.c	2022-04-13 02:50:08.792765951 +0100
@@ -200,11 +200,6 @@
 	UNUSED(size);
 	struct VFileFD* vfd = (struct VFileFD*) vf;
 #ifndef _WIN32
-#ifdef __HAIKU__
-	futimens(vfd->fd, NULL);
-#else
-	futimes(vfd->fd, NULL);
-#endif
 	if (buffer && size) {
 		return msync(buffer, size, MS_ASYNC) == 0;
 	}
