diff -ru libogg-1.3.4/config.sub libogg-1.3.4-patched/config.sub
--- libogg-1.3.4/config.sub	2021-03-08 15:40:56.932153900 +0100
+++ libogg-1.3.4-patched/config.sub	2021-03-08 15:40:43.633856800 +0100
@@ -1333,7 +1333,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
-	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | bsd* | mach* | minix* | genix* | ultrix* | serenity* | irix* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
