diff -ru make/arscan.c make-serenity/arscan.c
--- make/arscan.c	2016-05-21 22:21:52.000000000 +0200
+++ make-serenity/arscan.c	2019-07-21 16:11:09.161133749 +0200
@@ -331,7 +331,7 @@
 #endif
 
 #ifndef WINDOWS32
-# if !defined (__ANDROID__) && !defined (__BEOS__)
+# if !defined (__ANDROID__) && !defined (__BEOS__) && !defined (__serenity__)
 #  include <ar.h>
 # else
    /* These platforms don't have <ar.h> but have archives in the same format
diff -ru make/config/config.sub make-serenity/config/config.sub
--- make/config/config.sub	2016-05-22 15:27:16.000000000 +0200
+++ make-serenity/config/config.sub	2019-07-21 16:15:39.469065104 +0200
@@ -1374,6 +1374,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
