From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Sergey Bugaev <bugaevc@gmail.com>
Date: Wed, 2 Oct 2019 22:50:20 +0300
Subject: [PATCH] Disable backtrace on serenity

---
 tcc.h | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tcc.h b/tcc.h
index ccb3b60..8cd1b19 100644
--- a/tcc.h
+++ b/tcc.h
@@ -184,7 +184,7 @@ extern long double strtold (const char *__nptr, char **__endptr);
 # endif
 #endif
 
-#if defined TCC_IS_NATIVE && !defined CONFIG_TCCBOOT
+#if defined TCC_IS_NATIVE && !defined CONFIG_TCCBOOT && !defined __serenity__
 # define CONFIG_TCC_BACKTRACE
 # if (defined TCC_TARGET_I386 || defined TCC_TARGET_X86_64 || \
       defined TCC_TARGET_ARM || defined TCC_TARGET_ARM64 || \
