--- tinycc-dev/tcc.h	2019-05-14 22:40:39.000000000 +0300
+++ tinycc-dev-serenity/tcc.h	2019-10-02 13:37:16.919990212 +0300
@@ -162,7 +162,7 @@
 # endif
 #endif
 
-#if defined TCC_IS_NATIVE && !defined CONFIG_TCCBOOT
+#if defined TCC_IS_NATIVE && !defined CONFIG_TCCBOOT && !defined __serenity__
 # define CONFIG_TCC_BACKTRACE
 # if (defined TCC_TARGET_I386 || defined TCC_TARGET_X86_64) \
   && !defined TCC_UCLIBC && !defined TCC_MUSL
