From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Larkin <45925460+larb0b@users.noreply.github.com>
Date: Tue, 24 Sep 2019 02:56:39 -0400
Subject: [PATCH] Disable toilet fonts

---
 Makefile | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile b/Makefile
index 5a51d51..5044b4c 100644
--- a/Makefile
+++ b/Makefile
@@ -26,7 +26,7 @@ LDFLAGS =
 
 # Feature flags:
 #   define TLF_FONTS to use TOIlet TLF fonts
-XCFLAGS	= -DTLF_FONTS
+#XCFLAGS	= -DTLF_FONTS
 
 # Where to install files
 prefix	= /usr/local
