From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Sun, 9 Jan 2022 23:01:32 +0100
Subject: [PATCH] Teach configure about serenity

---
 configure | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/configure b/configure
index 5c28f7e35038236debbdeb3c396ce65f1189b13d..5bbf819fe31d24ac215d59f730bd64eac69c0bfd 100755
--- a/configure
+++ b/configure
@@ -5843,6 +5843,9 @@ EOF
 			mingw*)
 				OPENGL_LIBS="-lopengl32"
 				;;
+			serenity*)
+				OPENGL_LIBS="-lgl"
+				;;
 			*)
 				OPENGL_LIBS="-lGL"
 				;;
