From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Sun, 9 Jan 2022 23:18:02 +0100
Subject: [PATCH] Remove SDL timer lock

---
 backends/timer/sdl/sdl-timer.cpp | 6 ------
 1 file changed, 6 deletions(-)

diff --git a/backends/timer/sdl/sdl-timer.cpp b/backends/timer/sdl/sdl-timer.cpp
index f9d79ac..ce1af0e 100644
--- a/backends/timer/sdl/sdl-timer.cpp
+++ b/backends/timer/sdl/sdl-timer.cpp
@@ -29,11 +29,7 @@
 
 #include "common/textconsole.h"
 
-OSystem::MutexRef timerMutex;
-
 static Uint32 timer_handler(Uint32 interval, void *param) {
-	Common::StackLock lock(timerMutex);
-
 	((DefaultTimerManager *)param)->handler();
 	return interval;
 }
@@ -49,8 +45,6 @@ SdlTimerManager::SdlTimerManager() {
 }
 
 SdlTimerManager::~SdlTimerManager() {
-	Common::StackLock lock(timerMutex);
-
 	// Removes the timer callback
 	SDL_RemoveTimer(_timerID);
 
