--- scummvm-2.5.1/configure	2021-12-24 22:19:28.000000000 +0000
+++ scummvm-2.5.1-patched/configure	2022-01-09 21:55:40.753734211 +0000
@@ -3998,7 +3998,7 @@
 	amigaos* | cygwin* | dreamcast | ds | gamecube | mingw* | morphos | n64 | ps3 | psp2 | psp | riscos | wii)
 		_posix=no
 		;;
-	3ds | android | androidsdl | beos* | bsd* | darwin* | freebsd* | gnu* | gph-linux | haiku* | hpux* | iphone | ios7 | irix*| k*bsd*-gnu* | linux* | maemo | mint* | netbsd* | openbsd* | solaris* | sunos* | switch | uclinux*)
+	3ds | android | androidsdl | beos* | bsd* | darwin* | freebsd* | gnu* | gph-linux | haiku* | hpux* | iphone | ios7 | irix*| k*bsd*-gnu* | linux* | maemo | mint* | netbsd* | openbsd* | serenity* | solaris* | sunos* | switch | uclinux*)
 		_posix=yes
 		;;
 	os2-emx*)
@@ -5538,6 +5538,9 @@
 		mingw*)
 			OPENGL_LIBS="-lopengl32"
 			;;
+		serenity*)
+			OPENGL_LIBS="-lgl"
+			;;
 		*)
 			OPENGL_LIBS="-lGL"
 			;;
