--- scummvm-2.5.0/audio/fmopl.cpp	2021-10-01 13:52:37.000000000 +0200
+++ scummvm-2.5.0-patched/audio/fmopl.cpp	2021-10-24 16:20:17.548613828 +0200
@@ -69,7 +69,6 @@
 
 const Config::EmulatorDescription Config::_drivers[] = {
 	{ "auto", "<default>", kAuto, kFlagOpl2 | kFlagDualOpl2 | kFlagOpl3 },
-	{ "mame", _s("MAME OPL emulator"), kMame, kFlagOpl2 },
 #ifndef DISABLE_DOSBOX_OPL
 	{ "db", _s("DOSBox OPL emulator"), kDOSBox, kFlagOpl2 | kFlagDualOpl2 | kFlagOpl3 },
 #endif
