diff -Naur libksba-1.5.1/build-aux/config.sub libksba-1.5.1.serenity/build-aux/config.sub
--- libksba-1.5.1/build-aux/config.sub	2016-08-22 10:40:58.000000000 +0200
+++ libksba-1.5.1.serenity/build-aux/config.sub	2021-04-14 02:31:15.924387614 +0200
@@ -1381,7 +1381,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
