--- php-8.0.6/configure	2021-06-04 20:36:10.053986191 +0200
+++ php-8.0.6-patched/configure	2021-06-04 20:36:20.769935808 +0200
@@ -91478,8 +91478,8 @@
 CXXFLAGS="$CXXFLAGS $standard_libtool_flag \$(PROF_FLAGS)"
 
 if test "$PHP_PHAR" != "no" && test "$PHP_CLI" != "no"; then
-  pharcmd=pharcmd
-  pharcmd_install=install-pharcmd
+  pharcmd=
+  pharcmd_install=
 else
   pharcmd=
   pharcmd_install=
