--- mpfr-4.1.0/config.sub	2020-07-10 12:59:02.000000000 +0100
+++ mpfr-4.1.0.serenity/config.sub	2021-07-28 18:05:59.462525103 +0100
@@ -1390,7 +1390,7 @@
 	      | -powermax* | -dnix* | -nx6 | -nx7 | -sei* | -dragonfly* \
 	      | -skyos* | -haiku* | -rdos* | -toppers* | -drops* | -es* \
 	      | -onefs* | -tirtos* | -phoenix* | -fuchsia* | -redox* | -bme* \
-	      | -midnightbsd*)
+	      | -midnightbsd* | -serenity*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
