From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Mon, 27 Mar 2023 20:04:41 +1100
Subject: [PATCH] Error on unknown arch

---
 include/private/gcconfig.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/private/gcconfig.h b/include/private/gcconfig.h
index b410f188f91960977a11695bf2a9da4639a32e94..2e36e697a0ab1ba89db9c18a4caa35d381f8ec90 100644
--- a/include/private/gcconfig.h
+++ b/include/private/gcconfig.h
@@ -351,6 +351,8 @@ EXTERN_C_BEGIN
 #        define I386
 #    elif defined(__x86_64__)
 #        define X86_64
+#    else
+#        error "Invalid arch for serenity"
 #    endif
 #endif
 # if (defined(__BEOS__) || defined(__HAIKU__)) && defined(_X86_)
