From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Michael Manganiello <adamantike@users.noreply.github.com>
Date: Sun, 27 Mar 2022 12:52:11 -0300
Subject: [PATCH] Disable SDL's accelerated rendering

---
 src/engine/screen.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/engine/screen.cpp b/src/engine/screen.cpp
index 9cd9ccb..f4fe315 100644
--- a/src/engine/screen.cpp
+++ b/src/engine/screen.cpp
@@ -1005,10 +1005,10 @@ namespace
         int renderFlags() const
         {
             if ( _isVSyncEnabled ) {
-                return ( SDL_RENDERER_ACCELERATED | SDL_RENDERER_PRESENTVSYNC );
+                return ( SDL_RENDERER_SOFTWARE | SDL_RENDERER_PRESENTVSYNC );
             }
 
-            return SDL_RENDERER_ACCELERATED;
+            return SDL_RENDERER_SOFTWARE;
         }
 
         void _createPalette()
