From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Tue, 27 Apr 2021 14:58:09 +0200
Subject: [PATCH] Remove uses of ipv6

---
 tclUnixSock.c | 5 +++++
 1 file changed, 5 insertions(+)

diff --git a/tclUnixSock.c b/tclUnixSock.c
index 1a54914..121f8a4 100644
--- a/tclUnixSock.c
+++ b/tclUnixSock.c
@@ -706,6 +706,7 @@ static inline int
 IPv6AddressNeedsNumericRendering(
     struct in6_addr addr)
 {
+#ifndef __serenity__
     if (IN6_ARE_ADDR_EQUAL(&addr, &in6addr_any)) {
         return 1;
     }
@@ -716,11 +717,15 @@ IPv6AddressNeedsNumericRendering(
      */
 
     if (!IN6_IS_ADDR_V4MAPPED(&addr)) {
+#else
         return 0;
+#endif
+#ifndef __serenity__
     }
 
     return (addr.s6_addr[12] == 0 && addr.s6_addr[13] == 0
             && addr.s6_addr[14] == 0 && addr.s6_addr[15] == 0);
+#endif
 }
 #if defined (__clang__) || ((__GNUC__)  && ((__GNUC__ > 4) || ((__GNUC__ == 4) && (__GNUC_MINOR__ > 5))))
 #pragma GCC diagnostic pop
