From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Sun, 21 Jun 2020 18:41:01 -0400
Subject: [PATCH] Disable crossbuild

(?)
---
 configure.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure.py b/configure.py
index cded265..134db91 100755
--- a/configure.py
+++ b/configure.py
@@ -687,7 +687,7 @@ n.build('all', 'phony', all_targets)
 n.close()
 print('wrote %s.' % BUILD_FILENAME)
 
-if options.bootstrap:
+if options.bootstrap and False:
     print('bootstrap complete.  rebuilding...')
 
     rebuild_args = []
