From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Sun, 21 Jun 2020 18:41:01 -0400
Subject: [PATCH] Disable load average calculation

---
 src/util.cc | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/src/util.cc b/src/util.cc
index c76f730..5ec2bb8 100644
--- a/src/util.cc
+++ b/src/util.cc
@@ -588,11 +588,14 @@ double GetLoadAverage() {
 #else
 double GetLoadAverage() {
   double loadavg[3] = { 0.0f, 0.0f, 0.0f };
+  /*
+   * FIXME: serenity way of getting loadavg
   if (getloadavg(loadavg, 3) < 0) {
     // Maybe we should return an error here or the availability of
     // getloadavg(3) should be checked when ninja is configured.
     return -0.0f;
   }
+  */
   return loadavg[0];
 }
 #endif // _WIN32
