From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Nico Weber <thakis@chromium.org>
Date: Sun, 21 Jun 2020 18:41:01 -0400
Subject: [PATCH] Include unistd.h for stat()

---
 src/disk_interface.cc | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/disk_interface.cc b/src/disk_interface.cc
index 49af001..cb1fad5 100644
--- a/src/disk_interface.cc
+++ b/src/disk_interface.cc
@@ -21,6 +21,7 @@
 #include <string.h>
 #include <sys/stat.h>
 #include <sys/types.h>
+#include <unistd.h>
 
 #ifdef _WIN32
 #include <sstream>
