diff -Naur ninja-1.8.2/src/ninja.cc ninja-1.8.2.serenity/src/ninja.cc
--- ninja-1.8.2/src/ninja.cc	2017-09-11 03:20:10.000000000 +0200
+++ ninja-1.8.2.serenity/src/ninja.cc	2021-04-11 23:54:18.156949715 +0200
@@ -1207,6 +1207,7 @@
 
 }  // anonymous namespace
 
+__attribute__((visibility("default")))
 int main(int argc, char** argv) {
 #if defined(_MSC_VER)
   // Set a handler to catch crashes not caught by the __try..__except
