diff --git a/config.sub b/config.sub
index a44fd8ae..eb9be19b 100755
--- a/config.sub
+++ b/config.sub
@@ -1367,7 +1367,7 @@ case $os in
 	     | powermax* | dnix* | nx6 | nx7 | sei* | dragonfly* \
 	     | skyos* | haiku* | rdos* | toppers* | drops* | es* \
 	     | onefs* | tirtos* | phoenix* | fuchsia* | redox* | bme* \
-	     | midnightbsd* | amdhsa* | unleashed* | emscripten*)
+	     | midnightbsd* | amdhsa* | unleashed* | emscripten* | *serenity* )
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	qnx*)
