From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Thu, 27 Jul 2023 16:49:13 +0200
Subject: [PATCH] Disable nanosleep

---
 config_h.SH | 1 -
 1 file changed, 1 deletion(-)

diff --git a/config_h.SH b/config_h.SH
index 5880dc532de82a795f0ab3c9d65d84b3679c24b5..f63bf2aa1a848677c71e98017fe4e48f0e5cc8e6 100755
--- a/config_h.SH
+++ b/config_h.SH
@@ -3116,7 +3116,6 @@ sed <<!GROK!THIS! >$CONFIG_H -e 's!^#undef\(.*/\)\*!/\*#define\1 \*!' -e 's!^#un
  *	This symbol, if defined, indicates that the nanosleep
  *	system call is available to sleep with 1E-9 sec accuracy.
  */
-#$d_nanosleep HAS_NANOSLEEP		/**/
 
 /* HAS_NEARBYINT:
  *	This symbol, if defined, indicates that the nearbyint routine is
