From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Beckett Normington <beckett@b0ba.dev>
Date: Tue, 25 Jul 2023 17:00:13 -0400
Subject: [PATCH] pp_sys: Avoid redefining `h_errno`

---
 pp_sys.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pp_sys.c b/pp_sys.c
index b18911b3f4b047fa9b8422f66530cbee094af824..0bef81722379e4f28e18f622f9e1c977a9debd63 100644
--- a/pp_sys.c
+++ b/pp_sys.c
@@ -69,7 +69,7 @@
    compiling multithreaded and singlethreaded ($ccflags et al).
    HOST_NOT_FOUND is typically defined in <netdb.h>.
 */
-#if defined(HOST_NOT_FOUND) && !defined(h_errno) && !defined(__CYGWIN__)
+#if defined(HOST_NOT_FOUND) && !defined(h_errno) && !defined(__CYGWIN__) && !defined(__serenity__)
 extern int h_errno;
 #endif
 
