--- libtool-2.4.6/libltdl/config/config.sub	2010-09-22 07:46:09.000000000 -0700
+++ libtool-2.4.6/libltdl/config/config.sub	2021-04-11 18:46:03.374661972 -0700
@@ -1301,6 +1301,7 @@
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
 	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	      | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
