diff -Naur xz-5.2.5/build-aux/config.sub xz-5.2.5.serenity/build-aux/config.sub
--- xz-5.2.5/build-aux/config.sub	2020-03-17 15:29:35.000000000 +0100
+++ xz-5.2.5.serenity/build-aux/config.sub	2021-04-18 21:28:26.405741777 +0200
@@ -1363,7 +1363,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
