From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Schumacher <timschumi@gmx.de>
Date: Wed, 10 Nov 2021 02:07:36 +0100
Subject: [PATCH] Remove oversized dirfd cache

---
 src/safe.c | 12 +-----------
 1 file changed, 1 insertion(+), 11 deletions(-)

diff --git a/src/safe.c b/src/safe.c
index eda99224b5e7ae83527f7e8ff0571aa9b918602b..1921e39e0cf77fc62de42b8c51168d0de50a913c 100644
--- a/src/safe.c
+++ b/src/safe.c
@@ -103,17 +103,7 @@ static void free_cached_dirfd (struct cached_dirfd *entry)
 
 static void init_dirfd_cache (void)
 {
-  struct rlimit nofile;
-
-  if (getrlimit (RLIMIT_NOFILE, &nofile) == 0)
-    {
-      if (nofile.rlim_cur == RLIM_INFINITY)
-        max_cached_fds = RLIM_INFINITY;
-      else
-	max_cached_fds = MAX (nofile.rlim_cur / 4, min_cached_fds);
-    }
-  else
-    max_cached_fds = min_cached_fds;
+  max_cached_fds = 64;
 
   cached_dirfds = hash_initialize (min_cached_fds, nullptr,
 				   hash_cached_dirfd,
