From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Valtteri Koskivuori <vkoskiv@gmail.com>
Date: Tue, 25 Feb 2020 22:03:47 +0200
Subject: [PATCH] Add a dummy configure file

---
 configure | 1 +
 1 file changed, 1 insertion(+)
 create mode 100644 configure

diff --git a/configure b/configure
new file mode 100644
index 0000000..04bfb39
--- /dev/null
+++ b/configure
@@ -0,0 +1 @@
+#nop
