--- c-ray-master/src/libraries/asprintf.h	2021-03-13 00:16:23.782000000 +0100
+++ c-ray-master/src/libraries/asprintf.h	2021-03-13 00:16:41.350553067 +0100
@@ -33,7 +33,7 @@
  * GNU-C-compatible compilers implement these with the same names, thus we
  * don't have to do anything
  */
-#ifdef _MSC_VER
+#if defined(_MSC_VER) || defined(__serenity__)
 int cray_vasprintf(char **strp, const char *format, va_list ap) {
 	int len = vscprintf(format, ap);
 	if (len == -1)
