--- c-ray-master/CMakeLists.txt	2021-03-13 00:40:48.474000000 +0100
+++ c-ray-master/CMakeLists.txt	2021-03-13 00:41:04.770789629 +0100
@@ -38,7 +38,7 @@
 	set(CMAKE_EXE_LINKER_FLAGS_RELEASE "/DEBUG /MANIFEST:NO /INCREMENTAL:NO /OPT:REF,ICF" CACHE STRING "" FORCE)
 else()
 	# set(CMAKE_C_FLAGS "-Wall -Wextra -pedantic -Wconversion -std=gnu99")
-	set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-missing-field-initializers -std=gnu99 -march=native -mtune=native -D_GNU_SOURCE")
+	set(CMAKE_C_FLAGS "-Wall -Wextra -Wno-missing-field-initializers -std=gnu99 -mtune=native -D_GNU_SOURCE")
 	set(CMAKE_C_FLAGS_RELEASE "-O3 -ftree-vectorize")
 	set(CMAKE_C_FLAGS_DEBUG "${CMAKE_C_FLAGS_DEBUG} -O0 -g")
 	if (ASAN)
@@ -71,7 +71,7 @@
 target_include_directories(c-ray PRIVATE ${c-ray_SOURCE_DIR}/src)
 target_include_directories(c-ray PRIVATE ${c-ray_SOURCE_DIR}/tests)
 if (NOT MSVC)
-	target_link_libraries(c-ray PRIVATE -lpthread -lm)
+	target_link_libraries(c-ray PRIVATE -lSDL2 -lgui -lgfx -lipc -lcore -lpthread -lstdc++ -lm)
 endif ()
 
 include(CheckIPOSupported)
