--- c-ray-master/src/renderer/renderer.c	2021-03-13 22:08:24.699323180 +0100
+++ c-ray-master/src/renderer/renderer.c	2021-03-13 22:08:53.513237904 +0100
@@ -108,7 +108,7 @@
 			float sps = (1000000.0f/usPerRay) * r->prefs.threadCount;
 			char rem[64];
 			smartTime((msecTillFinished) / r->prefs.threadCount, rem);
-			logr(info, "[%s%.0f%%%s] μs/path: %.02f, etf: %s, %.02lfMs/s %s        \r",
+			logr(info, "[%s%.0f%%%s] us/path: %.02f, etf: %s, %.02lfMs/s %s        \r",
 				 KBLU,
 				 interactive ? ((float)r->state.finishedPasses / (float)r->prefs.sampleCount) * 100.0f :
 							   ((float)r->state.finishedTileCount / (float)r->state.tileCount) * 100.0f,
