--- c-ray-master/src/utils/ui.c	2021-03-13 00:30:25.488000000 +0100
+++ c-ray-master/src/utils/ui.c	2021-03-13 00:30:41.056572879 +0100
@@ -120,7 +120,7 @@
 		return;
 	}
 	//Init renderer
-	gdisplay->renderer = SDL_CreateRenderer(gdisplay->window, -1, SDL_RENDERER_ACCELERATED);
+	gdisplay->renderer = SDL_CreateRenderer(gdisplay->window, -1, SDL_RENDERER_SOFTWARE);
 	if (gdisplay->renderer == NULL) {
 		logr(warning, "Renderer couldn't be created, error: \"%s\"\n", SDL_GetError());
 		destroyDisplay();
