diff -ur Angband-4.2.3-orig/config.sub Angband-4.2.3/config.sub
--- Angband-4.2.3-orig/config.sub	2021-08-01 01:29:20.000000000 +0200
+++ Angband-4.2.3/config.sub	2021-09-04 19:01:55.625519348 +0200
@@ -1337,7 +1337,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -serenity* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* \
