From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 9 Jul 2021 04:56:55 +0430
Subject: [PATCH] fs: Stub out unsupported syscalls

---
 src/unix/fs.c | 9 ++++++---
 1 file changed, 6 insertions(+), 3 deletions(-)

diff --git a/src/unix/fs.c b/src/unix/fs.c
index 717f3fab36939e017f09e0ac5f2a2f658140ae38..3fe0f11c75b404a5655df1c5c644e0b852ca94a4 100644
--- a/src/unix/fs.c
+++ b/src/unix/fs.c
@@ -77,7 +77,8 @@
       defined(__MVS__)    || \
       defined(__NetBSD__) || \
       defined(__HAIKU__)  || \
-      defined(__QNX__)
+      defined(__QNX__)    || \
+      defined(__serenity__)
 # include <sys/statvfs.h>
 #else
 # include <sys/statfs.h>
@@ -683,7 +684,8 @@ static int uv__fs_statfs(uv_fs_t* req) {
     defined(__MVS__)    || \
     defined(__NetBSD__) || \
     defined(__HAIKU__)  || \
-    defined(__QNX__)
+    defined(__QNX__)    || \
+    defined(__serenity__)
   struct statvfs buf;
 
   if (0 != statvfs(req->path, &buf))
@@ -705,7 +707,8 @@ static int uv__fs_statfs(uv_fs_t* req) {
     defined(__OpenBSD__)  || \
     defined(__NetBSD__)   || \
     defined(__HAIKU__)    || \
-    defined(__QNX__)
+    defined(__QNX__)      || \
+    defined(__serenity__)
   stat_fs->f_type = 0;  /* f_type is not supported. */
 #else
   stat_fs->f_type = buf.f_type;
