From ef46efb9fb0a1faa306fb1e74a0b44faf6c6d41f Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 9 Jul 2021 04:56:55 +0430
Subject: [PATCH 2/7] fs: Stub out unsupported syscalls

---
 src/unix/fs.c | 24 +++++++++++++++++++++++-
 1 file changed, 23 insertions(+), 1 deletion(-)

diff --git a/src/unix/fs.c b/src/unix/fs.c
index eb17fb4..93b457b 100644
--- a/src/unix/fs.c
+++ b/src/unix/fs.c
@@ -1666,7 +1680,9 @@ static void uv__fs_work(struct uv__work* w) {
     X(COPYFILE, uv__fs_copyfile(req));
     X(FCHMOD, fchmod(req->file, req->mode));
     X(FCHOWN, fchown(req->file, req->uid, req->gid));
+#ifndef __serenity__
     X(LCHOWN, lchown(req->path, req->uid, req->gid));
+#endif
     X(FDATASYNC, uv__fs_fdatasync(req));
     X(FSTAT, uv__fs_fstat(req->file, &req->statbuf));
     X(FSYNC, uv__fs_fsync(req));
@@ -1805,7 +1823,9 @@ int uv_fs_lchown(uv_loop_t* loop,
                  uv_uid_t uid,
                  uv_gid_t gid,
                  uv_fs_cb cb) {
+#ifndef __serenity__
   INIT(LCHOWN);
+#endif
   PATH;
   req->uid = uid;
   req->gid = gid;
-- 
2.32.0

