From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 9 Jul 2021 04:57:31 +0430
Subject: [PATCH] stream: Don't use AF_INET6

---
 src/unix/stream.c | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/src/unix/stream.c b/src/unix/stream.c
index 619b625..10cd4f7 100644
--- a/src/unix/stream.c
+++ b/src/unix/stream.c
@@ -994,13 +994,19 @@ uv_handle_type uv__handle_type(int fd) {
       case AF_UNIX:
         return UV_NAMED_PIPE;
       case AF_INET:
+#ifndef __serenity__
       case AF_INET6:
+#endif
         return UV_TCP;
       }
   }
 
   if (type == SOCK_DGRAM &&
-      (ss.ss_family == AF_INET || ss.ss_family == AF_INET6))
+      (ss.ss_family == AF_INET
+#ifndef __serenity__
+       || ss.ss_family == AF_INET6
+#endif
+    ))
     return UV_UDP;
 
   return UV_UNKNOWN_HANDLE;
