From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 9 Jul 2021 05:01:05 +0430
Subject: [PATCH] build: Add SerenityOS platform definitions

---
 CMakeLists.txt | 15 +++++++++++++++
 1 file changed, 15 insertions(+)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ac52412..d285ed9 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -378,6 +378,21 @@ if(CMAKE_SYSTEM_NAME STREQUAL "QNX")
   list(APPEND uv_libraries socket)
 endif()
 
+if(CMAKE_SYSTEM_NAME STREQUAL "SerenityOS")
+  list(APPEND uv_headers
+    include/uv/posix.h
+  )
+  list(APPEND uv_sources
+    src/unix/posix-hrtime.c
+    src/unix/posix-poll.c
+    src/unix/no-fsevents.c
+    src/unix/no-proctitle.c
+  )
+  list(APPEND uv_libraries
+    dl
+  )
+endif()
+
 if(APPLE OR CMAKE_SYSTEM_NAME MATCHES "DragonFly|FreeBSD|Linux|NetBSD|OpenBSD")
   list(APPEND uv_test_libraries util)
 endif()
