From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Ali Mohammad Pur <ali.mpfard@gmail.com>
Date: Fri, 9 Jul 2021 05:02:01 +0430
Subject: [PATCH] include: Teach the header about serenity

---
 include/uv/unix.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/uv/unix.h b/include/uv/unix.h
index ea37d78..65e3212 100644
--- a/include/uv/unix.h
+++ b/include/uv/unix.h
@@ -69,6 +69,8 @@
       defined(__QNX__)    || \
       defined(__GNU__)
 # include "uv/posix.h"
+#elif defined(__serenity__)
+# include "uv/posix.h"
 #endif
 
 #ifndef NI_MAXHOST
