From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: circl <circl.lastname@gmail.com>
Date: Sun, 20 Mar 2022 17:16:16 +0100
Subject: [PATCH] Open links and directories using open(1)

---
 src/common/Platform.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/common/Platform.cpp b/src/common/Platform.cpp
index d341561..d32615c 100644
--- a/src/common/Platform.cpp
+++ b/src/common/Platform.cpp
@@ -137,7 +137,7 @@ void OpenURI(ByteString uri)
 	{
 		fprintf(stderr, "cannot open URI: ShellExecute(...) failed\n");
 	}
-#elif defined(MACOSX)
+#elif defined(MACOSX) || defined(__serenity__)
 	if (system(("open \"" + uri + "\"").c_str()))
 	{
 		fprintf(stderr, "cannot open URI: system(...) failed\n");
