diff -Naur nano-4.5/lib/fnmatch_loop.c nano-4.5.serenity/lib/fnmatch_loop.c
--- nano-4.5/lib/fnmatch_loop.c	2021-04-12 03:21:50.410647764 +0200
+++ nano-4.5.serenity/lib/fnmatch_loop.c	2021-04-12 03:20:03.688773104 +0200
@@ -16,6 +16,9 @@
 
 /* Match STRING against the file name pattern PATTERN, returning zero if
    it matches, nonzero if not.  */
+#ifdef __serenity__
+#define FNM_EXTMATCH 0
+#endif
 static int EXT (INT opt, const CHAR *pattern, const CHAR *string,
                 const CHAR *string_end, bool no_leading_period, int flags)
      internal_function;
