From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Mon, 9 May 2022 00:29:43 +0200
Subject: [PATCH] Build: Allow `CFLAGS` env var to be set

---
 configure | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/configure b/configure
index 226d00e..05ebf5d 100755
--- a/configure
+++ b/configure
@@ -2805,7 +2805,7 @@ else
   RANLIB="$ac_cv_prog_RANLIB"
 fi
 
-CFLAGS="-DSEPARATOR=\"'/'\""
+CFLAGS="$CFLAGS -DSEPARATOR=\"'/'\""
 
 # Check whether --enable-warn or --disable-warn was given.
 if test "${enable_warn+set}" = set; then
