From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Jelle Raaijmakers <jelle@gmta.nl>
Date: Mon, 9 May 2022 00:30:04 +0200
Subject: [PATCH] Build: Replace `-lGL` with `-lgl` to reference our LibGL

---
 configure | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/configure b/configure
index 226d00eb04f048a51cfbc742ccfe62cfb7ad6750..086139a0e045fe6ad914fc641625fc281ca630e7 100755
--- a/configure
+++ b/configure
@@ -3130,7 +3130,7 @@ if test "${ac_cv_lib_GL_main+set}" = set; then
   echo $ECHO_N "(cached) $ECHO_C" >&6
 else
   ac_check_lib_save_LIBS=$LIBS
-LIBS="-lGL  $LIBS"
+LIBS="-lgl  $LIBS"
 cat >conftest.$ac_ext <<_ACEOF
 #line 3135 "configure"
 #include "confdefs.h"
@@ -3171,7 +3171,7 @@ if test $ac_cv_lib_GL_main = yes; then
 #define HAVE_LIBGL 1
 EOF
 
-  LIBS="-lGL $LIBS"
+  LIBS="-lgl $LIBS"
 
 else
   { { echo "$as_me:3177: error: OpenGL is not installed" >&5
