--- gltron-0.70/config.sub	2002-03-23 17:40:17.000000000 +0000
+++ gltron-0.70-patched/config.sub	2022-01-09 23:05:14.342418312 +0000
@@ -1070,7 +1070,7 @@
 	      | -interix* | -uwin* | -rhapsody* | -darwin* | -opened* \
 	      | -openstep* | -oskit* | -conix* | -pw32* | -nonstopux* \
 	      | -storm-chaos* | -tops10* | -tenex* | -tops20* | -its* \
-	      | -os2* | -vos*)
+	      | -os2* | -vos* | -serenity*)
 	# Remember, each alternative MUST END IN *, to match a version number.
 		;;
 	-qnx*)
