From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Noah Rosamilia <ivoahivoah@gmail.com>
Date: Thu, 6 Jan 2022 18:28:55 -0500
Subject: [PATCH] Don't build the html target by default

---
 Makefile.in | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/Makefile.in b/Makefile.in
index e49616a..8d03935 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -67,7 +67,7 @@ CFLAGS = $(FLAGS) $(PROF) $(CYG) $(WARN) $(DEFS) $(LIBDIRPATH)
 ##
 ## How to build the target:
 ##
-all:		npiet npiet-foogol html
+all:		npiet npiet-foogol
 
 npiet:		npiet.o
 	$(CC) $(LDFLAGS) $(PROF) -o npiet npiet.o $(LIBS)
