From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Noah Rosamilia <ivoahivoah@gmail.com>
Date: Thu, 6 Jan 2022 18:28:55 -0500
Subject: [PATCH] Remove a 'malloc.h' include

---
 npiet-foogol.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/npiet-foogol.c b/npiet-foogol.c
index 4d9174a..70fc2cd 100644
--- a/npiet-foogol.c
+++ b/npiet-foogol.c
@@ -108,7 +108,7 @@ char *version = "v1.3";
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
-#include <malloc.h>
+// #include <malloc.h>
 #include <limits.h>
 #include <time.h>
 
