From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Sat, 17 Apr 2021 15:40:17 +0200
Subject: [PATCH] Don't use the asm version of rdtsc

---
 src/cpu.cpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/cpu.cpp b/src/cpu.cpp
index d95c704..762da79 100644
--- a/src/cpu.cpp
+++ b/src/cpu.cpp
@@ -40,7 +40,7 @@ unsigned __int64 ottd_rdtsc();
 #endif
 
 /* rdtsc for all other *nix-en (hopefully). Use GCC syntax */
-#if (defined(__i386__) || defined(__x86_64__)) && !defined(RDTSC_AVAILABLE)
+#if (defined(__i386__) || defined(__x86_64__)) && !defined(__serenity__) && !defined(RDTSC_AVAILABLE)
 uint64 ottd_rdtsc()
 {
 	uint32 high, low;
