From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: xSlendiX <gamingxslendix@gmail.com>
Date: Sat, 18 Sep 2021 16:19:50 +0300
Subject: [PATCH] Include strings.h

---
 src/libmodplug/stdafx.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/src/libmodplug/stdafx.h b/src/libmodplug/stdafx.h
index d45d949..23c66fc 100644
--- a/src/libmodplug/stdafx.h
+++ b/src/libmodplug/stdafx.h
@@ -62,6 +62,7 @@ inline void ProcessPlugins(int n) {}
 #include <stdlib.h>
 #include <stdio.h>
 #include <string.h>
+#include <strings.h>
 #ifdef HAVE_MALLOC_H
 #include <malloc.h>
 #endif
