diff -Naur libmodplug-0.8.8.5/config.sub libmodplug-0.8.8.5.serenity/config.sub
--- libmodplug-0.8.8.5/config.sub	2017-03-17 09:34:37.000000000 +0100
+++ libmodplug-0.8.8.5.serenity/config.sub	2021-04-14 01:12:00.264606451 +0200
@@ -1381,7 +1381,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
