From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andreas Kling <kling@serenityos.org>
Date: Tue, 15 Dec 2020 01:06:18 +0100
Subject: [PATCH] Include ar.h for serenity as well

---
 src/arscan.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/arscan.c b/src/arscan.c
index b6ed45a87754085aa1776c85e0181ba6edb0d7e6..2ae76f8134a581f3607a7d0c507d1d2e21dc3abc 100644
--- a/src/arscan.c
+++ b/src/arscan.c
@@ -331,7 +331,7 @@ ar_scan (const char *archive, ar_member_func_t function, const void *varg)
 #endif
 
 #ifndef WINDOWS32
-# if !defined (__ANDROID__) && !defined (__BEOS__)
+# if !defined (__ANDROID__) && !defined (__BEOS__) && !defined(__serenity__)
 #  include <ar.h>
 # else
    /* These platforms don't have <ar.h> but have archives in the same format
