From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Fabian Dellwing <fabian.dellwing@gmail.com>
Date: Tue, 25 Apr 2023 15:35:52 +0200
Subject: [PATCH] Remove `_GNU_SOURCE` as we are not GNU. With it we segfault
 after calling `basename()`.

---
 ssmtp.c | 1 -
 1 file changed, 1 deletion(-)

diff --git a/ssmtp.c b/ssmtp.c
index af4d1e58d28fa9450bfc6a80fbacc75ca28c2220..985858f42420a11f3a78ec8f8c281091e17fdc54 100644
--- a/ssmtp.c
+++ b/ssmtp.c
@@ -11,7 +11,6 @@
 
 */
 #define VERSION "2.64"
-#define _GNU_SOURCE
 
 #include <sys/socket.h>
 #include <netinet/in.h>
