From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Liav A <liavalb@gmail.com>
Date: Thu, 7 Jul 2022 23:49:02 +0300
Subject: [PATCH] Use LibMath frexp function

---
 Makefile | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/Makefile b/Makefile
index 4dc4f9a3b57b19582576cf3bf69f7ef6004b1c2c..ae10548430b7c3f6140c36631cc41f0200a71a95 100644
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,6 @@
 CC ?= gcc
 CFLAGS_BENCH ?= -std=gnu99 -O3
-LFLAGS_BENCH ?= -lpng
+LFLAGS_BENCH ?= -lpng -lm
 CFLAGS_CONV ?= -std=c99 -O3
 
 TARGET_BENCH ?= qoibench
@@ -15,7 +15,7 @@ $(TARGET_BENCH):$(TARGET_BENCH).c
 
 conv: $(TARGET_CONV)
 $(TARGET_CONV):$(TARGET_CONV).c
-	$(CC) $(CFLAGS_CONV) $(CFLAGS) $(TARGET_CONV).c -o $(TARGET_CONV)
+	$(CC) $(CFLAGS_CONV) $(CFLAGS) $(TARGET_CONV).c -o $(TARGET_CONV) -lm
 
 .PHONY: clean
 clean:
