From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andrew Kaster <akaster@serenityos.org>
Date: Tue, 13 Feb 2024 12:31:25 -0700
Subject: [PATCH] tools: Pass product name to update-metadata fallback path

---
 tools/wpt/update.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/tools/wpt/update.py b/tools/wpt/update.py
index 4dba7e69df8376e7e8566525ab083e6e0348a92f..17cd74e37fd79b839167feedc105ce986e89322e 100644
--- a/tools/wpt/update.py
+++ b/tools/wpt/update.py
@@ -44,7 +44,7 @@ def update_expectations(_, **kwargs):
     update_properties = metadata.get_properties(properties_file=kwargs["properties_file"],
                                                 extra_properties=kwargs["extra_property"],
                                                 config=kwargs["config"],
-                                                product=kwargs["product"])
+                                                product=kwargs["product"].name)
 
     manifest_update(kwargs["test_paths"])
     metadata.update_expected(kwargs["test_paths"],
