From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Sun, 11 Apr 2021 23:15:42 +0200
Subject: [PATCH] Don't use fcntl on serenity

---
 include/unixconf.h | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/include/unixconf.h b/include/unixconf.h
index e7ee915659b4b791ca286494514a7795d2a82f88..5f1e92a8335f6e81b580325463a0c5cff8a20c15 100644
--- a/include/unixconf.h
+++ b/include/unixconf.h
@@ -240,8 +240,10 @@
  * Comment out the USE_FCNTL if for some reason you have a strange
  * OS/filesystem combination for which fcntl(2) does not work. */
 #ifdef POSIX_TYPES
+#ifndef __serenity__
 #define USE_FCNTL
 #endif
+#endif
 
 /*
  * The remainder of the file should not need to be changed.
