From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brendan Coles <bcoles@gmail.com>
Date: Thu, 28 Oct 2021 11:27:43 +0000
Subject: [PATCH] Remove android-specific SDL hint

---
 src/input.cpp | 4 ----
 1 file changed, 4 deletions(-)

diff --git a/src/input.cpp b/src/input.cpp
index b5ae21e24b802245e4d8a102b2ff820d4ceeea94..b6ac4947cc5f7991b7f8b79da34da24aee5d048d 100644
--- a/src/input.cpp
+++ b/src/input.cpp
@@ -89,10 +89,6 @@ void input_init(void)
   // It also keeps the mouse within the window in software mode.
   SDL_SetHint(SDL_HINT_MOUSE_RELATIVE_MODE_WARP, "1");
 
-  // TODO: don't attempt to simulate mouse events from touch events -
-  // fake mouse events often are de-centered
-  SDL_SetHint(SDL_HINT_ANDROID_SEPARATE_MOUSE_AND_TOUCH, "0");
-
   /* Touch devices */
   {
     int i;
