From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Justin Brown <behindcurtain3@gmail.com>
Date: Tue, 8 Aug 2023 18:47:06 -0700
Subject: [PATCH] Skip including execinfo.h for Serenity

---
 src/core/backtrace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/backtrace.c b/src/core/backtrace.c
index 19f9b134c8ff41a1939ed04db0b2c8a94ec4be0b..28f1ff15d68a49514e6c9827833802897f22b8b8 100644
--- a/src/core/backtrace.c
+++ b/src/core/backtrace.c
@@ -4,7 +4,7 @@
 
 #if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__OpenBSD__) && \
     !defined(__vita__) && !defined(__SWITCH__) && !defined(__ANDROID__) && \
-    !defined(__HAIKU__) && !defined(__EMSCRIPTEN__) && \
+    !defined(__HAIKU__) && !defined(__EMSCRIPTEN__) && !defined(__serenity__) && \
     (!defined(__ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__) || __ENVIRONMENT_MAC_OS_X_VERSION_MIN_REQUIRED__ >= 1050)
 
 #include <execinfo.h>
