From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Justin Brown <behindcurtain3@gmail.com>
Date: Tue, 8 Aug 2023 18:47:06 -0700
Subject: [PATCH] Skip including execinfo.h for Serenity

---
 src/core/backtrace.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/core/backtrace.c b/src/core/backtrace.c
index 025bac7c7cc5058a8372ecfc3862ac31a6cb1310..693bb5c1ff0c1b43d58803d51c95a3ae9e5165f0 100644
--- a/src/core/backtrace.c
+++ b/src/core/backtrace.c
@@ -4,7 +4,7 @@
 
 #if defined(__GNUC__) && !defined(__MINGW32__) && !defined(__OpenBSD__) && \
     !defined(__vita__) && !defined(__SWITCH__) && !defined(__ANDROID__) && \
-    !defined(__HAIKU__) && !defined(__EMSCRIPTEN__)
+    !defined(__HAIKU__) && !defined(__EMSCRIPTEN__) && !defined(__serenity__)
 
 #include <execinfo.h>
 
