diff --git a/src/options.c b/src/options.c
index 1bfe473..ba13dab 100644
--- a/src/options.c
+++ b/src/options.c
@@ -130,7 +130,7 @@ static int ini_process_##data_type(const char* curr_name, const char* value, con
 	if(strcasecmp(curr_name, option_name) == 0) { \
 		if (strcasecmp(value, "default") != 0) { \
 			int named_value = ini_get_named_value(value, value_names); \
-			*target = (named_value == INI_NO_VALID_NAME) ? ((data_type) strtoimax(value, NULL, 0)) : ((data_type) named_value); \
+			*target = (named_value == INI_NO_VALID_NAME) ? ((data_type) strtol(value, NULL, 0)) : ((data_type) named_value); \
 		} \
 		return 1; /* finished; don't look for more possible options that curr_name can be */ \
 	} \
