From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Dan MacDonald <allcoms@gmail.com>
Date: Sun, 8 Dec 2019 17:25:35 +0000
Subject: [PATCH] Disable backtracing

---
 src/common.cpp | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/src/common.cpp b/src/common.cpp
index 33df846..4dd9e9c 100644
--- a/src/common.cpp
+++ b/src/common.cpp
@@ -1,6 +1,6 @@
 #include "common.h"
 
-#include <execinfo.h>
+// #include <execinfo.h>
 #include <signal.h>
 
 //
@@ -96,7 +96,7 @@ void errno_fail(int errno_val, char const *format, ...) {
 static void fatal_signal_handler(int sig) {
     // Use non-async-signal-safe functions. Likely to work in practice.
 
-    static void *backtrace_buffer[100];
+    /*static void *backtrace_buffer[100];
     static char addr2line_cmd_buf[100];
 
     fprintf(stderr, "caught fatal signal '%s'. Backtrace:\n\n", strsignal(sig));
@@ -123,7 +123,7 @@ static void fatal_signal_handler(int sig) {
             fputs("failed to write address to addr2line\n", stderr);
             abort();
         }
-
+    */
     abort();
 }
 
