--- ./src/common.cpp	2016-09-18 10:02:04.000000000 +0100
+++ ./src/common-serenity.cpp	2019-11-02 16:58:54.509792761 +0000
@@ -1,6 +1,6 @@
 #include "common.h"
 
-#include <execinfo.h>
+// #include <execinfo.h>
 #include <signal.h>
 
 //
@@ -96,7 +96,7 @@
 static void fatal_signal_handler(int sig) {
     // Use non-async-signal-safe functions. Likely to work in practice.
 
-    static void *backtrace_buffer[100];
+    /*static void *backtrace_buffer[100];
     static char addr2line_cmd_buf[100];
 
     fprintf(stderr, "caught fatal signal '%s'. Backtrace:\n\n", strsignal(sig));
@@ -124,7 +124,7 @@
             abort();
         }
 
-    abort();
+    abort();*/
 }
 
 static void install_fatal_signal_handler(int sig) {
