--- ./Makefile	2016-09-18 10:02:04.000000000 +0100
+++ ./Makefile-serenity	2019-11-04 13:43:24.842907912 +0000
@@ -58,7 +58,7 @@
 objects     = $(c_objects) $(cpp_objects)
 deps        = $(addprefix $(BUILD_DIR)/,$(c_sources:=.d) $(cpp_sources:=.d))
 
-LDLIBS := $(shell sdl2-config --libs) -lrt
+LDLIBS := -lSDL2 -lgui -lipc -ldraw -lcore -lpthread
 
 ifeq ($(INCLUDE_DEBUGGER),1)
     LDLIBS += -lreadline
@@ -105,12 +105,12 @@
 endif
 
 ifneq ($(findstring debug,$(CONF)),)
-    compile_flags += -ggdb
+    compile_flags += -ggdb -fno-rtti
 endif
 ifneq ($(findstring release,$(CONF)),)
     # Including -Ofast when linking (by including $(optimizations)) gives a
     # different binary size. Might be worth investigating why.
-    compile_flags += $(optimizations) -DOPTIMIZING
+    compile_flags += $(optimizations) -DOPTIMIZING -fno-rtti
     link_flags    += $(optimizations) -fuse-linker-plugin
 endif
 
@@ -175,3 +175,6 @@
 
 .PHONY: clean
 clean: ; $(q)-rm -rf $(BUILD_DIR)
+
+install:
+	install $(BUILD_DIR)/${EXECUTABLE} ${DESTDIR}/bin/${EXECUTABLE}
