From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: demostanis <demostanis@protonmail.com>
Date: Sat, 10 Sep 2022 20:19:23 +0200
Subject: [PATCH] Use /proc/self/exe on SerenityOS

---
 src/file.cc | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/file.cc b/src/file.cc
index 58eb92844051059e66a7f0cc47f70091b821f81f..0531938bf0dd7bf1a8ac3d7be7bec8b629aaabd1 100644
--- a/src/file.cc
+++ b/src/file.cc
@@ -621,7 +621,7 @@ FsStatus get_fs_status(StringView filename)
 String get_kak_binary_path()
 {
     char buffer[2048];
-#if defined(__linux__) or defined(__CYGWIN__) or defined(__gnu_hurd__)
+#if defined(__linux__) or defined(__CYGWIN__) or defined(__gnu_hurd__) or defined(__serenity__)
     ssize_t res = readlink("/proc/self/exe", buffer, 2048);
     kak_assert(res != -1);
     buffer[res] = '\0';
