diff -Naur libgcrypt-1.9.2/build-aux/config.sub libgcrypt-1.9.2.serenity/build-aux/config.sub
--- libgcrypt-1.9.2/build-aux/config.sub	2019-06-23 17:35:08.000000000 +0200
+++ libgcrypt-1.9.2.serenity/build-aux/config.sub	2021-04-14 01:14:04.895316310 +0200
@@ -1381,7 +1381,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* | -serenity* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
