From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Wed, 16 Jun 2021 11:23:34 +0200
Subject: [PATCH] Make it use software rendering

---
 src/res.c | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/res.c b/src/res.c
index ef4945ab4d115c8d2877be4f74b4596e4cb37c77..6c46184f12e006a8a1258df5ed7772aa2407d88b 100644
--- a/src/res.c
+++ b/src/res.c
@@ -89,6 +89,8 @@ Mix_Music *bgms[AUDIO_BGM_SIZE];
 int soundsCount;
 Mix_Chunk *sounds[AUDIO_SOUND_SIZE];
 
+#define SOFTWARE_ACC
+
 bool init() {
   // Initialization flag
   bool success = true;
