From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bertalan <dani@danielbertalan.dev>
Date: Fri, 5 Nov 2021 22:32:10 +0100
Subject: [PATCH] Add some compat macros for serenity

---
 portable.h | 10 +++++++++-
 1 file changed, 9 insertions(+), 1 deletion(-)

diff --git a/portable.h b/portable.h
index 5c86edd..c840c69 100644
--- a/portable.h
+++ b/portable.h
@@ -61,7 +61,7 @@
 #define _PW_NAME_LEN	MAXLOGNAME
 #elif defined(__sun)
 #define _PW_NAME_LEN	LOGNAME_MAX
-#elif defined(__hpux)
+#elif defined(__hpux) || defined(__serenity__)
 #define _PW_NAME_LEN	8
 #else
 #define _PW_NAME_LEN	MAXLOGNAME - 1
@@ -116,6 +116,14 @@
 #define nice(x) (int)0
 #endif /* __HAIKU__ */
 
+#ifdef __serenity__
+#define _PATH_DEFPATH "/usr/bin:/usr/local/bin"
+#define _PATH_STDPATH "/usr/bin:/bin:/usr/sbin:/sbin"
+#define _CS_PATH 1
+#define WCOREFLAG 0200
+#define WCOREDUMP(x) ((x) & WCOREFLAG)
+#endif /* __serenity__ */
+
 #ifndef HAVE_SETRESGID
 #define setresgid(x, y, z)	setgid(x); setegid(y); setgid(z)
 #endif /* !HAVE_SETRESGID */
