From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Daniel Bertalan <dani@danielbertalan.dev>
Date: Fri, 5 Nov 2021 22:32:10 +0100
Subject: [PATCH] Add pconfig.h

---
 pconfig.h | 30 ++++++++++++++++++++++++++++++
 1 file changed, 30 insertions(+)
 create mode 100644 pconfig.h

diff --git a/pconfig.h b/pconfig.h
new file mode 100644
index 0000000..3274014
--- /dev/null
+++ b/pconfig.h
@@ -0,0 +1,30 @@
+#ifndef __dead
+#define __dead __attribute__((__noreturn__))
+#endif
+
+/* #define __attribute__(x) */
+
+#define HAVE_ASPRINTF
+/* #define HAVE_CONFSTR */
+/* #define NO_CURSES */
+#define HAVE_NCURSES
+#define issetugid(x) 0
+/* #define HAVE_ISSETUGID */
+#define HAVE_GETAUXVAL
+/* #define HAVE_PLEDGE */
+/* #define HAVE_REALLOCARRAY */
+#define HAVE_SETRESGID
+#define HAVE_SETRESUID
+/* #define HAVE_SIG_T */
+/* #define HAVE_SRAND_DETERMINISTIC */
+#define HAVE_ST_MTIM
+/* #define HAVE_ST_MTIMESPEC */
+/* #define HAVE_STRAVIS */
+/* #define HAVE_STRLCAT */
+#define HAVE_STRLCPY
+/* #define HAVE_STRTONUM */
+/* #define HAVE_STRUNVIS */
+#define HAVE_SIGLIST
+#define HAVE_SIGNAME
+#define HAVE_TIMERADD
+#define HAVE_TIMERCLEAR
