diff --git a/pconfig.h b/pconfig.h
new file mode 100644
index 0000000..2dd229b
--- /dev/null
+++ b/pconfig.h
@@ -0,0 +1,30 @@
+#ifndef __dead
+#define __dead __attribute__((__noreturn__))
+#endif
+
+/* #define __attribute__(x) */
+
+#define HAVE_ASPRINTF
+/* #define HAVE_CONFSTR */
+/* #define NO_CURSES */
+#define HAVE_NCURSES
+#define issetugid(x) 0
+/* #define HAVE_ISSETUGID */
+#define HAVE_GETAUXVAL
+/* #define HAVE_PLEDGE */
+/* #define HAVE_REALLOCARRAY */
+#define HAVE_SETRESGID
+#define HAVE_SETRESUID
+/* #define HAVE_SIG_T */
+/* #define HAVE_SRAND_DETERMINISTIC */
+#define HAVE_ST_MTIM
+/* #define HAVE_ST_MTIMESPEC */
+/* #define HAVE_STRAVIS */
+/* #define HAVE_STRLCAT */
+#define HAVE_STRLCPY
+/* #define HAVE_STRTONUM */
+/* #define HAVE_STRUNVIS */
+#define HAVE_SIGLIST
+#define HAVE_SIGNAME
+#define HAVE_TIMERADD
+#define HAVE_TIMERCLEAR
+#define HAVE_TIMERSUB
