From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Beckett Normington <beckett@b0ba.dev>
Date: Sun, 27 Aug 2023 20:52:19 -0400
Subject: [PATCH] libs/whereami: Add support for Serenity

---
 src/libs/whereami/whereami.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/libs/whereami/whereami.c b/src/libs/whereami/whereami.c
index 390ea5fef36e512cca78b1fcea1464ae2d14a6d2..9140bfc520a758fcd3865ca811e12a9c6ce199b8 100644
--- a/src/libs/whereami/whereami.c
+++ b/src/libs/whereami/whereami.c
@@ -160,7 +160,7 @@ int WAI_PREFIX(getModulePath)(char* out, int capacity, int* dirname_length)
   return length;
 }
 
-#elif defined(__linux__) || defined(__CYGWIN__) || defined(__sun) || defined(WAI_USE_PROC_SELF_EXE)
+#elif defined(__linux__) || defined(__CYGWIN__) || defined(__sun) || defined(WAI_USE_PROC_SELF_EXE) || defined(__serenity__)
 
 #include <stdio.h>
 #include <stdlib.h>
