From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Thu, 4 May 2023 22:35:56 +0200
Subject: [PATCH] [bios.cpp] Serenity has time.h

---
 src/ints/bios.cpp | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/ints/bios.cpp b/src/ints/bios.cpp
index 4ccbe1609ba11d5d69f59ed15735e5a88cbb162a..257f8ce10bf5d5b8c9f145647764d8542e55fc51 100644
--- a/src/ints/bios.cpp
+++ b/src/ints/bios.cpp
@@ -33,7 +33,7 @@
 #include "serialport.h"
 #include <time.h>
 
-#if defined(HAVE_CLOCK_GETTIME) && !defined(WIN32)
+#if defined(HAVE_CLOCK_GETTIME) || defined(__serenity__) && !defined(WIN32)
 //time.h is already included
 #else
 #include <sys/timeb.h>
@@ -496,7 +496,7 @@ static void BIOS_HostTimeSync() {
 	Bit32u milli = 0;
 	// TODO investigate if clock_gettime and ftime can be replaced
 	// by using C++11 chrono
-#if defined(HAVE_CLOCK_GETTIME) && !defined(WIN32)
+#if defined(HAVE_CLOCK_GETTIME) || defined(__serenity__) && !defined(WIN32)
 	struct timespec tp;
 	clock_gettime(CLOCK_REALTIME,&tp);
 
