From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Fri, 18 Jun 2021 15:06:37 +0200
Subject: [PATCH] Manually hardcode SDL2 library name

---
 configure.ac | 6 ++----
 1 file changed, 2 insertions(+), 4 deletions(-)

diff --git a/configure.ac b/configure.ac
index 28e9281..1cc3691 100644
--- a/configure.ac
+++ b/configure.ac
@@ -45,10 +45,8 @@ CFLAGS="$CFLAGS $PTHREAD_CFLAGS"
 
 dnl Check for SDL
 SDL_VERSION=2.0.2
-AM_PATH_SDL($SDL_VERSION,
-            :,
-	    AC_MSG_ERROR([*** SDL version $SDL_VERSION not found!])
-)
+SDL_LIBS="-lSDL2"
+SDL_CFLAGS=""
 PRESDL_LIBS="$LIBS"
 LIBS="$LIBS $SDL_LIBS"
 CPPFLAGS="$CPPFLAGS $SDL_CFLAGS"
