From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Fri, 5 May 2023 07:36:53 +0200
Subject: [PATCH] [meson.build] Add 'serenity' to list of os families

---
 meson.build | 1 +
 1 file changed, 1 insertion(+)

diff --git a/meson.build b/meson.build
index d0794072239b7b48597cf299d9297fbc1a59d983..75b3c752c34e892e4215065d8bf32cafad9952c2 100644
--- a/meson.build
+++ b/meson.build
@@ -50,6 +50,7 @@ os_family_name = {
   'netbsd'    : 'BSD',
   'openbsd'   : 'BSD',
   'dragonfly' : 'BSD',
+  'serenity'  : 'LINUX',
 }.get(host_machine.system(), 'UNKNOWN_OS')
 conf_data.set(os_family_name, 1)
 
