From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Kenneth Myhra <kennethmyhra@gmail.com>
Date: Fri, 5 May 2023 07:45:29 +0200
Subject: [PATCH] [libserial.cpp] Patch out ioctl flags we currently do not
 support

---
 src/hardware/serialport/libserial.cpp | 16 ++++++++++++++++
 1 file changed, 16 insertions(+)

diff --git a/src/hardware/serialport/libserial.cpp b/src/hardware/serialport/libserial.cpp
index a3dad0980308089a0869001cac31b5c7661dd5df..7899e9cf9bfed065831a23d45aaeac907eee7b97 100644
--- a/src/hardware/serialport/libserial.cpp
+++ b/src/hardware/serialport/libserial.cpp
@@ -360,6 +360,7 @@ void SERIAL_getErrorString(char* buffer, size_t length) {
 }
 
 int SERIAL_getmodemstatus(COMPORT port) {
+#ifndef __serenity__
 	long flags = 0;
 	ioctl (port->porthandle, TIOCMGET, &flags);
 	int retval = 0;
@@ -368,6 +369,9 @@ int SERIAL_getmodemstatus(COMPORT port) {
 	if (flags & TIOCM_RI) retval |= SERIAL_RI;
 	if (flags & TIOCM_CD) retval |= SERIAL_CD;
 	return retval;
+#else
+	return 0;
+#endif
 }
 
 bool SERIAL_sendchar(COMPORT port, char data) {
@@ -463,17 +467,29 @@ bool SERIAL_setCommParameters(COMPORT port,
 }
 
 void SERIAL_setBREAK(COMPORT port, bool value) {
+#ifdef __serenity__
+	// Do nothing.
+#else
 	ioctl(port->porthandle, value?TIOCSBRK:TIOCCBRK);
+#endif
 }
 
 void SERIAL_setDTR(COMPORT port, bool value) {
+#ifdef __serenity__
+	// Do nothing.
+#else
 	long flag = TIOCM_DTR;
 	ioctl(port->porthandle, value?TIOCMBIS:TIOCMBIC, &flag);
+#endif
 }
 
 void SERIAL_setRTS(COMPORT port, bool value) {
+#ifdef __serenity__
+	// Do nothing.
+#else
 	long flag = TIOCM_RTS;
 	ioctl(port->porthandle, value?TIOCMBIS:TIOCMBIC, &flag);
+#endif
 }
 
 #endif
