diff -Naur mc-4.8.27/config/config.sub mc-4.8.27.serenity/config/config.sub
--- mc-4.8.27/config/config.sub	2021-08-15 16:02:28.000000000 +0200
+++ mc-4.8.27.serenity/config/config.sub	2021-08-18 22:35:10.630187143 +0200
@@ -1333,7 +1333,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
-	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* \
 	     | sym* | kopensolaris* | plan9* \
