From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Brian Callahan <bcallah@openbsd.org>
Date: Sun, 10 May 2020 12:00:03 -0400
Subject: [PATCH] Add an install target

---
 Makefile | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/Makefile b/Makefile
index 93cac80..5d68e52 100644
--- a/Makefile
+++ b/Makefile
@@ -6,5 +6,9 @@ OBJS =	str.o tr.o
 all: ${OBJS}
 	${CC} ${LDFLAGS} -o ${PROG} ${OBJS} -L/usr/local/lib -lpuffy
 
+install:
+	install -c -m 755 tr ${DESTDIR}/usr/local/bin
+	install -c -m 644 tr.1 ${DESTDIR}/usr/local/share/man/man1
+
 clean:
 	rm -f ${PROG} ${OBJS}
