From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Tim Ledbetter <timledbetter@gmail.com>
Date: Sat, 5 Aug 2023 09:55:31 +0100
Subject: [PATCH] Use SDL software renderer

---
 doomgeneric/doomgeneric_sdl.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/doomgeneric/doomgeneric_sdl.c b/doomgeneric/doomgeneric_sdl.c
index fd4236a83be0f99fce777c494f3b92faa341567f..725339d818fd6c969f431c181eb5c69fd7c765c6 100644
--- a/doomgeneric/doomgeneric_sdl.c
+++ b/doomgeneric/doomgeneric_sdl.c
@@ -141,7 +141,7 @@ void DG_Init(){
                             );
 
   // Setup renderer
-  renderer =  SDL_CreateRenderer( window, -1, SDL_RENDERER_ACCELERATED);
+  renderer =  SDL_CreateRenderer( window, -1, SDL_RENDERER_SOFTWARE);
   // Clear winow
   SDL_RenderClear( renderer );
   // Render the rect to the screen
