From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gunnar@beutner.name>
Date: Sun, 23 Oct 2022 12:00:56 +0200
Subject: [PATCH] Fix building Boost.Python

---
 tools/build/src/tools/python.jam | 1 +
 1 file changed, 1 insertion(+)

diff --git a/tools/build/src/tools/python.jam b/tools/build/src/tools/python.jam
index b0f6c2c6c29d03c1f86ff650158a4b04b1e0822c..824d1455dd086d36f9e561e961bab1fb73ece37c 100644
--- a/tools/build/src/tools/python.jam
+++ b/tools/build/src/tools/python.jam
@@ -716,6 +716,7 @@ local rule configure ( version ? : cmd-or-prefix ? : includes * : libraries ? :
     local interpreter-cmd ;
 
     local target-os = [ feature.get-values target-os : $(condition) ] ;
+    target-os = "serenity" ; # FIXME
     target-os ?= [ feature.defaults target-os ] ;
     target-os = $(target-os:G=) ;
 
