From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Gunnar Beutner <gbeutner@serenityos.org>
Date: Wed, 16 Jun 2021 11:08:32 +0200
Subject: [PATCH] Include sys/select.h

---
 SDLnetsys.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/SDLnetsys.h b/SDLnetsys.h
index 6f6dfae..bd9139d 100644
--- a/SDLnetsys.h
+++ b/SDLnetsys.h
@@ -46,6 +46,7 @@ typedef int socklen_t;
 #include <iphlpapi.h>
 #else /* UNIX */
 #include <sys/types.h>
+#include <sys/select.h>
 #ifdef __FreeBSD__
 #include <sys/socket.h>
 #endif
