From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: EWouters <6179932+EWouters@users.noreply.github.com>
Date: Sat, 2 Sep 2023 19:05:41 +0200
Subject: [PATCH] Include stdio.h in `programs/util.h`

---
 programs/util.h | 1 +
 1 file changed, 1 insertion(+)

diff --git a/programs/util.h b/programs/util.h
index 8234646bf3dc61e482f8331795b8bf54a7aa0746..4a8596169126b19f7612a743a0b4576c5c97c414 100644
--- a/programs/util.h
+++ b/programs/util.h
@@ -24,6 +24,7 @@ extern "C" {
 #include <sys/types.h>    /* stat, utime */
 #include <sys/stat.h>     /* stat, chmod */
 #include "../lib/common/mem.h"          /* U64 */
+#include <stdio.h>
 
 
 /*-************************************************************
