--- zstd-1.5.1/programs/platform.h.orig	2022-01-08 01:30:53.468728340 +0000
+++ zstd-1.5.1/programs/platform.h	2022-01-08 01:31:01.480646197 +0000
@@ -98,7 +98,11 @@
 #    endif
 #    include <unistd.h>  /* declares _POSIX_VERSION */
 #    if defined(_POSIX_VERSION)  /* POSIX compliant */
-#      define PLATFORM_POSIX_VERSION _POSIX_VERSION
+#      if defined(__serenity__)
+#        define PLATFORM_POSIX_VERSION 1
+#      else
+#        define PLATFORM_POSIX_VERSION _POSIX_VERSION
+#      endif
 #    else
 #      define PLATFORM_POSIX_VERSION 1
 #    endif
