From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Andreas Kling <kling@serenityos.org>
Date: Mon, 8 Mar 2021 13:15:35 +0100
Subject: [PATCH] Skip building the playmus and playwave utilities

---
 Makefile.in | 16 +---------------
 1 file changed, 1 insertion(+), 15 deletions(-)

diff --git a/Makefile.in b/Makefile.in
index 006a8fb..8cbee0c 100644
--- a/Makefile.in
+++ b/Makefile.in
@@ -37,8 +37,6 @@ SDL_LIBS = @SDL_LIBS@
 TARGET  = libSDL2_mixer.la
 OBJECTS = @OBJECTS@
 VERSION_OBJECTS = @VERSION_OBJECTS@
-PLAYWAVE_OBJECTS = @PLAYWAVE_OBJECTS@
-PLAYMUS_OBJECTS = @PLAYMUS_OBJECTS@
 
 SRC_DIST =			\
 	.gitmodules		\
@@ -64,8 +62,6 @@ SRC_DIST =			\
 	external/download.sh	\
 	include/SDL_mixer.h	\
 	mingw			\
-	playmus.c		\
-	playwave.c		\
 	sdl2_mixer-config-version.cmake.in \
 	sdl2_mixer-config.cmake.in \
 	src			\
@@ -80,7 +76,7 @@ LT_REVISION = @LT_REVISION@
 LT_EXTRA    = @LT_EXTRA@
 LT_LDFLAGS  = -no-undefined -rpath $(libdir) -release $(LT_RELEASE) -version-info $(LT_CURRENT):$(LT_REVISION):$(LT_AGE) $(LT_EXTRA)
 
-all: $(srcdir)/configure Makefile $(objects)/$(TARGET) $(objects)/playwave$(EXE) $(objects)/playmus$(EXE)
+all: $(srcdir)/configure Makefile $(objects)/$(TARGET)
 
 $(srcdir)/configure: $(srcdir)/configure.ac
 	@echo "Warning, configure is out of date, please re-run autogen.sh"
@@ -97,12 +93,6 @@ $(objects)/.created:
 $(objects)/$(TARGET): $(OBJECTS) $(VERSION_OBJECTS)
 	$(LIBTOOL) --mode=link $(CC) -o $@ $(OBJECTS) $(VERSION_OBJECTS) $(LDFLAGS) $(EXTRA_LDFLAGS) $(LT_LDFLAGS) $(SDL_LIBS) $(MIXER_LIBS)
 
-$(objects)/playwave$(EXE): $(objects)/playwave.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playwave.lo $(SDL_CFLAGS) $(objects)/$(TARGET) $(SDL_LIBS) $(LDFLAGS)
-
-$(objects)/playmus$(EXE): $(objects)/playmus.lo $(objects)/$(TARGET)
-	$(LIBTOOL) --mode=link $(CC) -o $@ $(objects)/playmus.lo $(SDL_CFLAGS) $(objects)/$(TARGET) $(SDL_LIBS) $(LDFLAGS)
-
 install: all install-hdrs install-lib #install-bin
 install-hdrs:
 	$(SHELL) $(auxdir)/mkinstalldirs $(includedir)/SDL2
@@ -120,8 +110,6 @@ install-lib: $(objects)/$(TARGET)
 	$(LIBTOOL) --mode=install $(INSTALL) $(objects)/$(TARGET) $(libdir)/$(TARGET)
 install-bin:
 	$(SHELL) $(auxdir)/mkinstalldirs $(bindir)
-	$(LIBTOOL) --mode=install $(INSTALL) -m 755 $(objects)/playwave$(EXE) $(bindir)/playwave$(EXE)
-	$(LIBTOOL) --mode=install $(INSTALL) -m 755 $(objects)/playmus$(EXE) $(bindir)/playmus$(EXE)
 
 uninstall: uninstall-hdrs uninstall-lib uninstall-bin
 uninstall-hdrs:
@@ -139,8 +127,6 @@ uninstall-hdrs:
 uninstall-lib:
 	$(LIBTOOL) --mode=uninstall rm -f $(libdir)/$(TARGET)
 uninstall-bin:
-	rm -f $(bindir)/playwave$(EXE)
-	rm -f $(bindir)/playmus$(EXE)
 
 clean:
 	rm -rf $(objects)
