diff -ru SDL2_mixer-2.0.4/build-scripts/config.sub SDL2_mixer-2.0.4-patched/build-scripts/config.sub
--- SDL2_mixer-2.0.4/build-scripts/config.sub	2018-10-31 15:59:00.000000000 +0100
+++ SDL2_mixer-2.0.4-patched/build-scripts/config.sub	2021-03-07 17:50:05.149501049 +0100
@@ -1408,7 +1408,7 @@
 	# The portable systems comes first.
 	# Each alternative MUST END IN A *, to match a version number.
 	# -sysv* is not here because it comes later, after sysvr4.
-	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -irix* \
+	-gnu* | -bsd* | -mach* | -minix* | -genix* | -ultrix* | -serenity* | -irix* \
 	      | -*vms* | -sco* | -esix* | -isc* | -aix* | -cnk* | -sunos | -sunos[34]*\
 	      | -hpux* | -unos* | -osf* | -luna* | -dgux* | -auroraux* | -solaris* \
 	      | -sym* | -kopensolaris* | -plan9* \
