--- libarchive-3.4.0/build/autoconf/config.sub.orig	2019-10-03 23:14:02.281000072 +0100
+++ libarchive-3.4.0/build/autoconf/config.sub	2019-10-03 23:14:36.896816822 +0100
@@ -2,7 +2,7 @@
 # Configuration validation subroutine script.
 #   Copyright 1992-2019 Free Software Foundation, Inc.
 
-timestamp='2019-05-23'
+timestamp='2019-10-12'
 
 # This file is free software; you can redistribute it and/or modify it
 # under the terms of the GNU General Public License as published by
@@ -1342,6 +1342,7 @@
 	# Each alternative MUST end in a * to match a version number.
 	# sysv* is not here because it comes later, after sysvr4.
 	gnu* | bsd* | mach* | minix* | genix* | ultrix* | irix* \
+	     | serenity* \
 	     | *vms* | esix* | aix* | cnk* | sunos | sunos[34]*\
 	     | hpux* | unos* | osf* | luna* | dgux* | auroraux* | solaris* 
 	     | sym* | kopensolaris* | plan9* \
