--- libarchive-3.4.0/libarchive/archive_read_disk_posix.c	2019-06-12 18:05:16.000000000 -0300
+++ libarchive-3.4.0-new/libarchive/archive_read_disk_posix.c	2019-10-12 22:30:12.774683968 -0300
@@ -1984,7 +1984,7 @@
 #  else
 		/* No way to get a trusted value of maximum filename
 		 * length. */
-		t->current_filesystem->name_max = PATH_MAX;
+		t->current_filesystem->name_max = 4096; /* Toolchain workaround */
 #  endif /* NAME_MAX */
 #  if defined(_PC_NAME_MAX)
 	else
