From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: Rodrigo Tobar <rtobarc@gmail.com>
Date: Wed, 8 Sep 2021 00:13:16 +0800
Subject: [PATCH] Make configure pass all dependency checks for serenity

---
 configure | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/configure b/configure
index b593652..28269b6 100755
--- a/configure
+++ b/configure
@@ -6166,6 +6166,10 @@ bsdi[45]*)
   lt_cv_file_magic_test_file=/shlib/libc.so
   ;;
 
+*serenity*)
+  lt_cv_deplibs_check_method=pass_all
+  ;;
+
 cygwin*)
   # func_win32_libid is a shell function defined in ltmain.sh
   lt_cv_deplibs_check_method='file_magic ^x86 archive import|^x86 DLL'
