From 54f21f92a7076d5e853f9ca37c57f34d54d01c67 Mon Sep 17 00:00:00 2001
From: Rodrigo Tobar <rtobarc@gmail.com>
Date: Wed, 8 Sep 2021 00:13:16 +0800
Subject: [PATCH 1/2] Teach config.sub about serenity

---
 config.sub | 8 +++++++-
 1 file changed, 7 insertions(+), 1 deletion(-)

diff --git a/config.sub b/config.sub
index c765b34..c524678 100755
--- a/config.sub
+++ b/config.sub
@@ -117,7 +117,7 @@ maybe_os=`echo $1 | sed 's/^\(.*\)-\([^-]*-[^-]*\)$/\2/'`
 case $maybe_os in
   nto-qnx* | linux-gnu* | linux-android* | linux-dietlibc | linux-newlib* | \
   linux-musl* | linux-uclibc* | uclinux-uclibc* | uclinux-gnu* | kfreebsd*-gnu* | \
-  knetbsd*-gnu* | netbsd*-gnu* | \
+  knetbsd*-gnu* | netbsd*-gnu* | *serenity* | \
   kopensolaris*-gnu* | \
   storm-chaos* | os2-emx* | rtmk-nova*)
     os=-$maybe_os
@@ -1277,6 +1277,9 @@ case $basic_machine in
 	we32k)
 		basic_machine=we32k-att
 		;;
+	*serenity*)
+		basic_machine="${SERENITY_ARCH}-pc"
+		;;
 	sh[1234] | sh[24]a | sh[24]aeb | sh[34]eb | sh[1234]le | sh[23]ele)
 		basic_machine=sh-unknown
 		;;
@@ -1404,6 +1407,9 @@ case $os in
 	-linux*)
 		os=`echo $os | sed -e 's|linux|linux-gnu|'`
 		;;
+	*-serenity*)
+		os=-serenity
+		;;
 	-sunos5*)
 		os=`echo $os | sed -e 's|sunos5|solaris2|'`
 		;;
-- 
2.36.1

